
<!DOCTYPE html>
<html lang="en">

<head>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-119794696-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'UA-119794696-1');
    </script>


    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="shortcut icon" href="//igv.org/web/img/favicon.ico">
    <title>ImmuneRegulation</title>

    <!-- Bootstrap CSS - NOT REQUIRED FOR IGV -->
    <link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    
    <!-- jQuery UI CSS -->
    <link rel="stylesheet" type="text/css" href="//ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/themes/smoothness/jquery-ui.css"/>

    <!-- Font Awesome CSS -->
    <link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css">

    <link rel="shortcut icon" href="//igv.org/web/img/favicon.ico">

    <!-- IGV CSS -->
    <link rel="stylesheet" type="text/css" href="https://igv.org/web/beta/igv-beta.css">

    <!-- DataTables CSS -->
   <link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/v/dt/dt-1.10.18/cr-1.5.0/fh-3.1.4/r-2.2.2/sl-1.3.0/datatables.min.css"/>

    <!--link rel="stylesheet" type="text/css" src="./css/jquery.dataTables.min.css"-->
    <link rel="stylesheet" type="text/css" src="./css/typeahead.css">

    <!-- jQuery JS -->
    <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
    <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/jquery-ui.min.js"></script>

    <!-- Tooltips -->
    <script src="./js/popper.min.js"></script>

    <!-- Bootstrap JS -  NOT REQUIRED FOR IGV -->
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>


    <!-- IGV JS  -->
    <script src="./js/igv-immuneregulation-Nov2019.js"></script>

    <script src="./js/savePng.js"></script>
    <script src="./js/jsPDF2.js"></script>
    <script src="./js/jsPDF.js"></script>
    <script src="./js/svg2pdf.js"></script>
    <script src="./js/saveSvgAsPng.js"></script>

    <script src="./js/savePdf.js"></script>
    <script src="./js/html2canvas.js"></script>
    <script src="./js/FileSaver.js"></script>

    <!-- D3 JS -->
    <script src='https://d3js.org/d3.v3.min.js'></script>

    <!-- Bootstrap Multi-Select -->

    <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-multiselect/0.9.13/js/bootstrap-multiselect.min.js" integrity="sha256-1fK87Pt9T7XXn/Vj3CWSfJlZkL5f73092A8YL7p7avc=" crossorigin="anonymous"></script>

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-multiselect/0.9.13/css/bootstrap-multiselect.css" integrity="sha256-O1jJhOIWWDc3wq75g7aXDl7aEQT2XPkieX/Mz/u4hQ0=" crossorigin="anonymous"/>    
    

    <link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
    
    <!-- DataTables JS -->
    <script type="text/javascript" src="https://cdn.datatables.net/v/dt/dt-1.10.18/cr-1.5.0/fh-3.1.4/r-2.2.2/sl-1.3.0/datatables.min.js"></script>
    
    <script src="https://cdn.rawgit.com/eligrey/canvas-toBlob.js/f1a01896135ab378aa5c0118eadd81da55e698d8/canvas-toBlob.js"></script>
    <script src="https://cdn.rawgit.com/eligrey/FileSaver.js/e9d941381475b5df8b7d7691013401e171014e89/FileSaver.min.js"></script>
    
    <script src="./js/type.js"></script>
    <script src="./js/genenames.js"></script>
    <script src="./js/uploadFile.js"></script>
    <script src="./js/uploadFile2.js"></script>
    <script src="./js/underscore-min.js"></script>
    <script src='https://cdnjs.cloudflare.com/ajax/libs/spin.js/2.0.1/spin.min.js'></script>

    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.min.js"></script>

    <script type="text/javascript">

        $( function() {
            $(document).tooltip({
              position: {
                my: "center bottom-20",
                at: "center top",
                using: function( position, feedback ) {
                  $( this ).css( position );
                  $( "<div>" )
                    .addClass( "arrow" )
                    .addClass( feedback.vertical )
                    .addClass( feedback.horizontal )
                    .appendTo( this );
                }
              }
            });
        } );

        $(document).ready(function() {
            $('#datasetT').multiselect({
                enableClickableOptGroups: true,
                enableCollapsibleOptGroups: true,
                buttonWidth: '350px',
                nonSelectedText: 'Choose dataset(s)'
            });
            $('.multiselect-container').find(".caret").click();

            $("body").tooltip({ selector: '[data-toggle=tooltip]' });

            $("#studyEntryRow").hide();

            $("#operationEntryRow").hide();

            //User Studies...
            $("#userListMain").hide();
            $("#userListCis").hide();
            $("#userListTrans").hide();

            $("#allStudies li:contains(User)").hide();


            //Update studyCountText 
            var allcheckboxes = document.querySelectorAll('input.studyopt');
            
            for(var i = 0; i < allcheckboxes.length; i++) {
                allcheckboxes[i].onclick = function() {
                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected"; 

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
                }
              
            } 
            

            //Emory Checkboxes 
            var emorycheckboxes = document.querySelectorAll('input.EmoryOption');
            var emorycheckall = document.getElementById('EmoryCheckAll');

            for(var i = 0; i < emorycheckboxes.length; i++) {
                emorycheckboxes[i].onclick = function() {
                    var emoryCheckCount = document.querySelectorAll('input.EmoryOption:checked').length;
                    emorycheckall.checked = emoryCheckCount > 0;
                    emorycheckall.indeterminate = emoryCheckCount > 0 && emoryCheckCount < emorycheckboxes.length;

                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
              }
            }

            emorycheckall.onclick = function() {
              for(var i = 0; i < emorycheckboxes.length; i++) {
                    emorycheckboxes[i].checked = this.checked;
              }

                var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";
                
                if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
            }

            //Mayo Checkboxes 
            var mayocheckboxes = document.querySelectorAll('input.MayoOption');
            var mayocheckall = document.getElementById('MayoCheckAll');

            for(var i = 0; i < mayocheckboxes.length; i++) {
                mayocheckboxes[i].onclick = function() {
                    var mayoCheckCount = document.querySelectorAll('input.MayoOption:checked').length;
                    mayocheckall.checked = mayoCheckCount > 0;
                    mayocheckall.indeterminate = mayoCheckCount > 0 && mayoCheckCount < mayocheckboxes.length;

                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
              }
            }

            mayocheckall.onclick = function() {
              for(var i = 0; i < mayocheckboxes.length; i++) {
                    mayocheckboxes[i].checked = this.checked;
              }

                var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
            } 


            //NIH Checkboxes 
            var nihcheckboxes = document.querySelectorAll('input.NihOption');
            var nihcheckall = document.getElementById('NihCheckAll');

            for(var i = 0; i < nihcheckboxes.length; i++) {
                nihcheckboxes[i].onclick = function() {
                    var nihCheckCount = document.querySelectorAll('input.NihOption:checked').length;
                    nihcheckall.checked = nihCheckCount > 0;
                    nihcheckall.indeterminate = nihCheckCount > 0 && nihCheckCount < nihcheckboxes.length;

                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
              }
            }

            nihcheckall.onclick = function() {
              for(var i = 0; i < nihcheckboxes.length; i++) {
                    nihcheckboxes[i].checked = this.checked;
              }

                var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
            }


            //Yale Checkboxes 
            var yalecheckboxes = document.querySelectorAll('input.YaleOption');
            var yalecheckall = document.getElementById('YaleCheckAll');

            for(var i = 0; i < yalecheckboxes.length; i++) {
                yalecheckboxes[i].onclick = function() {
                    var yaleCheckCount = document.querySelectorAll('input.YaleOption:checked').length;
                    yalecheckall.checked = yaleCheckCount > 0;
                    yalecheckall.indeterminate = yaleCheckCount > 0 && yaleCheckCount < yalecheckboxes.length;
                    
                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected"; 

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
              }
            }

            yalecheckall.onclick = function() {
                for(var i = 0; i < yalecheckboxes.length; i++) {
                    yalecheckboxes[i].checked = this.checked;
                    
                }
                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
            }


            //CHIKV Checkboxes 
            var chikvcheckboxes = document.querySelectorAll('input.ChikvOption');
            var chikvcheckall = document.getElementById('ChikvCheckAll');

            for(var i = 0; i < chikvcheckboxes.length; i++) {
                chikvcheckboxes[i].onclick = function() {
                    var chikvCheckCount = document.querySelectorAll('input.ChikvOption:checked').length;
                    chikvcheckall.checked = chikvCheckCount > 0;
                    chikvcheckall.indeterminate = chikvCheckCount > 0 && chikvCheckCount < chikvcheckboxes.length;

                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
              }
            }

            chikvcheckall.onclick = function() {
              for(var i = 0; i < chikvcheckboxes.length; i++) {
                    chikvcheckboxes[i].checked = this.checked;
              }

                var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
            }


            //ZIKV Checkboxes 
            var zikvcheckboxes = document.querySelectorAll('input.ZikvOption');
            var zikvcheckall = document.getElementById('ZikvCheckAll');

            for(var i = 0; i < zikvcheckboxes.length; i++) {
                zikvcheckboxes[i].onclick = function() {
                    var zikvCheckCount = document.querySelectorAll('input.ZikvOption:checked').length;
                    zikvcheckall.checked = zikvCheckCount > 0;
                    zikvcheckall.indeterminate = zikvCheckCount > 0 && zikvCheckCount < zikvcheckboxes.length;

                    var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                    document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                    if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
              }
            }

            zikvcheckall.onclick = function() {
              for(var i = 0; i < zikvcheckboxes.length; i++) {
                    zikvcheckboxes[i].checked = this.checked;
              }

                var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
                document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected";

                if(allcheckedcount > 0)
                        document.getElementById('deselectDiv').style.visibility = "visible";

                    else
                        document.getElementById('deselectDiv').style.visibility = "hidden";
            }



        });

        
    
    </script>

<style type="text/css">
    
.ui-tooltip, .arrow:after {
    background: white;
    border: 2px solid white;
  }
  .ui-tooltip {
    padding: 10px 20px;
    color: black;
    border-radius: 10px;
    font: bold 14px "Helvetica Neue", Sans-Serif;
    box-shadow: 0 0 7px black;
    opacity: 0.9;
  }
  .arrow {
    width: 70px;
    height: 16px;
    overflow: hidden;
    position: absolute;
    left: 50%;
    margin-left: -35px;
    bottom: -16px;
  }
  .arrow.top {
    top: -16px;
    bottom: auto;
  }
  .arrow.left {
    left: 20%;
  }
  .arrow:after {
    content: "";
    position: absolute;
    left: 20px;
    top: -20px;
    width: 25px;
    height: 25px;
    box-shadow: 6px 5px 9px -9px black;
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
    transform: rotate(45deg);
  }
  .arrow.top:after {
    bottom: -20px;
    top: auto;
  }


#custom-search-input {
    margin:0;
    margin-top: 10px;
    padding: 0;
}
 
#custom-search-input .search-query {
    padding-right: 3px;
    padding-right: 4px \9;
    padding-left: 3px;
    padding-left: 4px \9;
    /* IE7-8 doesn't have border-radius, so don't indent the padding */

    margin-bottom: 0;
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    border-radius: 3px;
}
 
#custom-search-input button {
    border: 0;
    background: none;
    /** belows styles are working good */
    padding: 2px 5px;
    margin-top: 2px;
    position: relative;
    left: -28px;
    /* IE7-8 doesn't have border-radius, so don't indent the padding */
    margin-bottom: 0;
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    border-radius: 3px;
    color:#D9230F;
}
 
.search-query:focus + button {
    z-index: 3;   
}

.graph-svg-component {
    background-color: AliceBlue;
}

.filter {
  display: inline-block;
}

#geneSearch, #uploadStudy, #saveModal, #vizListButton, #cisUpload, #cisSampleUpload, #transUpload, #transSampleUpload {
    background: #1974B8;  
}

#geneSearch:hover, #uploadStudy:hover, #saveModal:hover, #vizListButton:hover, #cisUpload:hover, #cisSampleUpload:hover, #transUpload:hover, #transSampleUpload:hover {
    background: #035b9c;   
}

#geneSearch:active, #vizListButton:active, #cisUpload:active, #cisSampleUpload:active, #transUpload:active, #transSampleUpload:active {
    background: #1974B8;   
}

.btn, .btn-primary {
    background: white;
    padding: 14px 24px;
    font-weight: 400;
    letter-spacing: 1px;
    border-radius: 0;
    border: 1 solid;
    border-color: #bbb;
}

.btn-group, .btn-group-vertical {
    margin-left: 10px;
}

.btn:focus, .btn:active:focus, .btn.active:focus {
    outline: 0 none;
    border: 2 solid;
    border-color: #bbb;
}

.btn-primary.active, .btn-primary:focus, .btn-primary:active, .open > .dropdown-toggle.btn-primary {
    background: lightgray;
    color: #000;
    
    box-shadow: none;
    border: 1 solid;
    border-color: #bbb;
}

.btn-primary:hover {
    background: #bbb;
    border: 1 solid;
    border-color: #bbb;
}

.btn-primary:active, .btn-primary.raised.active {
    background: lightgray;
    box-shadow: none;
    margin-bottom: -3px;
    margin-top: 3px;
    border: 1 solid;
    border-color: #bbb;
}

.btn-primary.disabled, .btn-primary.disabled:focus, .btn-primary.disabled:active {
    background: lightgray;
    color: white;
    opacity: .7;
    border: 1 solid;
    border-color: #bbb;
}

.multiselect {
    padding-bottom: 10px;
    padding-top: 10px;
}

.multiselect-clear-filter {
    padding-bottom: 5px;
    padding-top: 5px;
    padding-right: 10px;
    padding-left: 10px;
}

circle {
    fill-opacity: .7;
    fill: rgb(30,144,255);
    stroke: #aaa; 
}

circle:hover {
  fill: red;
  stroke: #ccc;
}

#circle-leg-1 {
    fill-opacity: .9;
    fill: rgb(200,200,200);
    stroke: #aaa;
}

#circle-leg-2 {
    fill-opacity: .9;
    fill: rgb(30,144,255);
    stroke: rgb(30,144,255);
}

#circle-leg-3 {
    fill-opacity: .9;
    fill: rgb(240,20,20);
    stroke: rgb(240,20,20);
}

#circle-leg-4 {
    fill-opacity: .9;
    fill: rgb(0,0,0);
    stroke: #aaa;
}

caption {
    font-size:medium;
    text-align: center;
    color: #035b9c;
    font-weight:500;

}

path.line {
  fill: none;
  stroke: #666;
  stroke-width: 1.5px;
}

path.area {
  fill: #e7e7e7;
}

.axis path,
.axis line {
    fill: none;
    stroke: black;
    shape-rendering: crispEdges;
}


.y.axis line,
.y.axis path {
  fill: none;
  stroke: #000;
}

.axis--y text {
  font: oblique 10px Georgia, serif;
}


.glyphicon + .tooltip > .tooltip-inner {
    /*position: absolute;*/
    background-color: white; 
      color: black; 
      border: 1px solid gray; 
      padding: 10px;
      font-size: 12px;
      height:auto;

}

.glyphicon + .tooltip {
   background-color: white;

}

.btn + .tooltip.bottom > .tooltip-inner {
    position: absolute;         
    text-align: left;         
    width: 160px;                    
    height: auto;                   
    padding: 10px;               
    font: 12px sans-serif;      
    background: lightsteelblue; 
    border: 0px;        
    /* box-shadow: 4px 4px 10px rgba(0, 0, 0, 0.4); */         
    pointer-events: none;  
    color:black;   
  }


div.tooltip {   
    position: absolute;         
    text-align: left;         
    width: 200px;                    
    height: auto;                   
    padding: 10px;               
    font: 12px sans-serif;      
    background: lightsteelblue; 
    border: 0px;        
    border-radius: 8px;
    /* box-shadow: 4px 4px 10px rgba(0, 0, 0, 0.4); */      
    pointer-events: none;  
    color:black;         
}

.glyphicon-arrow-down {
    font-size: large;
}

.glyphicon-arrow-down:hover {
    color: navy;
    font-size: x-large;
}

.glyphicon-arrow-up {
    font-size: large;
}

.glyphicon-arrow-up:hover {
    color: navy;
    font-size: x-large;
}

.badge {
    background: white;
    color: steelblue; 
    /*font-size: small; */
    padding-top: 5px;
    padding-bottom: 5px;
}

.navbar-nav:hover {
    background: #77b7ec;
}

.navbar-nav>.open {
    background: #1974B8;
}

.navbar-nav > li > a:active {
    font-size: larger;
}

.navbar-nav > li > a:focus {
    font-size: larger;

}

.navbar-inverse .navbar-nav>.open>a, .navbar-inverse .navbar-nav>.open>a:focus, .navbar-inverse .navbar-nav>.open>a:hover {
    background: #77b7ec;
}


</style>

</head>


<body role="document" style="background: #eee;">

<nav class="navbar navbar-inverse navbar-fixed-top" style= "margin-botom:0px;">

    <div class="container-fluid" style = "background: #1974B8;">
        
        <div id="navbar">
            <ul class="nav navbar-nav" style="margin-left:100px;">
                <li> 
                    <a href="index.html" style="padding:0px;"> <img src="images/Logo.png" alt="ImmuneRegulation" height="40" style="margin-right: 10px; margin-top:5px;"> </a> 
                </li>
                
                <li>   
                    <a href="index.html" class="navbar-brand" style="color: white;" role="button" aria-expanded="false">ImmuneRegulation </a>
                </li>
            </ul>

            <ul class="nav navbar-nav">
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle navbar-brand" style="color: white; margin-left: 10px;" data-toggle="dropdown" role="button" aria-expanded="false">Example Queries
                        <span class="caret"></span></a>
                    <ul class="dropdown-menu" style="margin-left: 10px; font-size:medium;" role="menu">
                        <li>
                            <a id="exQuery1" href="#" onclick="exampleQuery1();">Example 1 - Cis-eQTLs for JAK2 gene in monocyte exposure studies</a>
                        </li> 
                        <li>
                            <a id="exQuery2" href="#" onclick="exampleQuery2();">Example 2 - Trans-eQTLs for Cytokine Pathway genes in Whole-blood meta-analysis study</a>
                        </li> 
                        <li>
                            <a id="exQuery3" href="#" onclick="exampleQuery3();">Example 3 - Transcription Factor studies with hits for IL15RA gene </a>
                        </li>
                                              
                    </ul>
                </li>
            </ul>

            <!--ul class="nav navbar-nav">
                <li><a class="navbar-brand" style="color: white; margin-left: 10px;" href="metadata-all.html">Query Meta-data</a>
                    
                    </a>
                </li>
                
            </ul-->

            <ul class="nav navbar-nav">
                <li><a class="navbar-brand" style="color: white; margin-left: 10px;" href="tutorials.html">Tutorials</a>
                    
                    </a>
                </li>
                
            </ul>

            <ul class="nav navbar-nav">
                <li><a class="navbar-brand" style="color: white; margin-left: 10px;" href="feedback.html">Contact Us</a>
                    
                    </a>
                </li>
                
            </ul>

            <ul class="nav navbar-nav">
                <li><a class="navbar-brand" style="color: white; margin-left: 10px;" href="faq.html">FAQ</a>
                    
                    </a>
                </li>
                
            </ul>

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="http://www.mssm.edu" target="_blank" style = "padding-top: 0px; padding-bottom: 0px; margin-left: 10px;">
                        <img src="images/logo.gif" alt="Mount Sinai" height="40" style = "margin-left: 10px; margin-right: 100px; margin-top:5px; margin-bottom:5px;">
                    </a> 
                </li>
                
            </ul>

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a class="navbar-brand" target="_blank" style="color: white; margin-left: 10px; margin-right: 30px;" href="https://www.immunespace.org/">ImmuneSpace </a>
                </li>
            </ul>
            

        </div>
    </div>

</nav>



<div class="container" id = "mainContainer" style="background: white; padding-bottom: 50px;"> <!-- container start -->

    <div class="row" id="immRegExpRow" style="margin-top: 60px;margin-left:-5px;margin-right:-5px; padding-top:10px;">      
                <p style="font-size:medium; text-align:center;"><strong> immuneRegulation </strong> provides visualization, exploration, and download of large-number of regulatory data through a single, and intuitive interface.
                 </p>      
    </div>

    
    <div class="row" id="geneEntryRow" style="border:1px solid #888888; margin-top: 10px;margin-left:-5px;margin-right:-5px; padding-top:10px;">
        
        <div class="col-md-3" style="font-size:medium;">
            <span id="switchQueryText" style=""><strong> Enter </strong>
                <span> 
                    <div class="btn-group" role="group" aria-label="...">
                          <button id="enterGeneInput" onclick="switchQueryInputGene();" type="button" class="btn btn-default active" style="padding-top:12px; padding-bottom:12px;padding-left:8px;padding-right:8px; font-weight:500; border-radius:3px;">Gene(s)</button>
                          <button id="enterSNPInput" onclick="switchQueryInputSNP();" type="button" class="btn btn-default" style="padding-top:12px; padding-bottom:12px;padding-left:8px;padding-right:8px; font-weight:500; border-radius:3px;">SNP(s)</button>
                    </div> 
                </span>
                <span><strong> : </strong>
            </span>

            <p style = "margin-top:10px;"> <a style = "margin-left: 60px; padding-top:20 px;" href="metadata-all.html"> Query Meta-data </a></p>
        </div>
        
        <div class="col-md-6" style="font-size:medium;">
            <textarea id="geneVal" class="typeahead" placeholder="Enter HUGO Gene symbols" cols="57" rows="4" style="padding-top:0px; margin-bottom:5px; overflow:inherit;overflow-x: hidden;"></textarea>
        </div>
        
        <div class="col-md-3" id="extraGene" style="font-size:16px; padding-top:15px; color:red; visibility:hidden;">
                <p style=""> Please enter a single gene in the gene-box for 'cis-eQTL' queries. </p>               
        </div>
    
    </div>

    
    <div class="row" id= "browseRow" style="padding-bottom: 1px; margin-top: 20px; margin-bottom: 20px; margin-right: -7px; margin-left: -7px; padding-bottom: -10px; background: #ddd; border-style: solid; border-width: 1px; border-radius: 1px; border-color: #888888; height:445px; ">
        
        <div class="row" style="background: white; margin:0px; padding-bottom: 0px; border: 1px solid lightgray;">
            <div class="col-md-3" style = "font-size:large; padding-top:20px;">
                <span style="margin-right:5px;" class="glyphicon glyphicon-list"> </span> <strong> Study List: </strong>
            </div>
            
            <div class="col-md-2" style = "font-size:medium; text-align:center; padding-top:20px;">
                <p id="studyCountText"> 0 studies selected </p> 
            </div>
            
            <div class="col-md-2" id="deselectDiv" style = "font-size:medium; text-align:center; padding-top:20px; visibility:hidden;">
                <a href="#" onclick = "deselectStudies()">  Deselect all </a> 
            </div>
            
            <div class="col-md-3" style = "font-size:medium; text-align:right; padding-top:20px; opacity:0.5;">
                <p> <strong>Study Type: </strong> </p>
                
            </div>
            <div class="col-md-2" style = "font-size:medium; color:#337ab7; text-align:left; padding-top:20px; opacity:0.5;">
                <p id="queryType"> Not Selected  </p>
                
            </div>

            
        </div>

        <div class="row" style="padding:0px; overflow: hidden; height:91%; margin:0px;">
            <div class="col-md-3" style="padding:0px; overflow: scroll; height:inherit; background:white;">
                <ul class="list-group" style="margin-bottom: 0px; font-size:initial;">

                    <li class="list-group-item" id="userListMain" style="color:black; background:#a1cce4;">
                        <span class="badge" id="userListCount">0</span>
                            <a href="#" onClick = filterUser()> &#x21E3; User Studies </a>
                    </li>
                    <li class="list-group-item" id="userListCis" style="color:black; padding-left:35px;background:#a1cce4;">
                        <span class="badge" id="userListCisCount">0</span>
                            <a href="#" onClick = filterCisUser()> &#x21B3; cis-eQTL</a>
                    </li>
                    <li class="list-group-item" id="userListTrans" style="color:black; padding-left:35px;background:#a1cce4;">
                        <span class="badge" id="userListTransCount">0</span>
                            <a href="#" onClick = filterTransUser()> &#x21B3; trans-eQTL</a>
                    </li>
                    <li class="list-group-item" id="listHIPC" style="color:black; background:#a1cce4;">
                        <span class="badge">33</span>
                            <a href="#" onClick = filterHIPC()> 
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span--> &#x21E3; HIPC Studies </a>
                    </li>
                    <li class="list-group-item" id="listCisHIPC" style="color:black; padding-left:35px; background:#a1cce4;">
                        <span class="badge">22</span>
                            <a href="#" onClick = filterCisHIPC()>
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span> 
                                <span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span--> &#x21B3; cis-eQTL</a>
                    </li>
                    <li class="list-group-item" id="listTransHIPC" style="color:black; padding-left:35px; background:#a1cce4;">
                        <span class="badge">10</span>
                            <a href="#" onClick = filterTransHIPC()>
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span>
                                <span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span--> &#x21B3; trans-eQTL</a>
                    </li>
                    <li class="list-group-item" id="listHLAHIPC" style="color:black; padding-left:35px; background:#a1cce4;">
                        <span class="badge">1</span>
                            <a href="#" onClick = filterHLAType()>
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span>
                                <span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span--> &#x21B3; HLA Types</a>
                    </li>
                    <li class="list-group-item" id="listNonHIPC" style="color:black;background:#a1cce4;">
                        <span class="badge">124</span>
                            <a href="#" onClick = filterNonHIPC()>
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span--> &#x21E3; Non-HIPC Studies </a>
                    </li>
                    <li class="list-group-item" id="listCisNonHIPC" style="color:black; padding-left:35px;background:#a1cce4;">
                        <span class="badge">98</span>
                            <a href="#" onClick = filterCisNonHIPC()> 
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-menu-right"> </span>
                                <span style="margin-right:5px;" class="glyphicon glyphicon-option-horizontal"> </span--> &#x21B3; cis-eQTL</a>
                    </li>
                    <li class="list-group-item" id="listTransNonHIPC" style="color:black; padding-left:35px;background:#a1cce4;">
                        <span class="badge">26</span>
                            <a href="#" onClick = filterTransNonHIPC()> 
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-menu-right"> </span>
                                <span style="margin-right:5px;" class="glyphicon glyphicon-option-horizontal"> </span--> &#x21B3; trans-eQTL</a>
                    </li>
                    <li class="list-group-item" id="mainTF" style="color:black;background:#a1cce4;">
                        <span class="badge">1945</span>
                            <a href="#" onClick = filterTF()>
                                <!--span style="margin-right:5px;" class="glyphicon glyphicon-option-vertical"> </span--> &#x21E3; Transcription Factors </a>
                    </li>
                </ul>
            </div>

            <div class="col-md-9" style="padding-left:0px; overflow: scroll; height:inherit; padding-right:0px; background:white;">
                <ul class="list-group" id="allStudies" style="margin-bottom: 0px; font-size:initial;  ">
                    
                    <li class="list-group-item user" id="UserListTitle" style="color:black; padding-top:15px;padding-left:10px;">                        
                        <strong>User Studies</strong>                         
                    </li>

                    <li class="list-group-item user ucis" id="UserListCis0" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study101" value="on">
                        <span value="101" id="span1000">User</span>                          
                    </li>

                    <li class="list-group-item user ucis" id="UserListCis1" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study102" value="on">
                        <span value="102" id="span1001">User</span>                          
                    </li>

                    <li class="list-group-item user ucis" id="UserListCis2" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study103" value="on">
                        <span value="103" id="span1002">User</span>                          
                    </li>

                    <li class="list-group-item user utrans" id="UserListTrans0" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study111" value="on">
                        <span value="111" id="span1100">User</span>                          
                    </li>

                    <li class="list-group-item user utrans" id="UserListTrans1" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study112" value="on">
                        <span value="112" id="span1101">User</span>                          
                    </li>

                    <li class="list-group-item user utrans" id="UserListTrans2" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study113" value="on">
                        <span value="113" id="span1102">User</span>                          
                    </li>

                     <li class="list-group-item user" id="UserListCis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">                 
                    </li>


                    <li class="list-group-item non-hipc" style="color:black; padding-top:15px;padding-bottom:5px;padding-left:10px;">                        
                        <strong>Non-HIPC Studies</strong>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-bottom:0px;">                      
                        <strong>*** COVID-19 ***</strong>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study001" value="on">
                        <span value="001">Lung - Meta-analysis - European (PLoS Genetics, 2012) </span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 1111 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" aria-hidden="true" data-toggle="tooltip" title="Study includes 1,111 patients who passed clinical, genotyping and gene expression quality control assessments according to the three sites of recruitment, Laval University, University of British Columbia, and University of Groningen. The majority of the subjects were smokers or former-smokers. There were some differences in the clinical characteristics across the sites including age, lung function and smoking status. To account for this heterogeneity between sites, authors performed a meta-analysis rather than a pooled analysis for eQTL discovery."></span>   
                        <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003029" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study002" value="on">
                        <span value="002">Lung - Mixed ethnicity - GTEx Consortium v8 (bioRxiv, 2019) </span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 515 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" aria-hidden="true" data-toggle="tooltip" title="The Genotype-Tissue Expression (GTEx) project was established to characterize genetic effects on the transcriptome across human tissues, and to link these regulatory mechanisms to trait and disease associations. v8 data is based on 17,382 RNA-sequencing samples from 54 tissues of 948 post-mortem donors. Authors comprehensively characterize genetic associations for gene expression and splicing in cis and trans, showing that regulatory associations are found for almost all genes, and describe the underlying molecular mechanisms and their contribution to allelic heterogeneity and pleiotropy of complex traits. Leveraging the large diversity of tissues, they provide insights into the tissue-specificity of genetic effects, and show that cell type composition is a key factor in understanding gene regulatory mechanisms in human tissues."></span>   
                        <a href="https://www.biorxiv.org/content/10.1101/787903v1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-bottom:0px;">                      
                        <strong>Whole Blood</strong>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="framingham" value="on">
                        <span value="11">Whole Blood - European American (Framingham Heart Study, NHLBI)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 5257 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" aria-hidden="true" data-toggle="tooltip" title="Framingham Heart Study, under the direction of the National Heart, Lung and Blood Institute (NHLBI), formerly known as the National Heart Institute, has been committed to identifying the common factors or characteristics that contribute to cardiovascular disease (CVD). Study began in 1948 by recruiting an Original Cohort of 5,209 men and women between the ages of 30 and 62 from the town of Framingham, Massachusetts, who had not yet developed overt symptoms of cardiovascular disease or suffered a heart attack or stroke."></span>   
                        <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1142-6" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;border-bottom:0px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="gtex" value="on">
                        <span value="12">Whole Blood - Mixed ethnicity - GTEx Consortium v8 (bioRxiv, 2019)</span>    
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 670 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The Genotype-Tissue Expression (GTEx) project was established to characterize genetic effects on the transcriptome across human tissues, and to link these regulatory mechanisms to trait and disease associations. v8 data is based on 17,382 RNA-sequencing samples from 54 tissues of 948 post-mortem donors. Authors comprehensively characterize genetic associations for gene expression and splicing in cis and trans, showing that regulatory associations are found for almost all genes, and describe the underlying molecular mechanisms and their contribution to allelic heterogeneity and pleiotropy of complex traits. Leveraging the large diversity of tissues, they provide insights into the tissue-specificity of genetic effects, and show that cell type composition is a key factor in understanding gene regulatory mechanisms in human tissues." aria-hidden="true"></span>
                        <a href="https://gtexportal.org/home/" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
                    </li>
                   
                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="westra" value="on">
                        <span value="13">Whole Blood - Meta-analysis - European (Nature Genetics, 2013)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 5311 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors performed expression quantitative trait locus (eQTL) meta-analysis in non-transformed peripheral blood samples from 5,311 individuals with replication in 2,775 individuals. They identified and replicated trans eQTLs for 233 SNPs (reflecting 103 independent loci) that were previously associated with complex traits at genome-wide significance." aria-hidden="true"></span> 
                        <a href="http://genenetwork.nl/bloodeqtlbrowser/" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study14" value="on">
                        <span value="14">Whole Blood - European (Nature Genetics, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 2166 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors generated peripheral blood RNAâ€“seq data from 2,116 unrelated individuals and systematically identified context-dependent eQTLs using a hypothesis-free strategy that does not require previous knowledge of the identity of the modifiers." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/ng.3737" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study15" value="on">
                        <span value="15">Whole Blood - East Asian (PLoS Genetics, 2014)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 1799 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors use genome-wide data on SNPs and array-based expression measures from mononuclear cells obtained from a population-based cohort of 1,799 Bangladeshi individuals to characterize cis- and trans-eQTLs and determine if observed trans-eQTL associations are mediated by expression of transcripts in cis with the SNPs showing trans-association, using Sobel tests of mediation." aria-hidden="true"></span> 
                        <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004818" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study16" value="on">
                        <span value="16">Whole Blood - East Asian (PLoS One, 2014)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 298 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Using genotypes at 1.4 million SNPs and a comprehensive transcriptional profile of 15,454 coding genes and 6,113 lincRNA genes obtained from peripheral blood cells of 298 Japanese individuals, authors mapped expression quantitative trait loci (eQTLs)." aria-hidden="true"></span> 
                        <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0100924" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study17" value="on">
                        <span value="17">Whole Blood - Meta-analysis - Mixed (BioRxiv, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> (*10722 overlaps) 31684 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="eQTLGen Consortium has been set up to identify the downstream consequences of trait-related genetic variants. The consortium incorporates 37 datasets, with a total of 31,684 individuals. (Please note that some of these datasets, with a total of 10722 individuals, overlaps with other Whole Blood datasets listed in here.)" aria-hidden="true"></span> 
                        <a href="https://www.biorxiv.org/content/early/2018/10/19/447367" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study18" value="on">
                        <span value="18">Whole Blood - Mixed ethnicity - CAGE (AJHG, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 2765 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Using data from the Consortium for the Architecture of Gene Expression (CAGE), authors analyzed the mRNA levels for 36,778 transcript expression traits (probes) from 2,765 individuals to comprehensively investigate the genetic architecture and degree of missing heritability for gene expression in peripheral blood." aria-hidden="true"></span> 
                        <a href="https://doi.org/10.1016/j.ajhg.2016.12.008" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study19" value="on">
                        <span value="19">Whole Blood - Mixed ethnicity - LIFE-HEART (HMG, 2015)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 2112 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied the genetics of gene expression in PBMCs of 2112 individuals from the LIFE-Heart Study. They assessed the power of the study in comparison with previously published eQTL studies." aria-hidden="true"></span> 
                        <a href="https://doi.org/10.1093/hmg/ddv194" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study190" value="on">
                        <span value="190">Whole Blood - Mixed (Nature Genetics, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 242 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors describe analyses combining two previously published data sets derived from individuals in the TwinsUK cohort: RNA-seq from four tissues and WGS from the UK10K project." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/ng.3979" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <!--li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study191" value="on">
                        <span value="191">Whole Blood - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li-->

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">       
                        <strong>Monocytes</strong>     
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study21" value="on">
                        <span value="21">Monocytes - African American (Science, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 162 samples </span>
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
     
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study22" value="on">
                        <span value="22">Monocytes - East Asian (Science, 2014)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 155 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
   
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study23" value="on">
                        <span value="23">Monocytes - European American (Science, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 377 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>     
                    </li>
                    
                    <li class="list-group-item non-hipc trans" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study24" value="on">
                        <span value="24">Monocytes - Meta-analysis (Science, 2014)</span>   
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 696 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>     
                    </li>
                    
                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study25" value="on">
                        <span value="25">Monocytes - European (Nature Genetics, 2012)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 288 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Using paired purified primary monocytes and B cells, authors identify new predominantly cell typeâ€“specific cis and trans expression quantitative trait loci (eQTLs), including multi-locus trans associations to LYZ and KLF4 in monocytes and B cells, respectively. " aria-hidden="true"></span>
                        <a href="http://www.nature.com/ng/journal/v44/n5/full/ng.2205.html" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>     
                    </li>
                    
                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study26" value="on">
                        <span value="26">Monocytes - European American (Science, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 432 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="To systematically investigate the impact of immune stimulation upon regulatory variant activity, we exposed primary monocytes from 432 healthy Europeans to interferon-Î³ (IFN-Î³) or differing durations of lipopolysaccharide and mapped expression quantitative trait loci (eQTLs). More than half of cis-eQTLs identified, involving hundreds of genes and associated pathways, are detected specifically in stimulated monocytes." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/343/6175/1246949" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>       
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study27" value="on">
                        <span value="27">Monocytes - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>       
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study28" value="on">
                        <span value="28">Monocytes (non-classical) - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>       
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study29" value="on">
                        <span value="29"> Monocytes - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 194 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study290" value="on">
                        <span value="290"> Monocytes - splicing QTL - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 194 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study291" value="on">
                        <span value="291">Monocytes - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study292" value="on">
                        <span value="292">Monocytes (classical) - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study293" value="on">
                        <span value="293">Monocytes (non-classical) - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study294" value="on">
                        <span value="294">Monocytes - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">   
                        <strong>CD4+ T-cells</strong>     
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study31" value="on">
                        <span value="31">CD4+ T-cells - African American (Science, 2014)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 162 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>      
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study32" value="on">
                        <span value="32">CD4+ T-cells - East Asian (Science, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 155 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>    
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study33" value="on">
                        <span value="33">CD4+ T-cells - European American (Science, 2014)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 377 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>     
                    </li>
                    
                    <li class="list-group-item non-hipc trans" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study34" value="on">
                        <span value="34">CD4+ T-cells - Meta-analysis (Science, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 696 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Context-specific cis- and trans-eQTLs were identified, and cross-population mapping allowed, in some cases, putative functional assignment of candidate causal regulatory variants for disease-associated loci. Authors note an over-representation of T cellâ€“specific eQTLs among susceptibility alleles for autoimmune diseases and of monocyte-specific eQTLs among Alzheimerâ€™s and Parkinsonâ€™s disease variants. This polarization implicates specific immune cell types in these diseases and points to the need to identify the cell-autonomous effects of disease susceptibility variants." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/344/6183/519" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study35" value="on">
                        <span value="35">CD4+ T-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study36" value="on">
                        <span value="36">CD4+ naive Treg-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study37" value="on">
                        <span value="37">CD4+ memory Treg-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study38" value="on">
                        <span value="38">CD4+ Th1-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study39" value="on">
                        <span value="39">CD4+ Th1/17-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study390" value="on">
                        <span value="390">CD4+ Th17-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study391" value="on">
                        <span value="391">CD4+ Th2-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study392" value="on">
                        <span value="392">CD4+ memory Tfh-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study393" value="on">
                        <span value="393"> CD4+ T-Cells - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 171 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study394" value="on">
                        <span value="394"> CD4+ T-Cells - splicing QTL - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 171 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study395" value="on">
                        <span value="395">CD4+ T-Cells - alternative splicing QTL - European (PLOS Genetics, 2015)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 204 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors used the GenCord cohort of 204 newborn Europeansâ€™ lymphoblastoid cell lines, T-cells and fibroblasts derived from umbilical cords. To identify associations between SNPs and alternative splicing (asQTL, alternative splicing Quantitative Trait Locus), they calculated the Spearman Rank Correlation between alternative splicing levels and SNP genotypes within 1 Mb of TSSs." aria-hidden="true"></span> 
                        <a href="https://doi.org/10.1371/journal.pgen.1004461" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study396" value="on">
                        <span value="396">CD4+ T-Cells - European - Estonian Biobank (PLOS Genetics, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 313 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The aims of this study were to perform cis- and trans-eQTL mapping in purified CD4+ and CD8+ T cells and evaluate their pathogenic implications for autoimmune mechanisms. This study participants were healthy donors of the Estonian Genome Center of the University of Tartu. In total, 313 subjects were selected for the study, with median age 54 (standard deviation 17.8), 154 females and 159 males." aria-hidden="true"></span> 
                        <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006643" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study397" value="on">
                        <span value="397">CD4+ T-Cells - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                       
                        <strong>B-cells</strong>                          
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study41" value="on">
                        <span value="41">B-cells - European (Nature Genetics, 2012)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 288 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Using paired purified primary monocytes and B cells, authors identify new predominantly cell typeâ€“specific cis and trans expression quantitative trait loci (eQTLs), including multi-locus trans associations to LYZ and KLF4 in monocytes and B cells, respectively. " aria-hidden="true"></span>
                        <a href="http://www.nature.com/ng/journal/v44/n5/full/ng.2205.html" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study42" value="on">
                        <span value="42">B-cells - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study43" value="on">
                        <span value="43">B-cells - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                        
                        <strong>CD16+ Neutrophils</strong>                           
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study51" value="on">
                        <span value="51">CD16+ Neutrophils - European (Nature Communications, 2015)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 101 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors map expression quantitative trait loci (eQTL) in peripheral blood CD16+ neutrophils from 101 healthy European adults. They identify cis-eQTL for 3281 neutrophil-expressed genes including many implicated in neutrophil function, with 450 of these not previously observed in myeloid or lymphoid cells." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/ncomms8545" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study52" value="on">
                        <span value="52"> CD16+ Neutrophils - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 192 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study53" value="on">
                        <span value="53"> CD16+ Neutrophils - splicing QTL - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 192 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                       
                        <strong>Exposure Studies - Monocytes</strong>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study61" value="on">
                        <span value="61">Lipopolysaccharide (LPS) - 2hours - European (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 432 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="To systematically investigate the impact of immune stimulation upon regulatory variant activity, we exposed primary monocytes from 432 healthy Europeans to interferon-Î³ (IFN-Î³) or differing durations of lipopolysaccharide and mapped expression quantitative trait loci (eQTLs). More than half of cis-eQTLs identified, involving hundreds of genes and associated pathways, are detected specifically in stimulated monocytes." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/343/6175/1246949" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>
                    
                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study62" value="on">
                        <span value="62">Lipopolysaccharide (LPS) - 24hours - European (Science, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 432 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="To systematically investigate the impact of immune stimulation upon regulatory variant activity, we exposed primary monocytes from 432 healthy Europeans to interferon-Î³ (IFN-Î³) or differing durations of lipopolysaccharide and mapped expression quantitative trait loci (eQTLs). More than half of cis-eQTLs identified, involving hundreds of genes and associated pathways, are detected specifically in stimulated monocytes." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/343/6175/1246949" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>
                    
                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study63" value="on">
                        <span value="63">IFN-gamma - European (Science, 2014)</span>    
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 432 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="To systematically investigate the impact of immune stimulation upon regulatory variant activity, we exposed primary monocytes from 432 healthy Europeans to interferon-Î³ (IFN-Î³) or differing durations of lipopolysaccharide and mapped expression quantitative trait loci (eQTLs). More than half of cis-eQTLs identified, involving hundreds of genes and associated pathways, are detected specifically in stimulated monocytes." aria-hidden="true"></span>
                        <a href="http://science.sciencemag.org/content/343/6175/1246949" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                    
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study64" value="on">
                        <span value="64">Lipopolysaccharide (LPS) - 90 minutes - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study65" value="on">
                        <span value="65">Lipopolysaccharide (LPS) - 6 hours - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study66" value="on">
                        <span value="66">Muramyl-dipeptide (MDP) - 90 minutes - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study67" value="on">
                        <span value="67">Muramyl-dipeptide (MDP) - 6 hours - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study68" value="on">
                        <span value="68">5â€²-triphosphate RNA (RNA) - 90 minutes - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study69" value="on">
                        <span value="69">5â€²-triphosphate RNA (RNA) - 6 hours - European (Nature Communications, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 134 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="For the full eQTL cohort, they analyzed primary monocytes isolated from 134 healthy male individuals (185 before quality control), which were either left untreated (baseline) or stimulated with the same three pathogen-derived stimuli, and gene expression was profiled after 90â€‰min and 6â€‰h." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41467-017-00366-1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                       
                        <strong>Exposure Studies - Dendritic Cells </strong>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study131" value="on">
                        <span value="131">Lipopolysaccharide (LPS) - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 356 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study132" value="on">
                        <span value="132">Lipopolysaccharide (LPS) - response QTLs - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 356 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study133" value="on">
                        <span value="133">Influenza infection - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 342 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study134" value="on">
                        <span value="134">Influenza infection - response QTLs - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 342 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study135" value="on">
                        <span value="135">IFN-beta - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 284 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study136" value="on">
                        <span value="136">IFN-beta - response QTLs - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 284 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                       
                        <strong>Lymphoblastoid Cell Lines</strong>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study71" value="on">
                        <span value="71">LCL - European (Nature, 2013)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 373 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors report sequencing and deep analysis of messenger RNA and microRNA from lymphoblastoid cell lines of 462 individuals from the 1000 Genomes Projectâ€”the first uniformly processed high-throughput RNA-sequencing data from multiple human populations with high-quality genome sequences." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature12531" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study72" value="on">
                        <span value="72">LCL - African (Nature, 2013)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 89 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors report sequencing and deep analysis of messenger RNA and microRNA from lymphoblastoid cell lines of 462 individuals from the 1000 Genomes Projectâ€”the first uniformly processed high-throughput RNA-sequencing data from multiple human populations with high-quality genome sequences." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature12531" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study73" value="on">
                        <span value="73">LCL - European - twins study (Nature Genetics, 2012)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 856 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors designed the MuTHER project (Multiple Tissue Human Expression Resource) to develop a major resource of detailed genomic and transcriptome data from three disease-relevant tissues (adipose, lymphoblastoid cell lines (LCLs) and skin) originating from a cohort of 856 deeply phenotyped twins (one-third monozygotic and two-thirds dizygotic from the TwinsUK adult registry)." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/ng.2394" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study74" value="on">
                        <span value="74">LCL - Utah residents with Northern/Western European ancestry (Nature, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 99 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations. At the completion of the project, the genomes of 2,504 individuals from 26 populations were reconstructed using a combination of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature15393" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study75" value="on">
                        <span value="75">LCL - Han Chinese (Nature, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 103 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations. At the completion of the project, the genomes of 2,504 individuals from 26 populations were reconstructed using a combination of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature15393" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study76" value="on">
                        <span value="76">LCL - Gujarati Indians (Nature, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 103 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations. At the completion of the project, the genomes of 2,504 individuals from 26 populations were reconstructed using a combination of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature15393" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study77" value="on">
                        <span value="77">LCL - Japanese (Nature, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 104 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations. At the completion of the project, the genomes of 2,504 individuals from 26 populations were reconstructed using a combination of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature15393" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study78" value="on">
                        <span value="78">LCL - Luhya in Kenya (Nature, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 99 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations. At the completion of the project, the genomes of 2,504 individuals from 26 populations were reconstructed using a combination of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature15393" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study79" value="on">
                        <span value="79">LCL - Yoruba in Nigeria (Nature, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 108 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations. At the completion of the project, the genomes of 2,504 individuals from 26 populations were reconstructed using a combination of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/nature15393" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study790" value="on">
                        <span value="790">LCL - Mixed (Nature Genetics, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 506 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors describe analyses combining two previously published data sets derived from individuals in the TwinsUK cohort: RNA-seq from four tissues and WGS from the UK10K project." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/ng.3979" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study791" value="on">
                        <span value="791">LCL - Mixed - ALSPAC (PLOS Genetics, 2014)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 869 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors performed association analysis of single nucleotide polymorphisms (SNPs) and copy number variants (CNVs) with gene expression measured in 869 lymphoblastoid cell lines of the Avon Longitudinal Study of Parents and Children (ALSPAC) cohort in cis and in trans." aria-hidden="true"></span> 
                        <a href="https://doi.org/10.1371/journal.pgen.1004461" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study792" value="on">
                        <span value="792">LCL - alternative splicing QTL - European (PLOS Genetics, 2015)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 204 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors used the GenCord cohort of 204 newborn Europeansâ€™ lymphoblastoid cell lines, T-cells and fibroblasts derived from umbilical cords. To identify associations between SNPs and alternative splicing (asQTL, alternative splicing Quantitative Trait Locus), they calculated the Spearman Rank Correlation between alternative splicing levels and SNP genotypes within 1 Mb of TSSs." aria-hidden="true"></span> 
                        <a href="https://doi.org/10.1371/journal.pgen.1004461" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>
                    
                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                       
                        <strong>Epigenetics</strong>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study81" value="on">
                        <span value="81">methylation Quantitative Trait Locus (meQTL) - European (Nature Genetics, 2017)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 3841 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors show that disease-associated variants have widespread effects on DNA methylation in trans that likely reflect differential occupancy of trans binding sites by cis-regulated transcription factors." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/ng.3721" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>
                    
                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study82" value="on">
                        <span value="82"> expression Quantitative Trait Methylation (eQTM) - European (Nature Genetics, 2017)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 2101 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors show that disease-associated variants have widespread effects on DNA methylation in trans that likely reflect differential occupancy of trans binding sites by cis-regulated transcription factors." aria-hidden="true"></span>
                        <a href="https://www.nature.com/articles/ng.3721" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study83" value="on">
                        <span value="83"> Monocytes - meQTL - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 196 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study84" value="on">
                        <span value="84"> Monocytes - histone modification QTL (H3K4me1) - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 172 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study85" value="on">
                        <span value="85"> Monocytes - histone modification QTL (H3K27ac) - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 162 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study86" value="on">
                        <span value="86"> Neutrophils - meQTL - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 197 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study87" value="on">
                        <span value="87"> Neutrophils - histone modification QTL (H3K4me1) - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 173 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study88" value="on">
                        <span value="88"> Neutrophils - histone modification QTL (H3K27ac) - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 174 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study89" value="on">
                        <span value="89"> CD4+ T-cells - meQTL - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 133 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study890" value="on">
                        <span value="890"> CD4+ T-cells - histone modification QTL (H3K4me1) - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 104 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study891" value="on">
                        <span value="891"> CD4+ T-cells - histone modification QTL (H3K27ac) - Mixed - BLUEPRINT (Cell, 2016)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 142 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="BLUEPRINT Consortium carried out high-resolution genetic, epigenetic, and transcriptomic profiling in three major human immune cell types (CD14+ monocytes, CD16+ neutrophils, and naive CD4+ T cells) from up to 197 individuals. They assessed, quantitatively, the relative contribution of cis-genetic and epigenetic factors to transcription and evaluated their impact as potential sources of confounding in epigenome-wide association studies. Further, they characterized highly coordinated genetic effects on gene expression, methylation, and histone variation through quantitative trait locus (QTL) mapping and allele-specific (AS) analyses." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2016.10.026" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study892" value="on">
                        <span value="892"> LCL - meQTL - Yoruba (PLOS Genetics, 2014)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 64 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors collected methylation data at âˆ¼300,000 loci in lymphoblastoid cell lines (LCLs) from 64 HapMap Yoruba individuals, and genome-wide bisulfite sequence data in ten of these individuals. They identified (at an FDR of 10%) 13,915 cis methylation QTLs (meQTLs)â€”i.e., CpG sites in which changes in DNA methylation are associated with genetic variation at proximal loci. They found that meQTLs are frequently associated with changes in methylation at multiple CpGs across regions of up to 3 kb." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1371/journal.pgen.1004663" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study893" value="on">
                        <span value="893"> LCL - hmQTL (H3K4me3, H3K4me1, H3K27ac) - Yoruba (Cell, 2015)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 75 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors integrate chromatin profiling for three histone marks in lymphoblastoid cell lines (LCLs) from 75 sequenced individuals with LCL-specific Hi-C and ChIA-PET-based chromatin contact maps to uncover one of the largest collections of local and distal histone quantitative trait loci (hQTLs)." aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2015.07.048" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                       
                        <strong>Other</strong>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study91" value="on">
                        <span value="91">CD56dim CD16+ NK cell - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study92" value="on">
                        <span value="92">CD8+ T-cell - Mixed (Cell, 2018)</span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 91 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="In the first phase of the DICE (Database of Immune Cell EQTLs/Expression/Epigenomics) project, authors present transcriptomic data generated from 13 immune cell types isolated from 106 leukapheresis samples provided by 91 healthy subjects. " aria-hidden="true"></span>
                        <a href="https://doi.org/10.1016/j.cell.2018.10.022" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                        
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study93" value="on">
                        <span value="93">Fibroblast - alternative splicing QTL - European (PLOS Genetics, 2015)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 204 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors used the GenCord cohort of 204 newborn Europeansâ€™ lymphoblastoid cell lines, T-cells and fibroblasts derived from umbilical cords. To identify associations between SNPs and alternative splicing (asQTL, alternative splicing Quantitative Trait Locus), they calculated the Spearman Rank Correlation between alternative splicing levels and SNP genotypes within 1 Mb of TSSs." aria-hidden="true"></span> 
                        <a href="https://doi.org/10.1371/journal.pgen.1004461" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study003" value="on">
                        <span value="003">Prostate - Mixed ethnicity - GTEx Consortium v8 (bioRxiv, 2019) </span>  
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 221 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" aria-hidden="true" data-toggle="tooltip" title="The Genotype-Tissue Expression (GTEx) project was established to characterize genetic effects on the transcriptome across human tissues, and to link these regulatory mechanisms to trait and disease associations. v8 data is based on 17,382 RNA-sequencing samples from 54 tissues of 948 post-mortem donors. Authors comprehensively characterize genetic associations for gene expression and splicing in cis and trans, showing that regulatory associations are found for almost all genes, and describe the underlying molecular mechanisms and their contribution to allelic heterogeneity and pleiotropy of complex traits. Leveraging the large diversity of tissues, they provide insights into the tissue-specificity of genetic effects, and show that cell type composition is a key factor in understanding gene regulatory mechanisms in human tissues."></span>   
                        <a href="https://www.biorxiv.org/content/10.1101/787903v1" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study94" value="on">
                        <span value="94">CD8+ T-Cells - European - Estonian Biobank (PLOS Genetics, 2017)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 313 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="The aims of this study were to perform cis- and trans-eQTL mapping in purified CD4+ and CD8+ T cells and evaluate their pathogenic implications for autoimmune mechanisms. This study participants were healthy donors of the Estonian Genome Center of the University of Tartu. In total, 313 subjects were selected for the study, with median age 54 (standard deviation 17.8), 154 females and 159 males." aria-hidden="true"></span> 
                        <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006643" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study95" value="on">
                        <span value="95">CD8+ T-Cells - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study96" value="on">
                        <span value="96">NK Cells - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study97" value="on">
                        <span value="97">PBMC - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study98" value="on">
                        <span value="98">Dendritic Cells - European - single-cell - Lifelines Deep (Nature Genetics, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 45 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors studied celltype-specific effects of genetic variation on genome-wide gene expression by generating scRNA-seq data of ~25,000 PBMCs from 45 donors of the population-based cohort study Lifelines Deep. After quality control, they first assessed the extent to which previously reported cis-eQTLs from bulk whole blood, using either 94 DeepSAGE samples or 2,116 RNA-seq samples, also show significant effects in the scRNA-seq dataset. For this analysis, they treated the scRNA-seq data as representing bulk PBMCs (by averaging expression levels of all cells per gene per sample, referred to as â€˜bulk-like PBMCsâ€™). They detected 50 and 311 significant cis-eQTLs (gene-level FDR of 0.05) that were previously reported in the DeepSAGE and RNA-seq study, respectively. Although only small proportions (8% and 1%, respectively) of previously reported cis-eQTLs were significant in the scRNA-seq analysis, 96% and 90.4% had identical allelic directions as in the DeepSAGE and RNA-seq studies, respectively, indicating that these cis-eQTLs reflect similar regulatory effects. The few discordant eQTLs may reflect the slightly different sample composition of both datasets (PBMCs versus whole blood) and the relatively few sequence reads targeting the 3â€²-end of genes in the bulk RNA-seq dataset." aria-hidden="true"></span> 
                        <a href="https://www.nature.com/articles/s41588-018-0089-9" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item non-hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt Option" id="study99" value="on">
                        <span value="99">Dendritic Cells - Mixed (Science, 2014)</span>
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 528 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors derived dendritic cells (DCs) from peripheral blood monocytes of healthy individuals (295 Caucasians, 122 African Americans, 117 East Asians) and stimulated them with Escherichia coli lipopolysaccharide (LPS), influenza virus, or the cytokine interferon-Î² (IFN-Î²) to generate 1598 transcriptional profiles. They genotyped each of these individuals at sites of common genetic variation and identified the genetic variants that best explain variation in gene expression and gene induction between individuals. They then tested mechanistic predictions from these associations using synthetic promoter constructs and genome engineering." aria-hidden="true"></span>
                        <a href="https://science.sciencemag.org/content/343/6175/1246980" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                          
                    </li>

                    <li class="list-group-item non-hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px; border-top:0px;border-bottom:0px;">
                        <input type="checkbox" class="studyopt Option" id="study990" value="on">
                        <span value="990">Monocyte-derived macrophages (MDMs) - Mixed - Systemic Sclerosis (Ann Rheum Dis, 2018)</span> 
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 57 samples </span> 
                        <span class="glyphicon glyphicon-info-sign" title="Authors carried out RNA sequencing and genome-wide genotyping in MDMs from 57 patients with SSc and 15 controls. The differential expression and expression quantitative trait locus analysis in SSc was further integrated with epigenetic, expression and eQTL data from skin, monocytes, neutrophils and lymphocytes." aria-hidden="true"></span> 
                        <a href="https://pubmed.ncbi.nlm.nih.gov/29348297/" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span></span> </a>                         
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:10px; border-top:0px;">                       
                        <strong>HIPC Studies</strong>                         
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                        
                        <strong>Influenza Vaccination</strong>                         
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="MayoCheckAll" value="on">
                        <span value="210">Mayo - ImmuneSpace SDY67</span>  
                        <span style = "float:right;"> 
                            <span class="glyphicon glyphicon-info-sign" title="Aim 1: Characterize Immune Profiles Over Time, Aim 2: Correlate Immune Profiles with Vaccine Immunogenicity,Aim 3: Replication of Immune Profiles and Verification of Models" aria-hidden="true"></span>   
                            <a href="https://www.immunespace.org/project/Studies/SDY67/begin.view?" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span>
                            </a> 
                            <a href="./data/hipc/mayo/mayo-all.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                       
                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt MayoOption" id="study211" value="on">
                        <span value="211">Timepoint: Day 0</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 150 samples </span> 
                            <a href="./data/hipc/mayo/mayo-day0.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt MayoOption" id="study212" value="on">
                        <span value="212">Timepoint: Day 3</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 150 samples </span> 
                            <a href="./data/hipc/mayo/mayo-day3.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt MayoOption" id="study213" value="on">
                        <span value="213">Timepoint: Day 28</span>
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 150 samples </span>
                            <a href="./data/hipc/mayo/mayo-day28.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                          
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="NihCheckAll" value="on">
                        <span value="220">NIH - ImmuneSpace SDY80</span>  
                        <span style = "float:right;">  
                            <span class="glyphicon glyphicon-info-sign" title="The Center for Human Immunology, Autoimmunity, and Inflammatory Diseases proposes this protocol designed to obtain blood from healthy adult subjects (NIH employees) prior to vaccination and then at various time points after receiving the FDA-licensed seasonal and H1N1 influenza vaccine. These samples will be used to perform a comprehensive and detailed analysis of the immune system at baseline and in response to vaccination. " aria-hidden="true"></span>   
                            <a href="https://www.immunespace.org/project/Studies/SDY80/begin.view?" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span> 
                            </a> 
                            <a href="./data/hipc/nih/nih-all.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                       
                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt NihOption" id="study221" value="on">
                        <span value="221">Timepoint: Day -7</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 51 samples </span> 
                            <a href="./data/hipc/nih/nih-day-7.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt NihOption" id="study222" value="on">
                        <span value="222">Timepoint: Day 0</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 50 samples </span> 
                            <a href="./data/hipc/nih/nih-day0.cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt NihOption" id="study223" value="on">
                        <span value="223">Timepoint: Day 1</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 53 samples </span> 
                            <a href="./data/hipc/nih/nih-day1.cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt NihOption" id="study224" value="on">
                        <span value="224">Timepoint: Day 7</span>
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 49 samples </span> 
                            <a href="./data/hipc/nih/nih-day7.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                         
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt NihOption" id="study225" value="on">
                        <span value="225">Timepoint: Day 70</span>  
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 51 samples </span>
                            <a href="./data/hipc/nih/nih-day70.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                   
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="YaleCheckAll" value="on">
                        <span value="230">Yale - ImmuneSpace SDY400, SDY404</span>   
                        <span style = "float:right;">  
                            <span class="glyphicon glyphicon-info-sign" title=" Immunologic and genomic signatures of influenza vaccine response" aria-hidden="true"></span>   
                            <a href="https://www.immunespace.org/project/Studies/SDY400/begin.view?" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span>
                            </a> 
                            <a href="./data/hipc/yale/yale-all.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                  
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt YaleOption" id="study231" value="on">
                        <span value="231">Timepoint: Day 0</span>  
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 54 samples </span>  
                            <a href="./data/hipc/yale/yale-day0-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                      
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt YaleOption" id="study232" value="on">
                        <span value="232">Timepoint: Day 2</span>
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 55 samples </span>
                            <a href="./data/hipc/yale/yale-day2-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                          
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt YaleOption" id="study233" value="on">
                        <span value="233">Timepoint: Day 7</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 55 samples </span>
                            <a href="./data/hipc/yale/yale-day7-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                          
                    </li> 

                     <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt YaleOption" id="study234" value="on">
                        <span value="234">Timepoint: Day 28</span>
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 55 samples </span> 
                            <a href="./data/hipc/yale/yale-day28-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                          
                    </li> 


                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" id="EmoryCheckAll" value="on">
                        <span value="240">Emory - ImmuneSpace SDY56 </span>  
                        <span style = "float:right;">  
                       
                            <span class="glyphicon glyphicon-info-sign" title="Single center, open label study in which adult healthy volunteers with no contraindications to immunization will be vaccinated with TIV. Blood samples will be collected on Days D0 (at enrollment) and D1, D3, D7, D14, D30, D180 post vaccination to study innate and/or adaptive immunity markers. (Note: FDR cutoff value = 0.5 used for the calculation of cis-eQTL results from all Emory datasets.)" aria-hidden="true"></span>   
                            
                            <a href="https://www.immunespace.org/project/Studies/SDY56/begin.view?" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span>    
                            </a> 
                            <a href="./data/hipc/emory/emory-all.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                     
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt EmoryOption" id="study241" value="on">
                        <span value="241">Timepoint: Day 0</span>
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 55 samples </span>
                            <a href="./data/hipc/emory/emory-day0-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                          
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt EmoryOption" id="study242" value="on">
                        <span value="242">Timepoint: Day 1</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 53 samples </span>
                            <a href="./data/hipc/emory/emory-day1-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                         
                    </li>
                   
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt EmoryOption" id="study243" value="on">
                        <span value="243">Timepoint: Day 3</span>
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 53 samples </span> 
                            <a href="./data/hipc/emory/emory-day3-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                         
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt EmoryOption" id="study244" value="on">
                        <span value="244">Timepoint: Day 7</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 51 samples </span> 
                            <a href="./data/hipc/emory/emory-day7-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                       
                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt EmoryOption" id="study245" value="on">
                        <span value="245">Timepoint: Day 14</span>   
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 53 samples </span> 
                            <a href="./data/hipc/emory/emory-day14-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                       
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                        
                        <strong>Chikungunya Infection</strong>                         
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="ChikvCheckAll" value="on">
                        <span value="300">DHIPC - SDY1288</span>  
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 43 samples </span>
                            <span class="glyphicon glyphicon-info-sign" title="Study Title: Chikungunya virus infection immunoprofiling in pediatric cohort; Objectives: Identify biomarkers of acute CHIKV infection and correlate these with severity of disease." aria-hidden="true"></span>   
                            <a href="http://www.immport.org/immport-open/public/home/studySearch" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span>   
                            </a> 
                            <a href="./data/hipc/chikv/chik-all.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span> 

                    </li>
                    
                    <li class="list-group-item hipc cis" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt ChikvOption" id="study301" value="on">
                        <span value="301">Timepoint: Acute (Day 2-4)</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 41 samples </span> 
                            <a href="./data/hipc/chikv/chik-day1-cis.txt"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt ChikvOption" id="study302" value="on">
                        <span value="302">Timepoint: Convalescent (Day 14-17)</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 43 samples </span> 
                            <a href="./data/hipc/chikv/cihk-day9.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>

                    <li class="list-group-item hipc hla" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                        
                        <strong>Zika Infection</strong>                         
                    </li>

                    <li class="list-group-item hipc hla" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="ZikvCheckAll" value="on">
                        <span value="400">DHIPC : UC Berkeley &nbsp;&nbsp; - &nbsp;&nbsp;&nbsp;</span> 

                        <span style = "float:medium;"> (HLA Genotyping 
                            <a href="./data/hipc/zikv/Zika-HLA.xlsx"> 
                                <span class="glyphicon glyphicon-file" title="Download results as an Excel file" aria-hidden="true"></span>   
                            </a> 
                            <a href="./zika-hla.html" target="_blank"> 
                                <span class="glyphicon glyphicon-stats" title="Visualize results using histograms" aria-hidden="true"></span> 
                            </a>
                            )
                        </span>  
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 89 samples </span>
                            <span class="glyphicon glyphicon-info-sign" title="Zika virus infection immunoprofiling." aria-hidden="true"></span>   
                            <a href="http://www.immport.org/immport-open/public/home/studySearch" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span>   
                            </a> 
                            <a href="./data/hipc/zikv/zika-all.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span> 

                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt ZikvOption" id="study401" value="on">
                        <span value="401">Timepoint: Acute </span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 84 samples </span> 
                            <a href="./data/hipc/zikv/zika-acute.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>
                    
                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt ZikvOption" id="study402" value="on">
                        <span value="402">Timepoint: Late Acute </span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 88 samples </span> 
                            <a href="./data/hipc/zikv/zika-lateacute.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>

                    <li class="list-group-item hipc" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:50px;border-bottom:0px;border-top:0px;">
                        <input type="checkbox" class="studyopt ZikvOption" id="study403" value="on">
                        <span value="403">Timepoint: Convalescent</span> 
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 87 samples </span> 
                            <a href="./data/hipc/zikv/zika-convalescent.zip"> 
                                <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> 
                            </a>
                        </span>                        
                    </li>


                    <!--li class="list-group-item hipc hla noqtl" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                        
                        <strong>Kidney Transplant</strong>                         
                    </li>

                    <li class="list-group-item hipc hla noqtl" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="KidneyCheckAll" value="on">
                        <span value="410">HIPC : UCLA &nbsp;&nbsp; - &nbsp;&nbsp;&nbsp;</span> 

                        <span style = "float:medium;"> (HLA Genotyping 
                            <a href="./data/hipc/zikv/Kidney-HLA.xlsx"> 
                                <span class="glyphicon glyphicon-file" title="Download results as an Excel file" aria-hidden="true"></span>   
                            </a> 
                            <a href="./kidney-hla.html" target="_blank"> 
                                <span class="glyphicon glyphicon-stats" title="Visualize results using histograms" aria-hidden="true"></span> 
                            </a>
                            )
                        </span>  
                        <span style = "float:right;"> 
                            <span style = "color:gray; font-size:smaller;"> 54 samples </span>
                            <span class="glyphicon glyphicon-info-sign" title="Kidney transplantation immunoprofiling." aria-hidden="true"></span>   
                            <a href="http://www.immport.org/immport-open/public/home/studySearch" target="_blank"> 
                                <span class="glyphicon glyphicon-book" aria-hidden="true"></span>   
                            </a> 

                        </span> 

                    </li-->
                    
                    

                    <li class="list-group-item tr-fact" style="color:black;padding-top:5px; padding-bottom:5px; padding-left:10px; border-top:0px;">                       
                        <strong>Transcription Factors</strong>                         
                    </li>

                    <li class="list-group-item tr-fact" style="color:black;padding-top:15px; padding-bottom:5px; padding-left:30px; border-bottom:0px;">                        
                        <strong>ENCODE ChIP-Seq Experiments </strong>                         
                    </li> 

                    <li class="list-group-item tr-fact" style="color:black;padding-top:5px;padding-bottom:5px;padding-left:30px;border:0px;">
                        <input type="checkbox" id="TfAll" class="studyopt Option" value="on">
                        <span value="001">Transcription Factor Binding Sites </span>   
                        <span style = "float:right;"> <span style = "color:gray; font-size:smaller;"> 144 cell types, 736 targets </span>
                        <span class="glyphicon glyphicon-info-sign" title="ChIP-Seq Transcription Factor Binding Site experiments from ENCODE" aria-hidden="true"></span>   
                        <a href="https://www.encodeproject.org/matrix/?type=Experiment&replicates.library.biosample.donor.organism.scientific_name=Homo+sapiens&assay_title=ChIP-seq&target.investigated_as=transcription+factor&assembly=hg19&award.project=ENCODE&files.file_type=bed+narrowPeak" target="_blank"> <span class="glyphicon glyphicon-book" aria-hidden="true"></span>    </span>   </a>                   
                    </li>

                </ul>     
            </div>       
        </div>
    </div>

    
    <div class="row" id="studyEntryRow" style="border:1px solid #888888; margin-left:-5px;margin-right:-5px; margin-bottom: 20px;padding-top:10px;">
        <div class="col-md-3" style="font-size:medium;">
                <p style=""><strong> Please Select Study Type: </strong></p>               
        </div>
        
        <div class="col-md-4" style="font-size:medium;">                 
                        <label class="radio-inline"><input type="radio" onclick="selectCis()" name="studyType" id="item1" value="cis" checked>
                        cis-eQTL </label>    
                          
                        <label class="radio-inline"><input type="radio" onclick="selectTrans()" name="studyType" id="item2" value="trans">
                        trans-eQTL </label>             
        </div>
    </div>


    <div class="row" id="operationEntryRow" style="border:1px solid #888888; margin-left:-5px;margin-right:-5px; margin-bottom: 20px;padding-top:10px;">
        <div class="col-md-3" style="font-size:medium;">
                <p style=""><strong> Operation to use on Gene List: </strong></p>               
        </div>
        
        <div class="col-md-6" style="font-size:medium;">                 
                        <label class="radio-inline"><input type="radio" onclick="selectInt()" name="operationType" id="opitem1" value="intersect" >
                        Intersection </label>    
                          
                        <label class="radio-inline"><input type="radio" onclick="selectUnion()" name="operationType" id="opitem2" value="union" checked>
                        Union </label>             
        </div>
    </div>

    
    
    <div class="row" id="queryRow" style="margin-left:-5px;margin-right:-5px; padding-top:0px;">
        <div class="col-md-4" id="privDiv" style="padding-left:0px; text-align:left;">
            <label style="font-style:italic;font-size:medium;font-weight:500;"> Have private studies: </label>
        <a href="#" id="uploadStudy" class="btn btn-primary active" data-toggle="modal" data-target="#largeModal" style="width:140px; height:40px; padding:5px; margin:5px; margin-left: 0px; border: black; border-radius: 5px; color: white; font-size:medium; font-weight:400;"> Upload Study </a>
        </div>


        <div class="col-md-5" id="missingDataset" style="font-size:medium; padding-top:15px; color:red; visibility:hidden;">
                <p id="missingDataText" style=""> Please select 1 or more datasets. </p>               
        </div>

        <div class="col-md-3" id="submitButton" style="padding-left:0px; padding-right:0px; text-align:right;">
            <button onclick="submitQuery()" id="geneSearch" class="btn btn-primary active" style="width:200px; height:50px; padding:5px; margin:5px; margin-left: 0px; border: black; border-radius: 5px; color: white; font-size:large; font-weight:500;"> Submit Query </button>
        </div>
    </div>

    <div class="row" id="TFmultipleGeneRow" style="visibility:hidden; padding:5px; padding-left: 10px; margin-top: 10px; margin-left: 10px; margin-right: 10px; font-size:0.9vw; margin-bottom:15px; overflow-x: auto; border-color:rgb(238,238,238); border-width:1.8px;">

    </div>
    
    <div class="row" id="filterRow" style="margin-top: 10px; margin-left: -10px; margin-right: -10px; visibility:hidden;">

        <div class="col-md-6 filter" id="filterCol" style=" margin-left: 5px; border-style: double; border-width: 1px; border-color: #ccc; background: whitesmoke;">
           
            <label for="filterChr">SNP Chr:</label>

            <select name="sometext" id="filterChr" style="margin-top: 10px; margin-left: 10px; margin-right: 10px; font-size: 16px;width: 40px !important; min-width: 40px; max-width: 40px; size=1;">
        
                <option value="c1">1</option>
                <option value="c2">2</option>
                <option value="c3">3</option>
                <option value="c4">4</option>
                <option value="c5">5</option>
                <option value="c6">6</option>
                <option value="c7">7</option>
                <option value="c8">8</option>
                <option value="c9">9</option>
                <option value="c10">10</option>
                <option value="c11">11</option>
                <option value="c12">12</option>
                <option value="c13">13</option>
                <option value="c14">14</option>
                <option value="c15">15</option>
                <option value="c16">16</option>
                <option value="c17">17</option>
                <option value="c18">18</option>
                <option value="c19">19</option>
                <option value="c20">20</option>
                <option value="c21">21</option>
                <option value="c22">22</option>
                
            </select>

            <label for="filterChr">Position:</label>
            <input type="text" id="filterStart" name="FirstName" value="0" size="9" style="margin-left: 10px;">
            <text style="font-size: 12px; margin-top: 12px;"> - </text>
            <input type="text" id="filterEnd" name="FirstName" value="End" size="9">
        
            <button onclick="filter()" id="filter" class="btn btn-primary active" style="width:53px; height:30px; padding:5px; margin:5px; margin-left:10px;"> Filter </button> 
            
            <button onclick="unfilter()" id="unfilter" class="btn btn-primary disabled" style="width:53px; height:30px; padding:5px; margin:5px; ">  Undo </button>
        </div>

    </div> <!-- FilterRow finish -->


    <div class="row" id="plotRow" style="padding-top:0px;visibility:hidden;">
         
        <div class="col-md-6" id="plotCol" style="margin-left:90px; padding-top:10px; padding-right:0px;">
                <label style="">  Change Plot Type: </label>               
        

            <button onclick="manhattanPlot()" id="manPlot" class="btn btn-primary active" style="width:175px; height:40px; padding:5px; margin:5px; margin-left: 10px; font-size:medium; border-radius: 5px; font-weight:400;"> Manhattan Plot </button>
        </div>

    </div>

    
    

    <div class="row" id="transRow" style="visibility:hidden;">
          

    </div>


    <div class="row" id="multipleGeneRow" style="visibility:hidden; text-align:center; padding:5px; padding-left: 10px; margin-left: 10px; margin-right: 10px; font-size:0.9vw; margin-bottom:15px; overflow-x: auto; border-color:rgb(238,238,238); border-width:1.8px;">
          <!--textarea id="multipleGeneText" cols="100" rows="2" style="padding:5px; padding-left: 10px; margin-left: 10px; font-size:0.9vw; margin-bottom:15px;  overflow:inherit;overflow-x: hidden; border-color:rgb(238,238,238); border-width:1.8px;"></textarea-->

    </div>


    <div class="row" id="igvDiv" style="margin-top: -10px; margin-left: -5px; margin-right: -5px; padding:5px; visibility: hidden; border:1px solid lightgray">
    </div>


    <div class="row" id= "cisLegendRow" style = "margin-top:15px; margin-left:-5px; margin-right:-5px; border-style: solid; border-width: 1px; border-color: lightgray; visibility: hidden;">
        
        <div class="col-md-1" style="margin-top: 5px;">
            <strong> Legend: </strong>
        </div>
        
        <div class="col-md-5">
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-3" cx="10" cy="15" r="5"> </circle>
                </svg>
                <span> cis-eqtl for the queried gene common in multiple tracks </span>
            </div>
            
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-2" cx="10" cy="15" r="5"> </circle>
                </svg>
                 cis-eqtl for the queried gene in a single track
            </div>
            
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-1" cx="10" cy="15" r="5"> </circle>
                </svg>
                 cis-eqtl for a different gene in the genomic region 
            </div>
        </div>
        
        <div class="col-md-1">
        </div>
    
        <div class="col-md-5" style = "padding-left: 0px;text-align:right; margin-top:5px;">
                       
            <button onclick="saveSVG()" id="sSVG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as SVG" style="width:85px; height:30px; padding:5px; margin:5px; margin-left:10px;"> SVG  <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span></button> 
            
            <button onclick="saveCisPDF()" id="cisPDF" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PDF" style="width:85px; height:30px; padding:5px; margin:5px;"> PDF <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button> 

            <button onclick="saveCisPNG()" id="cisPNG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PNG" style="width:85px; height:30px; padding:5px; margin:5px;"> PNG <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>

            <button id="cisTXT" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title=" To save cis-eQTL results as text file, click on the gear icon on the right-side of track, and click &quot;Save Results&quot; menu item." style="width:85px; height:30px; padding:5px; margin:5px;"> TXT <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>
        </div>
    
    </div>

    <div class="row" id= "cisLegendRowSnp" style = "margin-top:15px; margin-left:-5px; margin-right:-5px; border-style: solid; border-width: 1px; border-color: lightgray; visibility: hidden;">
        
        <div class="col-md-1" style="margin-top: 5px;">
            <strong> Legend: </strong>
        </div>
        
        <div class="col-md-5">
            
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-2" cx="10" cy="15" r="5"> </circle>
                </svg>
                 cis-eqtl for the queried SNP
            </div>
            
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-1" cx="10" cy="15" r="5"> </circle>
                </svg>
                 cis-eqtl for a different SNP in the genomic region 
            </div>
        </div>
        
        <div class="col-md-1">
        </div>
    
        <div class="col-md-5" style = "padding-left: 0px;text-align:right; margin-top:5px;">
                       
            <button onclick="saveSVG()" id="sSVG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as SVG" style="width:85px; height:30px; padding:5px; margin:5px; margin-left:10px;"> SVG  <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span></button> 
            
            <button onclick="saveCisPDF()" id="cisPDF" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PDF" style="width:85px; height:30px; padding:5px; margin:5px;"> PDF <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button> 

            <button onclick="saveCisPNG()" id="cisPNG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PNG" style="width:85px; height:30px; padding:5px; margin:5px;"> PNG <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>

            <button id="cisTXT" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title=" To save cis-eQTL results as text file, click on the gear icon on the right-side of track, and click &quot;Save Results&quot; menu item." style="width:85px; height:30px; padding:5px; margin:5px;"> TXT <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>
        </div>
    
    </div>

    <div class="row" id= "transLegendRow" style = "margin-top:15px; margin-left:-5px; margin-right:-5px; padding-bottom: 10px; border-style: solid; border-width: 1px; border-color: lightgray; visibility: hidden;">
        
        <div class="col-md-4" style="margin-left: 15px;">
            
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-2" cx="10" cy="15" r="5"> </circle>
                </svg>
                <span style="font-weight:normal; font-size:14px;"> eQTL with NO GWAS hits </span>
            </div>
            
            <div class="row">
                <svg height="21" width="20">
                    <circle id="circle-leg-4" cx="10" cy="15" r="5"> </circle>
                </svg>
                 <span style="font-weight:normal; font-size:14px;"> eQTL with GWAS hit(s) </span>
            </div>
            
        </div>
        
    
        <div class="col-md-7" style = "padding-left: 0px; text-align:right; margin-top:5px; margin-left:20px; margin-right:0px;">
                       
            <button onclick="saveSVG()" id="sSVG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as SVG" style="width:65px; height:30px; padding:5px; margin:5px; margin-left:10px;"> SVG  <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span></button> 
            
            <button onclick="savePDF()" id="sPDF" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PDF" style="width:65px; height:30px; padding:5px; margin:5px;"> PDF <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button> 

            <button onclick="savePNG()" id="sPNG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PNG" style="width:65px; height:30px; padding:5px; margin:5px;"> PNG <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>

            <button onclick="saveText()" id="sTXT" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Results as Text File" style="width:65px; height:30px; padding:5px; margin:5px;"> TXT <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>
        </div>
    
    </div>


    <div class="row" id= "controlsRow" style = "margin-top:5px; visibility:hidden;">

        <div class="col-md-3">
        </div>

        <div class="col-md-2" style = "padding-left: 0px; padding-right: 0px;">
            <button onclick="first()" id="firstB" class="btn btn-primary active" style="width:35px; height:30px; padding:5px; margin:5px; margin-left:10px;"> << </button> 
            
            <button onclick="prev()" id="prevB" class="btn btn-primary active" style="width:35px; height:30px; padding:5px; margin:5px;"> < </button> 
        </div>
        
        <div class="col-md-2">
            <p id="displayRange" style="text-align: center; font-size: 14px; margin-top:10px; font-weight:500;"> 1 - 50 </p> 
        </div>

        <div class="col-md-2" style = "padding-left: 0px; padding-right: 0px;"> 
            <button onclick="next()" id="nextB" class="btn btn-primary active" style="width:35px; height:30px; padding:5px; margin:5px;"> > </button> 
            <button onclick="last()" id="lastB" class="btn btn-primary active" style="width:35px; height:30px; padding:5px; margin:5px;"> >> </button>
        </div>
    </div>

    <div class="row" id= "saveButtonsRow" style = "margin-top:5px; visibility:hidden;">
        <div class="col-md-3" id= "saveButtonsGap">
        </div>
        
        <div class="col-md-9" style = "padding-left: 0px;">
                       
            <button onclick="saveSVG()" id="sSVG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as SVG" style="width:85px; height:30px; padding:5px; margin:5px; margin-left:10px;"> SVG  <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span></button> 
            
            <button onclick="savePDF()" id="sPDF" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PDF" style="width:85px; height:30px; padding:5px; margin:5px;"> PDF <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button> 

            <button onclick="savePNG()" id="sPNG" class="btn btn-primary active" data-toggle="tooltip" data-placement="bottom" title="Save Graphics as PNG" style="width:85px; height:30px; padding:5px; margin:5px;"> PNG <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> </button>
        
        </div>
    
    </div>


    <select name="sometext" multiple="multiple" id="geneDataSelect" style="visibility:hidden; padding-bottom:10px; padding-top:10px; width:300px;">
        
    </select>

    <iframe id="gwasFile" src="./data/gwas.txt" style="display: none;"></iframe>


</div>  <!-- container finish-->



<div class="modal fade" id="largeModal" tabindex="-1" role="dialog" aria-labelledby="largeModal" aria-hidden="true">
  <div class="modal-dialog modal-lg">
    <div class="modal-content">
      
      <div class="modal-header">
        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
        <h3 class="modal-title" id="myModalLabel">Upload Private Study</h3>
      </div>
      
      <div class="modal-body">
        <h4 style="text-align:center;">Upload your study and visualize results within immuneRegulation interface</h4>
      </div>

        <div class="bs-example">
            <div class="panel-group" id="accordion">
                <div class="panel panel-default">
                    <div class="panel-heading">
                        <h4 class="panel-title">
                            <a data-toggle="collapse" data-parent="#accordion" href="#collapseOne">Upload cis-eQTL Data</a>
                        </h4>
                    </div>
        
                    <div id="collapseOne" class="panel-collapse collapse in">
                        <div class="panel-body">
                            <p>For performance reasons, ImmuneRegulation makes use of cis-eQTL data files that are compressed and indexed using tabix. </p> 
                            
                            <p> <strong> 1. </strong> Format your cis-eQTL results data accordingly with the following tab-separated columns (see example input):    </p>
                            
                            <p style="font-size:small;font-weight:bold; margin-bottom:0px; margin-left:20px;"> Chromosome <span style="display: inline-block; width: 3ch;">&#9;</span> Gene <span style="display: inline-block; width: 3ch;">&#9;</span> SNP-ID/Name <span style="display: inline-block; width: 3ch;">&#9;</span> SNP-Position <span style="display: inline-block; width: 3ch;">&#9;</span> SNP-Position <span style="display: inline-block; width: 3ch;">&#9;</span> P-value <span style="display: inline-block; width: 3ch;">&#9;</span> . <span style="display: inline-block; width: 3ch;">&#9;</span> . </p>
                            
                            <p style="font-size:small; margin-bottom:0px; margin-left:20px;">3 <span style="display: inline-block; width: 3ch;">&#9;</span>  CCDC80 <span style="display: inline-block; width: 3ch;">&#9;</span> rs2638022 <span style="display: inline-block; width: 3ch;">&#9;</span>  112265776 <span style="display: inline-block; width: 3ch;">&#9;</span>  112265776 <span style="display: inline-block; width: 3ch;">&#9;</span>  4.93996E-10 <span style="display: inline-block; width: 3ch;">&#9;</span> . <span style="display: inline-block; width: 3ch;">&#9;</span>  . </p>

                            <p style="font-size:small; margin-left:20px; margin-bottom: 20px;">5 <span style="display: inline-block; width: 3ch;">&#9;</span>  ERAP2 <span style="display: inline-block; width: 3ch;">&#9;</span> rs2287988 <span style="display: inline-block; width: 3ch;">&#9;</span>  96237326 <span style="display: inline-block; width: 3ch;">&#9;</span>  96237326 <span style="display: inline-block; width: 3ch;">&#9;</span>  4.5617E-8 <span style="display: inline-block; width: 3ch;">&#9;</span> . <span style="display: inline-block; width: 3ch;">&#9;</span>  . </p>
                            
                            <p style="font-size:small; margin-bottom:0px;"> <strong> <em> Column Definitions: </strong> </em> </p>
                            <p style="font-size:small; margin-bottom:20px;"> <strong> <em> Chromosome: </strong> </em> chromosome number for the SNP, <strong> <em> Gene: </strong> </em> HUGO gene symbol for the gene associated with cis-eQTL, <strong> <em> SNP-ID/Name: </strong> </em> rsid for the SNP, <strong> <em> SNP-Position: </strong> </em> SNP Position in 'Chromosome', <strong> <em> P-value: </strong> </em> p-value associated with the cis-eQTL </p>

                            <p> <strong> 2. </strong> Sort the above data based on genomic location, and then generate your compressed file and index file using tabix (refer to <a href= "http://www.htslib.org/doc/tabix.html" target="_blank"> Tabix manual </a> for details.) </p>
                            
                            <p> <strong> 3. </strong> Place your data files (i.e. compressed file and index file) in a web-accessible server that supports Cross-Origin Resource Sharing (CORS), and copy the link to your compressed file below:</p>
                            
                            <div class = "row">
                                <div class="col-md-7">
                                    <textarea id="cisDataVal" placeholder="Enter Link" cols="66" rows="2" style="padding-top:0px; margin-bottom:5px; overflow:inherit;overflow-x: hidden;"></textarea> 
                                </div>
                                <div class="col-md-2">
                                    <button onclick="submitCisData()" id="cisUpload" class="btn btn-primary active" style="width:130px; height:40px; padding:5px;  margin-left: 10px; border: black; border-radius: 5px; color: white; font-size:medium; font-weight:400;"> Upload </button>
                                </div>
                                <div class="col-md-2">
                                    <button onclick="sampleCisData()" id="cisSampleUpload" class="btn btn-primary active" style="width:130px; height:40px; padding:5px;  margin-left: 10px; border: black; border-radius: 5px; color: white; font-size:medium; font-weight:400;"> Load Sample </button>
                                </div>
                            </div>

                            <div id="cisfileList">
                                <p></p>
                            </div>
          
                        </div>
                    </div>
                </div>
      

                <div class="panel panel-default">
                    <div class="panel-heading">
                        <h4 class="panel-title">
                            <a data-toggle="collapse" data-parent="#accordion" href="#collapseTwo">Upload trans-eQTL Data</a>
                        </h4>
                    </div>
        
                    <div id="collapseTwo" class="panel-collapse collapse">
                        <div class="panel-body">
                            <p> <strong> 1. </strong> Please make sure that the data file you provide is a .tsv (tab-sepaeated text) file with the following columns (make sure you use the same column headers listed below; and see example input): </p>
                            
                            <p style="font-size:small;font-weight:bold; margin-bottom:0px; margin-left:20px;"> PValue <span style="display: inline-block; width: 5ch;">&#9;</span> SNPName <span style="display: inline-block; width: 5ch;">&#9;</span> SNPChr <span style="display: inline-block; width: 5ch;">&#9;</span> SNPChrPos <span style="display: inline-block; width: 5ch;">&#9;</span> Gene <span style="display: inline-block; width: 5ch;">&#9;</span> ProbeChr <span style="display: inline-block; width: 5ch;">&#9;</span> ProbeCenterChrPos </p>

                            <p style="font-size:small; margin-bottom:0px; margin-left:20px;">5.16389E-39 <span style="display: inline-block; width: 3ch;">&#9;</span>  rs2249563 <span style="display: inline-block; width: 3ch;">&#9;</span> 4 <span style="display: inline-block; width: 3ch;">&#9;</span>  39455378 <span style="display: inline-block; width: 3ch;">&#9;</span>  RPL9 <span style="display: inline-block; width: 3ch;">&#9;</span>  23 <span style="display: inline-block; width: 3ch;">&#9;</span> 23925996 </p>
                                             
                            <p style="font-size:small; margin-bottom:20px; margin-left:20px;">8.39797E-34 <span style="display: inline-block; width: 3ch;">&#9;</span>  rs4103004 <span style="display: inline-block; width: 3ch;">&#9;</span> 3 <span style="display: inline-block; width: 3ch;">&#9;</span>  20042834 <span style="display: inline-block; width: 3ch;">&#9;</span>  EFHB <span style="display: inline-block; width: 3ch;">&#9;</span>  3 <span style="display: inline-block; width: 3ch;">&#9;</span> 18996472 </p>

                            <p style="font-size:small; margin-bottom:0px;"> <strong> <em> Column Definitions: </strong> </em> </p>
                            <p style="font-size:small; margin-bottom:20px;"> <strong> <em> PValue: </strong> </em> p-value associated with the trans-eQTL, <strong> <em> SNPName: </strong> </em> rsid for the SNP, <strong> <em> SNPChr: </strong> </em> chromosome number for the SNP, <strong> <em> SNPChrPos: </strong> </em> SNP Position in SNPChr, <strong> <em> Gene: </strong> </em> HUGO gene symbol for the gene associated with trans-eQTL, <strong> <em> ProbeChr: </strong> </em> chromosome number for the probe in gene expression, <strong> <em> ProbeCenterChrPos: </strong> </em> center position -within ProbeChr- of the probe in gene expression </p>

                            <p> <strong> 2. </strong> Place your data file in a web-accessible server that supports Cross-Origin Resource Sharing (CORS), and copy the link to your file below:</p>
                            
                            <div class = "row">
                                <div class="col-md-7">
                                    <textarea id="transDataVal" placeholder="Enter Link" cols="66" rows="2" style="padding-top:0px; margin-bottom:5px; overflow:inherit;overflow-x: hidden;"></textarea> 
                                </div>
                                <div class="col-md-2">
                                    <button onclick="submitTransData()" id="transUpload" class="btn btn-primary active" style="width:130px; height:40px; padding:5px;  margin-left: 10px; border: black; border-radius: 5px; color: white; font-size:medium; font-weight:400;"> Upload </button>
                                </div>
                                <div class="col-md-2">
                                    <button onclick="sampleTransData()" id="transSampleUpload" class="btn btn-primary active" style="width:130px; height:40px; padding:5px;  margin-left: 10px; border: black; border-radius: 5px; color: white; font-size:medium; font-weight:400;"> Load Sample </button>
                                </div>
                            </div>

                            <div id="transfileList">
                                <p></p>
                            </div>

                        </div>
                    </div>
                </div>
      
      
            </div>

        <div class="modal-footer">
            <button type="button" class="btn btn-default" data-dismiss="modal" style="font-size:large;">Close</button>
            <button type="button" onclick="uploadStudy()" id="saveModal" class="btn btn-primary" style="font-size:large;">Save changes</button>
        </div>
    
        </div>
    </div>
</div>
</div>


<nav class="navbar navbar-inverse navbar-fixed-bottom" style= "margin:5px; border: 1 px solid; border-color: gray; height: 60px;">

    <nav class="nav-upper" style = "background: #3f97ca; height: 60px; padding-top:10px;">
        <div class="container-fluid" style = "text-align:center; color:white; font-size: normal;">
            <ul class="nav navbar-upper">
                <li> ImmuneRegulation <span> | </span> Version 1.0.0 </li>
                <li> <span><a href = "http://research.mssm.edu/gumuslab/" style = "color:white; text-decoration:underline;"> GÃ¼mÃ¼ÅŸ Lab </a></span> 
                    <span> | </span>Icahn School of Medicine at Mount Sinai </li>
            </ul>
        </div>
    </nav>

</nav>


<div class="container-fluid">

    <footer>
        <p></p>
    </footer>

</div>



<script type="text/javascript">

    var original, totalTrans, currenti = 0, currentj = 49;
    var geneData = [];
    var transData = [];
    var cisData = [];
    var SNP_interest;
    var SNP_hover;
    var svg;
    var wdth, hght = 410;
    var margin = {top: 30, right: 20, bottom: 30, left: 50};

    var highlightC = 0;
    var filterToggle = 0;

    var study = 0;
    var operation = 0;
    var datasetvalues = [];

    var gwasData = [];
    var gwasRead = 0;


    var plot = 1; //pvalue-order... 2: manhattan-plot 
    
    var reader = new FileReader();
    var userCisDataId = 0;
    var userTransDataId = 0;
    var userCisArr = [];
    var userTransArr = [];
    var userCisFilename = [];
    var userCisFileLink = [];
    var userTransFilename = [];
    var userTransFileLink = [];

    var userCisData = [];
    var userTransData = [];

    var viz = d3.select('#transDiv');

    var tfData = [];
    var tfList = [];
    var cellList = [];
    var targetList = [];

    var geneSnpToggle = 0;
    var searchURLHandled = 0;

    LoadGWAS();


    function LoadGWAS() 
    {
        d3.tsv("./data/gwas.txt", function(data) {
            gwasData.push(data); 

            // Handle URL input
            if(searchURLHandled == 0)
            {
                var searchText = window.location.search;
                var studyText = searchText.split("&")[0];
                var studyType = studyText.split("=")[1];
                var geneText = searchText.split("&")[1];
                var searchGene = geneText.split("=")[1];
                var datasetText = searchText.split("&")[2];
                var datasetVal = datasetText.split("=")[1];

                console.log('text:', searchText);
                console.log('study:', studyType);
                console.log('gene:', searchGene);
                console.log('dataset:', datasetVal);
                
                if(searchText != "")
                {
                    document.getElementById("geneSearch").innerText = "Modify Query";
                    document.getElementById("queryRow").style.marginTop = "60px";


                    searchGeneURL(studyType, searchGene, datasetVal);
                }

                searchURLHandled = 1;
            }
        });
    }



    function submitQuery() 
    { 
        // No Study (cis/trans) is selected, prompt the user to select study; default: 1 is assigned
        if(study == 0)
        {
            $("#studyEntryRow").show();
            study = 1;

        }

        // Multiple Genes selected for trans study, but the operation (intersection/union) is not defined yet; default: union is assigned 
        if(study == 2 && operation == 0)
        {
            var gene = document.getElementById("geneVal").value;    
            var geneArray = gene.split(',');

            if(geneArray.length > 2)
            {
                $("#operationEntryRow").show();
                operation = 2;
                return;
            }
            
        }


        if(document.getElementById("geneSearch").innerText == "Submit Query")
        {
            datasetvalues = []; //reset datasetvalues
            addDataSets();

             //Transcription Factors
            if(study == 3)
            {
                var gene = document.getElementById("geneVal").value;    
                var geneArray = gene.split(',');

                if(geneArray[geneArray.length-1] == "")
                {
                    geneArray.pop();
                }

                document.getElementById("TFmultipleGeneRow").style.display = "block";

                if(geneArray.length == 0)
                {
                    var extraGeneDiv = document.getElementById("extraGene");
                    extraGeneDiv.style.visibility = "visible";
                    extraGeneDiv.innerHTML = "Please enter gene name(s) in the gene-box for 'TF' queries.";
                    return true;
                }

                datasetvalues.push(001); // reserve dataset for TFs
                displayTransFact();
            }


            if(datasetvalues.length == 0)
            {
                var missingDataDiv = document.getElementById("missingDataset");
                missingDataDiv.style.visibility = "visible";
                return true;
            }

            else if(datasetvalues.length > 0)
            {
                var missingDataDiv = document.getElementById("missingDataset");
                missingDataDiv.style.visibility = "hidden";
            } 

            //Check if cis-eQTL
            if(study == 1)
            {
                var gene = document.getElementById("geneVal").value;    
                var geneArray = gene.split(',');

                if(geneArray[geneArray.length-1] == "")
                {
                    geneArray.pop();
                }

                document.getElementById("TFmultipleGeneRow").style.display = "none";
            }

            if(study == 2)
            {
                var extraGeneDiv = document.getElementById("extraGene");
                extraGeneDiv.style.visibility = "hidden";

                document.getElementById("TFmultipleGeneRow").style.display = "none";
            }

            var infoRow = document.getElementById("immRegExpRow");
            infoRow.style.display = "none";

            var browseRow = document.getElementById("browseRow");
            browseRow.style.display = "none";

            var geneRow = document.getElementById("geneEntryRow");
            geneRow.style.display = "none";

            var privDiv = document.getElementById("privDiv");
            privDiv.style.display = "none";

            var missingDataDiv = document.getElementById("missingDataset");
            missingDataDiv.style.display = "none";

            var multipleGeneNode = document.getElementById("multipleGeneRow");
            multipleGeneNode.style.visibility = "hidden";

            var submitButtonDiv = document.getElementById("submitButton");
            submitButtonDiv.style.textAlign = "left";

            if(document.getElementById("studyEntryRow").style.display == "none")
            {
                var queryRow = document.getElementById("queryRow");
                queryRow.style.marginTop = "60px";
            }
            
            $("#operationEntryRow").hide();

            document.getElementById("geneSearch").innerText = "Modify Query";

            if(gwasRead == 0)
            {
                LoadGWAS();
                //gwasRead = 1;
            }

            searchGene();
        }

        else if(document.getElementById("geneSearch").innerText == "Modify Query")
        {
            var browseRow = document.getElementById("browseRow");
            browseRow.style.display = "block";

            var geneRow = document.getElementById("geneEntryRow");
            geneRow.style.display = "block";
            geneRow.style.marginTop = "60px";

            var privDiv = document.getElementById("privDiv");
            privDiv.style.display = "block";

            var missingDataDiv = document.getElementById("missingDataset");
            missingDataDiv.style.display = "block";

            var submitButtonDiv = document.getElementById("submitButton");
            submitButtonDiv.style.textAlign = "right";

            var queryRow = document.getElementById("queryRow");
            queryRow.style.marginTop = "0px";
            document.getElementById("geneSearch").innerText = "Submit Query";

            if(study == 2)
            {
                var gene = document.getElementById("geneVal").value;    
                var geneArray = gene.split(',');

                if(geneArray.length > 2)
                {
                    $("#operationEntryRow").show();
                }
            }

        }

    }



    function filterUser()
    {
        $('.hipc').hide();
        
        $('.non-hipc').hide();
        $('.tr-fact').hide();
        study = 0;

        $("#studyEntryRow").hide();
        document.getElementById("item1").checked = true;

        $("#operationEntryRow").hide();

        $("#userListMain").show();
        $("#userListCis").show();
        $("#userListTrans").show();

        document.getElementById("userListMain").style.fontWeight = "bold";
        document.getElementById("userListCis").style.fontWeight = "bold";
        document.getElementById("userListTrans").style.fontWeight = "bold";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        uploadStudy();

        document.getElementById("queryType").innerText = "Cis-eQTL";
    }

    
    function filterCisUser()
    {
        $('.hipc').hide();
        $('.non-hipc').hide();
        $('.trans').hide();
        $('.utrans').hide();
        $('.tr-fact').hide();
        study = 1;

        $("#studyEntryRow").hide();
        $("#operationEntryRow").hide();

        var userListCount = document.getElementById('userListCount');
        userListCount.innerText = userCisArr.length + userTransArr.length;

        var userListCisCount = document.getElementById('userListCisCount');
        userListCisCount.innerText = userCisArr.length;

        var userListTransCount = document.getElementById('userListTransCount');
        userListTransCount.innerText = userTransArr.length;
        
        $("#UserListTitle").show();

        for(var i = 0; i < userCisArr.length; i++)
        {
            var sid = "UserListCis" + i;
            $('#'+ sid).show();

            var spanid = 'span100' + i;
            var userCisSpan = document.getElementById(spanid);
            userCisSpan.innerText = userCisFilename[i]  + " - (Cis)";
        } 

        document.getElementById("userListMain").style.fontWeight = "bold";
        document.getElementById("userListCis").style.fontWeight = "bold";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "larger";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Cis-eQTL";
    }

    
    function filterTransUser()
    {
        $('.hipc').hide();
        $('.non-hipc').hide();
        $('.cis').hide();
        $('.ucis').hide();
        $('.tr-fact').hide();
        study = 2;

        $("#studyEntryRow").hide();

        var gene = document.getElementById("geneVal").value;    
        var geneArray = gene.split(',');

        if(geneArray.length > 2)
        {
            $("#operationEntryRow").show();
            operation = 2;
        }

        var userListCount = document.getElementById('userListCount');
        userListCount.innerText = userCisArr.length + userTransArr.length;

        var userListCisCount = document.getElementById('userListCisCount');
        userListCisCount.innerText = userCisArr.length;

        var userListTransCount = document.getElementById('userListTransCount');
        userListTransCount.innerText = userTransArr.length;
        
        $("#UserListTitle").show();

        for(var i = 0; i < userTransArr.length; i++)
        {
            var sid = "UserListTrans" + i;
            $('#'+ sid).show();

            var spanid = 'span110' + i;
            var userTransSpan = document.getElementById(spanid);
            userTransSpan.innerText = userTransFilename[i]  + " - (Trans)";
        } 

        document.getElementById("userListMain").style.fontWeight = "bold";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "bold";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "larger";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Trans-eQTL";
        
    }

    
    function filterNonHIPC()
    {
        $('.hipc').hide();
        $('.user').hide();
        $('.non-hipc').show();
        $('.tr-fact').hide();
        study = 0;
      
        $("#studyEntryRow").show();
        document.getElementById("item1").checked = true;

        $("#operationEntryRow").hide();

        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "bold";
        document.getElementById("listCisNonHIPC").style.fontWeight = "bold";
        document.getElementById("listTransNonHIPC").style.fontWeight = "bold";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Not Selected";
    }

    
    function filterHIPC()
    {
        $('.non-hipc').hide();
        $('.user').hide();
        $('.hipc').show();
        $('.tr-fact').hide();
        study = 0;

        $("#studyEntryRow").show();
        document.getElementById("item1").checked = true;

        $("#operationEntryRow").hide();

        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "bold";
        document.getElementById("listCisHIPC").style.fontWeight = "bold";
        document.getElementById("listTransHIPC").style.fontWeight = "bold";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "bold";

        document.getElementById("queryType").innerText = "Not Selected";
    }


    function filterCisHIPC()
    {
        $('.hipc').show();
        $('.non-hipc').hide();
        $('.trans').hide();
        $('.user').hide();
        $('.tr-fact').hide();
        $('.noqtl').hide();
        study = 1;

        $("#studyEntryRow").hide();
        $("#operationEntryRow").hide();

        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "bold";
        document.getElementById("listCisHIPC").style.fontWeight = "bold";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "larger";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Cis-eQTL";
    }

    
    function filterTransHIPC()
    {
        $('.hipc').show();
        $('.non-hipc').hide();
        $('.cis').hide();
        $('.user').hide();
        $('.tr-fact').hide();
        $('.noqtl').hide();
        study = 2;

        $("#studyEntryRow").hide();

        var gene = document.getElementById("geneVal").value;    
        var geneArray = gene.split(',');

        if(geneArray.length > 2)
        {
            $("#operationEntryRow").show();
            operation = 2;
        }

        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "bold";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "bold";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "larger";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Trans-eQTL";
    }

    
    function filterCisNonHIPC()
    {
        $('.hipc').hide();
        $('.non-hipc').show();
        $('.trans').hide();
        $('.user').hide();
        $('.tr-fact').hide();
        study = 1;

        $("#studyEntryRow").hide();
        $("#operationEntryRow").hide();

        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "bold";
        document.getElementById("listCisNonHIPC").style.fontWeight = "bold";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontWeight = "normal";
        
        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "larger";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Cis-eQTL";
    }

    
    function filterTransNonHIPC()
    {
        $('.hipc').hide();
        $('.non-hipc').show();
        $('.cis').hide();
        $('.user').hide();
        $('.tr-fact').hide();
        study = 2;

        $("#studyEntryRow").hide();

        var gene = document.getElementById("geneVal").value;    
        var geneArray = gene.split(',');

        if(geneArray.length > 2) {
            $("#operationEntryRow").show();
            operation = 2;
        }

        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "bold";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "bold";
        document.getElementById("mainTF").style.fontWeight = "normal";
        

        document.getElementById("mainTF").style.fontSize = "initial";
        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "larger";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Trans-eQTL";
    }

    
    function filterTF()
    {

        $('.hipc').hide();
        $('.user').hide();
        $('.non-hipc').hide();
        
        $("#studyEntryRow").hide();

        $("#operationEntryRow").hide();

        $('.tr-fact').show();

        document.getElementById("mainTF").style.fontWeight = "bold";
        document.getElementById("mainTF").style.fontSize = "medium";


        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";

        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("listHLAHIPC").style.fontWeight = "normal";
        document.getElementById("listHLAHIPC").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "Transcription Factors";

        study = 3;
    }


    function filterHLAType()
    {
        $('.hipc').hide();
        $('.user').hide();
        $('.non-hipc').hide();
        $('.tr-fact').hide();
        
        $("#studyEntryRow").hide();

        $("#operationEntryRow").hide();

        $('.hla').show();

        document.getElementById("listHLAHIPC").style.fontWeight = "bold";
        document.getElementById("listHLAHIPC").style.fontSize = "larger";


        document.getElementById("userListMain").style.fontWeight = "normal";
        document.getElementById("userListCis").style.fontWeight = "normal";
        document.getElementById("userListTrans").style.fontWeight = "normal";
        document.getElementById("listHIPC").style.fontWeight = "normal";
        document.getElementById("listCisHIPC").style.fontWeight = "normal";
        document.getElementById("listTransHIPC").style.fontWeight = "normal";
        document.getElementById("listNonHIPC").style.fontWeight = "normal";
        document.getElementById("listCisNonHIPC").style.fontWeight = "normal";
        document.getElementById("listTransNonHIPC").style.fontWeight = "normal";

        document.getElementById("userListMain").style.fontSize = "initial";
        document.getElementById("userListCis").style.fontSize = "initial";
        document.getElementById("userListTrans").style.fontSize = "initial";
        document.getElementById("listHIPC").style.fontSize = "initial";
        document.getElementById("listCisHIPC").style.fontSize = "initial";
        document.getElementById("listTransHIPC").style.fontSize = "initial";
        document.getElementById("listNonHIPC").style.fontSize = "initial";
        document.getElementById("listCisNonHIPC").style.fontSize = "initial";
        document.getElementById("listTransNonHIPC").style.fontSize = "initial";

        document.getElementById("mainTF").style.fontWeight = "normal";
        document.getElementById("mainTF").style.fontSize = "initial";

        document.getElementById("queryType").innerText = "HLA Type";

        study = 4;
    }

    
    function switchQueryInputGene()
    {
        document.getElementById("enterGeneInput").className = "btn btn-default active";
        document.getElementById("enterSNPInput").className = "btn btn-default";
        document.getElementById("geneVal").placeholder = "Enter HUGO Gene symbols";
        geneSnpToggle = 0;
    }


    function switchQueryInputSNP()
    {
        document.getElementById("enterGeneInput").className = "btn btn-default";
        document.getElementById("enterSNPInput").className = "btn btn-default active";
        document.getElementById("geneVal").placeholder = "Enter SNP IDs";
        geneSnpToggle = 1;
    }


    function displayTransFact()
    {
        var filterRow = document.getElementById("filterRow"); 
            filterRow.style.height = "0px";
            filterRow.style.visibility = "hidden";

        var filterCol = document.getElementById("filterCol"); 
            filterCol.style.height = "0px";
            filterCol.style.visibility = "hidden";

        var plotRow = document.getElementById("plotRow");
            plotRow.style.height = "0px";
            plotRow.style.visibility = "hidden";
        
        var plotCol = document.getElementById("plotCol");
            plotCol.style.height = "0px";

        var gene = document.getElementById("geneVal").value;

        var lastChar = gene.substr(gene.length - 1);
        
        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);

        $('#TFmultipleGeneRow').empty();
        
        var geneArray = gene.split(',');


        if(geneArray.length > 1)
        {
            var TFmultipleGeneNode = document.getElementById("TFmultipleGeneRow");
            document.getElementById("TFmultipleGeneRow").style.display = "block";
            TFmultipleGeneNode.style.visibility = "visible";

            for (var i = 0; i < geneArray.length; i++)
            {
                var b = document.createElement('a');
                var linktextcontent = geneArray[i];
                var linkText = document.createTextNode(linktextcontent);
                b.style.textAlign = "center";
                if(i != 0)
                    b.style.marginLeft = "10px";
                b.appendChild(linkText);
                TFmultipleGeneNode.appendChild(b);
                
                b.onclick = function() {
                                gene = $(event.target).text();
                                var datafile = "./data/tf-meta/" + gene;

                                d3.json(datafile, function(data) {
                                    vizTF(data, gene);
                                });
                            }
            }

        }

        gene = geneArray[0];

        var datafile = "./data/tf-meta/" + gene;

        d3.json(datafile, function(data) {
            vizTF(data, gene);
        });
    }

    
    function vizTF(tfData, gene) 
    {
        tfList = [];
        cellList = [];
        targetList = [];

        $('#igvDiv').empty(); // Remove contents of igvDiv (cis eqtls)
            var igvDiv = document.getElementById("igvDiv"); 
            igvDiv.style.visibility = "hidden";

            $('#eqtlDataDiv').empty(); // Remove eqtl details 

            var transDiv = document.getElementById("transDiv");

            $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details
            
            if(transDiv === null) 
                createTransDiv();


        // TF TABLE INFO...
        var tfInfoNode = document.getElementById("tfInfoDiv");
        var tfInfoInnerNode = document.getElementById("tfInfoInnerDiv");

        if(tfInfoNode === null) {
            tfInfoNode = document.createElement('div'); 
            
            tfInfoNode.id = "tfInfoDiv";
            tfInfoNode.className = "row";
            tfInfoNode.style.marginTop = "0px";
            tfInfoNode.style.marginBottom = "20px";
            tfInfoNode.style.marginLeft = "-5px";
            tfInfoNode.style.fontSize = "14px";
            tfInfoNode.style.fontWeight = "bold";
            tfInfoNode.style.visibility = "visible";

            var transDiv = document.getElementById("transDiv");
            transDiv.appendChild(tfInfoNode);
        }

        if(tfInfoInnerNode === null) {
            tfInfoInnerNode = document.createElement('div');

            tfInfoInnerNode.id = "tfInfoInnerDiv";
            tfInfoInnerNode.className = "col-md-11";
            tfInfoInnerNode.style.border = "1px solid #ccc";
            tfInfoInnerNode.style.paddingLeft = "15px";
            tfInfoInnerNode.style.paddingRight = "15px";
            tfInfoInnerNode.style.paddingTop = "5px";
            tfInfoInnerNode.style.paddingBottom = "5px";

            tfInfoNode.appendChild(tfInfoInnerNode);
        }

        tfInfoInnerNode.style.fontSize = "16px";
        tfInfoInnerNode.style.fontWeight = "bold";
        tfInfoInnerNode.innerHTML = "Target Gene: " + gene; 


        var d = document.createElement('span');
        d.className = "badge";
        d.style.color = "black";
        d.style.fontSize = "15px";
        d.style.fontWeight = "normal";
        d.style.marginTop = "5px";
        d.style.marginBottom = "5px";
        d.style.marginLeft = "25px";
        var dtext = tfData.length + " total hits: ";
        var dnode = document.createTextNode(dtext);
        
        d.appendChild(dnode);
        tfInfoInnerNode.appendChild(d);

        // Append # of Cell Types
        var countCellData = _.countBy(tfData, function(obj){
            return obj.c.replace(" ", "_");
        }); 

        var countCellLen = Object.keys(countCellData).length;

        var d = document.createElement('span');
        d.className = "badge";
        d.style.background = "steelblue";
        d.style.color = "white";
        d.style.marginTop = "5px";
        d.style.marginBottom = "5px";
        d.style.marginLeft = "10px";
        var dtext = countCellLen + " cell types";
        var dnode = document.createTextNode(dtext);
        
        d.style.fontSize = "15px";
        d.style.fontWeight = "normal";
        d.appendChild(dnode);
        tfInfoInnerNode.appendChild(d);

        
        // Append # of Target Types
        var countTargetData = _.countBy(tfData, function(obj){
            return obj.t.replace(" ", "_");
        }); 

        var countTargetLen = Object.keys(countTargetData).length;

        var d = document.createElement('span');
        d.className = "badge";
        d.style.background = "steelblue";
        d.style.color = "white";
        d.style.marginTop = "5px";
        d.style.marginBottom = "5px";
        d.style.marginLeft = "10px";
        var dtext = countTargetLen + " targets";
        var dnode = document.createTextNode(dtext);
        d.style.fontSize = "15px";
        d.style.fontWeight = "normal";
        d.appendChild(dnode);
        tfInfoInnerNode.appendChild(d);

        // TF TABLE CREATION...    
        var tfTableNode = document.getElementById("tfTableDiv");

        if(tfTableNode === null) 
        {
            var transDiv = document.getElementById("transDiv");
            tfTableNode = document.createElement('table');

            tfTableNode.id = "tfTableDiv";
            tfTableNode.className = "display compact";
            tfTableNode.style.width = "100%";
            tfTableNode.style.border = "1px solid #ccc";
            tfTableNode.style.marginTop = "20px";
            tfTableNode.style.marginLeft = "0px";
            tfTableNode.style.padding = "5px";
            tfTableNode.style.fontSize = "normal";
            tfTableNode.style.visibility = "visible";

            transDiv.appendChild(tfTableNode);
        }

        //Table Initialization
        var header = tfTableNode.createTHead();                    
        var tblBody = document.createElement("tbody");
        tfTableNode.appendChild(tblBody);
        var th = document.createElement('th');
        th.innerHTML = "Gene";
        th.innerHTML = "P-Value";


        var table = $('#tfTableDiv').DataTable( {
                        "bInfo" : false,
                        data: tfData,
                        columnDefs: [ {
                        orderable: false,
                        className: 'select-checkbox',
                        targets:   0
                    } ],

                    select: {
                        style:    'multi',
                        selector: 'td:first-child'
                    },

                    order: [[ 1, 'asc' ]],
                    
                    columns: [
                        { data: "l", "title": 'Select',
                            "render": function ( data, type, full, meta ) {
                                return "";
                            }
                        },

                        { data: 'c', "title": 'Cell Type' },

                        { data: 't', "title": 'Transcription Factor' },

                        { data: 'l', "title": 'Study Details',  
                            "render": function ( data, type, full, meta ) {
                                    var text = "View: ENCODE";
                                    var pmed = "https://www.encodeproject.org/files/" + data;
                                    var pmedlink = "<a href = " + pmed + " target=\"_blank\">" + text + "</a>";
                                    return pmedlink;
                            } 
                        }           
                    ]
                } );

            table.on( 'select', function (e,dt,type,indexes) {
                var data = table.rows( indexes ).data();
                console.log('add:::', data[0]);
                addTF(this, data[0].l, data[0].c, data[0].t);
            } );

            table.on( 'deselect', function (e,dt,type,indexes) {
                var data = table.rows( indexes ).data();
                console.log('remove:::', data[0]);
                removeTF(this, data[0].l, data[0].c, data[0].t);
            } );

                            
            $('#tfTableDiv').append('<caption> Table: Transcription Factor ChIP-Seq Studies </caption>');

            var tableDiv = document.getElementById("tfTableDiv_wrapper");
            tableDiv.style.visibility = "visible";

        
        // Create button to Visualize Tracks
        var sButton = document.getElementById("geneSearch");
        var vizClone = sButton.cloneNode(true);
        vizClone.id = "vizListButton";
        vizClone.innerHTML = "Submit";

        vizClone.onclick = function() { vizTFTracks(gene); }
        
        vizClone.style.visibility = "visible";
        vizClone.style.marginTop = "20px";
        vizClone.style.height = "40px";
        vizClone.style.fontSize = "medium";
        vizClone.style.fontWeight = "400";

        transDiv.appendChild(vizClone);
    }


    function addTF(obj, tf, cell, target)
    {
        tfList.push(tf);
        cellList.push(cell);
        targetList.push(target);

        var vizButton = document.getElementById("vizListButton");

        if(tfList.length == 0)
            vizButton.innerHTML = "Submit";

        if(tfList.length > 0)
            vizButton.innerHTML = "Submit (" + tfList.length + ")" ;
    }


    function removeTF(obj, tf, cell, target)
    { 
        console.log('remove cell:', cell);
                console.log('remove target:', target);

        for( var i = 0; i < tfList.length; i++){ 
           if ( tfList[i] === tf) {
             tfList.splice(i, 1); 
           }
        }
        
        for( var i = 0; i < cellList.length; i++){ 
           if ( cellList[i] === cell) {
             cellList.splice(i, 1); 
           }
        }
        
        for( var i = 0; i < targetList.length; i++){ 
           if ( targetList[i] === target) {
             targetList.splice(i, 1); 
           }
        }

        var vizButton = document.getElementById("vizListButton");

        if(tfList.length == 0)
            vizButton.innerHTML = "Submit";

        if(tfList.length > 0)
            vizButton.innerHTML = "Submit (" + tfList.length + ")" ;
    }


    function vizTFTracks(gene)
    {
        var cisDiv = document.getElementById("igvDiv");
        cisDiv.style.visibility = "visible";
        cisDiv.style.marginTop = "10px";

        var div = $("#igvDiv")[0],
                options = {
                    showNavigation: true,
                    showRuler: true,
                    locus: gene,
                    showCursorTrackingGuide: true,
                    showCenterGuide: true,
                    reference: {
                        id: "hg19"
                    },
                    
                    formats: {
                            gwasSNPS: {
                                coords: 0,
                                chr: 0,
                                start: 1,
                                end: 2,
                                fields: ["Chr", "chromStart", "chromEnd", "Strongest_SNP_risk_allele", "Disease_Phenotype", "P-value", "Odds ratio or beta", "PUBMEDID"]
                            }
                        },
                    
                    tracks: [

                            {
                                featureType: 'annotation',
                                format: 'gwasSNPS',
                                indexed: false,
                                color: 'rgb(100,200,200)',
                                displayMode: 'EXPANDED',
                                url: "./data/gwas.txt",
                                name: 'GWAS Catalog',
                                height: "150",
                                order: 9999
                            },

                            {
                                name: "Genes",
                                type: "annotation",
                                format: "gtf",
                                sourceType: "file",
                                height: "300",
                                url: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg19/genes/gencode.v18.annotation.sorted.gtf.gz",
                                indexURL: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg19/genes/gencode.v18.annotation.sorted.gtf.gz.tbi",
                                visibilityWindow: 10000000,
                                order: Number.MAX_VALUE,
                                order: 10000,
                                displayMode: "EXPANDED"
                            }
                    ]
                };

        igv.createBrowser(div, options);

        setTimeout(loadTFTracks, 1000, tfList, cellList, targetList);

        gwasRead = 1;

    }


    function loadTFTracks(tfList, cellList, targetList)
    {

        for(var i = 0; i < tfList.length; i++)
        {

            var trackName = "Cell: " + cellList[i] + "<br/>" + "TF: " + targetList[i];
            var trackUrl = "https://www.encodeproject.org/files/" + tfList[i] + "/@@download/" + tfList[i] + ".bed.gz";

            igv.browser.loadTrack({
                                featureType: 'annotation',
                                format: 'bed',
                                indexed: false,
                                color: 'rgb(100,200,200)',
                                displayMode: 'EXPANDED',
                                url: trackUrl,
                                name: trackName,
                                height: "100"
                                });

        }
    }


    function selectCis()
    {
        $('.cis').show();
        $('.trans').hide();
        study = 1;

        $("#operationEntryRow").hide();

        document.getElementById("queryType").innerText = "Cis-eQTL";
    }


    function selectTrans()
    {
        $('.trans').show();
        $('.cis').hide();
        study = 2;

        var gene = document.getElementById("geneVal").value;    
        var geneArray = gene.split(',');

        if(geneArray.length > 2) {
            $("#operationEntryRow").show();
            operation = 2;
        }

        document.getElementById("queryType").innerText = "Trans-eQTL";
    }

    
    function selectInt()
    {
        operation = 1;
    }

    
    function selectUnion()
    {
        operation = 2;
    }


    function exampleQuery1()
    {
        filterCisNonHIPC();
        datasetvalues = [];
        deselectStudies();
        geneSnpToggle = 1;
        switchQueryInputGene();

        // Set datasets
        datasetvalues.push(61);
        datasetvalues.push(62);
        datasetvalues.push(63);
        document.getElementById("study61").checked = true;
        document.getElementById("study62").checked = true;
        document.getElementById("study63").checked = true;

        var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
        document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected"; 
        document.getElementById('deselectDiv').style.visibility = "visible";

        document.getElementById("geneVal").innerHTML = "JAK2"; // Set Gene Value
        document.getElementById("geneVal").value = "JAK2";

            $('#transRow').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details 
            $('#cisLegendClone').remove();
            document.getElementById("TFmultipleGeneRow").style.display = "none";

            var filterRow = document.getElementById("filterRow"); 
            filterRow.style.height = "0px";
            filterRow.style.visibility = "hidden";

            var plotRow = document.getElementById("plotRow");
            plotRow.style.height = "0px";
            plotRow.style.visibility = "hidden";

            var manplot = document.getElementById("manPlot");
            manplot.style.height = "0px";

            var filterCol = document.getElementById("filterCol"); 
            filterCol.style.height = "0px";
            filterCol.style.visibility = "hidden";
            
            removeControls();

            var infoRow = document.getElementById("immRegExpRow");
            infoRow.style.display = "none";

            var browseRow = document.getElementById("browseRow");
            browseRow.style.display = "none";

            var geneRow = document.getElementById("geneEntryRow");
            geneRow.style.display = "none";

            var privDiv = document.getElementById("privDiv");
            privDiv.style.display = "none";

            var missingDataDiv = document.getElementById("missingDataset");
            missingDataDiv.style.display = "none";

            var submitButtonDiv = document.getElementById("submitButton");
            submitButtonDiv.style.textAlign = "left";

            if(document.getElementById("studyEntryRow").style.display == "none")
            {
                var queryRow = document.getElementById("queryRow");
                queryRow.style.marginTop = "60px";
            }
            
            $("#operationEntryRow").hide();

            document.getElementById("geneSearch").innerText = "Modify Query";

            if(gwasRead == 0)
            {
                LoadGWAS();
                gwasRead = 1;
            }


            drawCis(datasetvalues);
    }


    function exampleQuery2()
    {
        filterTransNonHIPC();
        datasetvalues = [];
        deselectStudies();
        geneSnpToggle = 1;
        switchQueryInputGene();

        // Set datasets
        datasetvalues.push(13);
        document.getElementById("westra").checked = true;

        var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
        document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected"; 
        document.getElementById('deselectDiv').style.visibility = "visible";

        document.getElementById("geneVal").innerHTML = "IFNA1,IFNB1,IFNG,IL10,IL12A,IL12B,IL13,IL15,IL16,IL17A,IL18,IL1A,IL2,IL3,IL4,IL5,IL6,IL8,IL9,LTA,TNF,TXLNA,"; // Set Gene Value
        document.getElementById("geneVal").value = "IFNA1,IFNB1,IFNG,IL10,IL12A,IL12B,IL13,IL15,IL16,IL17A,IL18,IL1A,IL2,IL3,IL4,IL5,IL6,IL8,IL9,LTA,TNF,TXLNA,"; // Set Gene Value


        var gene = document.getElementById("geneVal").value;
        var lastChar = gene.substr(gene.length - 1);

        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);

        operation = 2;

            $('#igvDiv').empty(); // Remove contents of igvDiv (cis eqtls)
            var igvDiv = document.getElementById("igvDiv"); 
            igvDiv.style.visibility = "hidden";
            document.getElementById("TFmultipleGeneRow").style.display = "none";

            //$('#eqtlDataDiv').empty(); // Remove eqtl details 

            var transDiv = document.getElementById("transDiv");
            $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details
            $('#cisLegendClone').remove();
            
            if(transDiv === null) 
                createTransDiv();

            
            if(operation == 2 || operation == 0)
            {
                var plotRow = document.getElementById("plotRow"); 
                plotRow.style.visibility = "visible";
                
                var plotCol = document.getElementById("plotCol");
                plotCol.style.height = "50px";

                var manplot = document.getElementById("manPlot");
                manplot.style.height = "40px";

                var filterCol = document.getElementById("filterCol"); 
                filterCol.style.height = "45px";
                filterCol.style.visibility = "visible";
            }

            var vizRow = document.getElementById("transRow"); 
            vizRow.style.visibility = "visible";

            var infoRow = document.getElementById("immRegExpRow");
            infoRow.style.display = "none";

            var browseRow = document.getElementById("browseRow");
            browseRow.style.display = "none";

            var geneRow = document.getElementById("geneEntryRow");
            geneRow.style.display = "none";

            var privDiv = document.getElementById("privDiv");
            privDiv.style.display = "none";

            var multipleGeneNode = document.getElementById("multipleGeneRow");
            multipleGeneNode.style.visibility = "hidden";

            var TFmultipleGeneNode = document.getElementById("TFmultipleGeneRow");
            document.getElementById("TFmultipleGeneRow").style.display = "none";

            var missingDataDiv = document.getElementById("missingDataset");
            //missingDataDiv.style.display = "none";
            missingDataDiv.style.visibility = "visible";
            var missingDataText = document.getElementById("missingDataText");
            missingDataText.innerHTML = "Loading, please wait...";

            var submitButtonDiv = document.getElementById("submitButton");
            submitButtonDiv.style.textAlign = "left";

            if(document.getElementById("studyEntryRow").style.display == "none")
            {
                var queryRow = document.getElementById("queryRow");
                queryRow.style.marginTop = "60px";
            }
            
            $("#operationEntryRow").hide();

            document.getElementById("geneSearch").innerText = "Modify Query";

            if(gwasRead == 0)
            {
                LoadGWAS();
                //gwasRead = 1;
            }
 
            addTrans(gene);

    }


    function exampleQuery3()
    {
        filterTF();
        datasetvalues = [];
        deselectStudies();
        geneSnpToggle = 1;
        switchQueryInputGene();

        // Set datasets
        datasetvalues.push(001);
        document.getElementById("TfAll").checked = true;

        var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
        document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected"; 
        document.getElementById('deselectDiv').style.visibility = "visible";

        document.getElementById("geneVal").innerHTML = "IL15RA"; // Set Gene Value
        document.getElementById("geneVal").value = "IL15RA"; // Set Gene Value
         

            $('#transRow').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details 
            
            $('#igvDiv').empty(); // Remove contents of igvDiv (cis eqtls)
            var igvDiv = document.getElementById("igvDiv"); 
            igvDiv.style.visibility = "hidden";
            $('#cisLegendClone').remove();

            var filterRow = document.getElementById("filterRow"); 
            filterRow.style.height = "0px";
            filterRow.style.visibility = "hidden";

            var plotRow = document.getElementById("plotRow");
            plotRow.style.height = "0px";
            plotRow.style.visibility = "hidden";

            var manplot = document.getElementById("manPlot");
            manplot.style.height = "0px";

            var filterCol = document.getElementById("filterCol"); 
            filterCol.style.height = "0px";
            filterCol.style.visibility = "hidden";
            
            removeControls();

            var infoRow = document.getElementById("immRegExpRow");
            infoRow.style.display = "none";

            var browseRow = document.getElementById("browseRow");
            browseRow.style.display = "none";

            var geneRow = document.getElementById("geneEntryRow");
            geneRow.style.display = "none";

            var privDiv = document.getElementById("privDiv");
            privDiv.style.display = "none";

            var missingDataDiv = document.getElementById("missingDataset");
            missingDataDiv.style.display = "none";

            var multipleGeneNode = document.getElementById("multipleGeneRow");
            multipleGeneNode.style.visibility = "hidden";

            var TFmultipleGeneNode = document.getElementById("TFmultipleGeneRow");
            document.getElementById("TFmultipleGeneRow").style.display = "none";

            var submitButtonDiv = document.getElementById("submitButton");
            submitButtonDiv.style.textAlign = "left";

            if(document.getElementById("studyEntryRow").style.display == "none")
            {
                var queryRow = document.getElementById("queryRow");
                queryRow.style.marginTop = "60px";
            }
            
            $("#operationEntryRow").hide();

            document.getElementById("geneSearch").innerText = "Modify Query";

            if(gwasRead == 0)
            {
                LoadGWAS();
                gwasRead = 1;
            }
 
            displayTransFact();
    }


    function addDataSets()
    {
        if(document.getElementById("study101").checked == true){
             datasetvalues.push(101);
             study = 1;
         }

         if(document.getElementById("study111").checked == true) {
             datasetvalues.push(111);
             study = 2;
         }

         if(document.getElementById("study102").checked == true){
             datasetvalues.push(102);
             study = 1;
         }

         if(document.getElementById("study112").checked == true){
             datasetvalues.push(112);
             study = 2;
         }

         if(document.getElementById("study103").checked == true){
             datasetvalues.push(103);
             study = 1;
         }

         if(document.getElementById("study113").checked == true){
             datasetvalues.push(113);
             study = 2;
         }

        if(document.getElementById("framingham").checked == true)
             datasetvalues.push(11);

        if(document.getElementById("study001").checked == true)
             datasetvalues.push(001);

        if(document.getElementById("study002").checked == true)
             datasetvalues.push(002);

        if(document.getElementById("study003").checked == true)
             datasetvalues.push(003); 

        if(document.getElementById("gtex").checked == true)
             datasetvalues.push(12);

        if(document.getElementById("westra").checked == true)
             datasetvalues.push(13);

         if(document.getElementById("study14").checked == true)
             datasetvalues.push(14);

         if(document.getElementById("study15").checked == true)
             datasetvalues.push(15);

         if(document.getElementById("study16").checked == true)
             datasetvalues.push(16);

         if(document.getElementById("study17").checked == true)
             datasetvalues.push(17);

         if(document.getElementById("study18").checked == true)
             datasetvalues.push(18);

         if(document.getElementById("study19").checked == true)
             datasetvalues.push(19);

         if(document.getElementById("study190").checked == true)
             datasetvalues.push(190);

         if(document.getElementById("study21").checked == true)
             datasetvalues.push(21);

         if(document.getElementById("study22").checked == true)
             datasetvalues.push(22);

         if(document.getElementById("study23").checked == true)
             datasetvalues.push(23);

         if(document.getElementById("study24").checked == true)
             datasetvalues.push(24);

         if(document.getElementById("study25").checked == true)
             datasetvalues.push(25);

         if(document.getElementById("study26").checked == true)
             datasetvalues.push(26);

         if(document.getElementById("study27").checked == true)
             datasetvalues.push(27);

         if(document.getElementById("study28").checked == true)
             datasetvalues.push(28);

         if(document.getElementById("study29").checked == true)
             datasetvalues.push(29);

         if(document.getElementById("study290").checked == true)
             datasetvalues.push(290);

         if(document.getElementById("study31").checked == true)
             datasetvalues.push(31);

         if(document.getElementById("study32").checked == true)
             datasetvalues.push(32);

         if(document.getElementById("study33").checked == true)
             datasetvalues.push(33);

         if(document.getElementById("study34").checked == true)
             datasetvalues.push(34);

         if(document.getElementById("study35").checked == true)
             datasetvalues.push(35);

         if(document.getElementById("study36").checked == true)
             datasetvalues.push(36);

         if(document.getElementById("study37").checked == true)
             datasetvalues.push(37);

         if(document.getElementById("study38").checked == true)
             datasetvalues.push(38);

         if(document.getElementById("study39").checked == true)
             datasetvalues.push(39);

         if(document.getElementById("study390").checked == true)
             datasetvalues.push(390);

         if(document.getElementById("study391").checked == true)
             datasetvalues.push(391);

         if(document.getElementById("study392").checked == true)
             datasetvalues.push(392);

         if(document.getElementById("study393").checked == true)
             datasetvalues.push(393);

         if(document.getElementById("study394").checked == true)
             datasetvalues.push(394);

         if(document.getElementById("study395").checked == true)
             datasetvalues.push(395);

         if(document.getElementById("study41").checked == true)
             datasetvalues.push(41);

         if(document.getElementById("study42").checked == true)
             datasetvalues.push(42);

         if(document.getElementById("study51").checked == true)
            datasetvalues.push(51);

         if(document.getElementById("study52").checked == true)
             datasetvalues.push(52);

         if(document.getElementById("study53").checked == true)
             datasetvalues.push(53);

         if(document.getElementById("study61").checked == true)
             datasetvalues.push(61);

         if(document.getElementById("study62").checked == true)
             datasetvalues.push(62);

         if(document.getElementById("study63").checked == true)
             datasetvalues.push(63);

         if(document.getElementById("study71").checked == true)
             datasetvalues.push(71);

         if(document.getElementById("study72").checked == true)
             datasetvalues.push(72);

         if(document.getElementById("study73").checked == true)
             datasetvalues.push(73);

         if(document.getElementById("study74").checked == true)
             datasetvalues.push(74);

         if(document.getElementById("study75").checked == true)
             datasetvalues.push(75);

         if(document.getElementById("study76").checked == true)
             datasetvalues.push(76);

         if(document.getElementById("study77").checked == true)
             datasetvalues.push(77);

         if(document.getElementById("study78").checked == true)
             datasetvalues.push(78);

         if(document.getElementById("study79").checked == true)
             datasetvalues.push(79);

         if(document.getElementById("study790").checked == true)
             datasetvalues.push(790);

         if(document.getElementById("study791").checked == true)
             datasetvalues.push(791);

         if(document.getElementById("study792").checked == true)
             datasetvalues.push(792);

         if(document.getElementById("study81").checked == true)
             datasetvalues.push(81);

         if(document.getElementById("study82").checked == true)
             datasetvalues.push(82);

         if(document.getElementById("study83").checked == true)
             datasetvalues.push(83);

         if(document.getElementById("study84").checked == true)
             datasetvalues.push(84);

         if(document.getElementById("study85").checked == true)
             datasetvalues.push(85);

         if(document.getElementById("study86").checked == true)
             datasetvalues.push(86);

         if(document.getElementById("study87").checked == true)
             datasetvalues.push(87);

         if(document.getElementById("study88").checked == true)
             datasetvalues.push(88);

         if(document.getElementById("study89").checked == true)
             datasetvalues.push(89);

         if(document.getElementById("study890").checked == true)
             datasetvalues.push(890);

         if(document.getElementById("study891").checked == true)
             datasetvalues.push(891);

         if(document.getElementById("study892").checked == true)
             datasetvalues.push(892);

         if(document.getElementById("study893").checked == true)
             datasetvalues.push(893);

         if(document.getElementById("study91").checked == true)
             datasetvalues.push(91);

         if(document.getElementById("study92").checked == true)
             datasetvalues.push(92);

         if(document.getElementById("study93").checked == true)
             datasetvalues.push(93);

         if(document.getElementById("study211").checked == true)
             datasetvalues.push(211);

         if(document.getElementById("study212").checked == true)
             datasetvalues.push(212);

         if(document.getElementById("study213").checked == true)
             datasetvalues.push(213);

         if(document.getElementById("study221").checked == true)
             datasetvalues.push(221);

         if(document.getElementById("study222").checked == true)
             datasetvalues.push(222);

         if(document.getElementById("study223").checked == true)
             datasetvalues.push(223);

         if(document.getElementById("study224").checked == true)
             datasetvalues.push(224);

         if(document.getElementById("study225").checked == true)
             datasetvalues.push(225);

         if(document.getElementById("study231").checked == true)
             datasetvalues.push(231);

         if(document.getElementById("study232").checked == true)
             datasetvalues.push(232);

         if(document.getElementById("study233").checked == true)
             datasetvalues.push(233);

         if(document.getElementById("study234").checked == true)
             datasetvalues.push(234);

         if(document.getElementById("study241").checked == true)
             datasetvalues.push(241);

         if(document.getElementById("study242").checked == true)
             datasetvalues.push(242);

         if(document.getElementById("study243").checked == true)
             datasetvalues.push(243);

         if(document.getElementById("study244").checked == true)
             datasetvalues.push(244);

         if(document.getElementById("study245").checked == true)
             datasetvalues.push(245);

         if(document.getElementById("study301").checked == true)
             datasetvalues.push(301);

         if(document.getElementById("study302").checked == true)
             datasetvalues.push(302);

         if(document.getElementById("study401").checked == true)
             datasetvalues.push(401);

         if(document.getElementById("study402").checked == true)
             datasetvalues.push(402);

         if(document.getElementById("study403").checked == true)
             datasetvalues.push(403);

         // Datasets added in July 2020

        if(document.getElementById("study94").checked == true)
            datasetvalues.push(94);

        if(document.getElementById("study396").checked == true)
            datasetvalues.push(396);

        //if(document.getElementById("study191").checked == true)
          //  datasetvalues.push(191);

        if(document.getElementById("study291").checked == true)
            datasetvalues.push(291);

        if(document.getElementById("study292").checked == true)
            datasetvalues.push(292);

        if(document.getElementById("study293").checked == true)
            datasetvalues.push(293);

        if(document.getElementById("study397").checked == true)
            datasetvalues.push(397);

        if(document.getElementById("study95").checked == true)
            datasetvalues.push(95);

        if(document.getElementById("study43").checked == true)
            datasetvalues.push(43);

        if(document.getElementById("study96").checked == true)
            datasetvalues.push(96);

        if(document.getElementById("study97").checked == true)
            datasetvalues.push(97);

        if(document.getElementById("study98").checked == true)
            datasetvalues.push(98);

        if(document.getElementById("study990").checked == true)
            datasetvalues.push(990);

        if(document.getElementById("study294").checked == true)
            datasetvalues.push(294);

        if(document.getElementById("study64").checked == true)
            datasetvalues.push(64);

        if(document.getElementById("study65").checked == true)
            datasetvalues.push(65);

        if(document.getElementById("study66").checked == true)
            datasetvalues.push(66);

        if(document.getElementById("study67").checked == true)
            datasetvalues.push(67);

        if(document.getElementById("study68").checked == true)
            datasetvalues.push(68);

        if(document.getElementById("study69").checked == true)
            datasetvalues.push(69);

        if(document.getElementById("study99").checked == true)
            datasetvalues.push(99);

        if(document.getElementById("study131").checked == true)
            datasetvalues.push(131);

        if(document.getElementById("study132").checked == true)
            datasetvalues.push(132);

        if(document.getElementById("study133").checked == true)
            datasetvalues.push(133);

        if(document.getElementById("study134").checked == true)
            datasetvalues.push(134);

        if(document.getElementById("study135").checked == true)
            datasetvalues.push(135);

        if(document.getElementById("study136").checked == true)
            datasetvalues.push(136);

    }

    
    function deselectStudies() {
        var allcheckboxes = document.querySelectorAll('input.studyopt');
            
        for(var i = 0; i < allcheckboxes.length; i++)
            allcheckboxes[i].checked = false;

        var emorycheckall = document.getElementById('EmoryCheckAll');
        emorycheckall.checked = false;

        var mayocheckall = document.getElementById('MayoCheckAll');
        mayocheckall.checked = false;

        var nihcheckall = document.getElementById('NihCheckAll');
        nihcheckall.checked = false;

        var yalecheckall = document.getElementById('YaleCheckAll');
        yalecheckall.checked = false;

        var chikvcheckall = document.getElementById('ChikvCheckAll');
        chikvcheckall.checked = false;

        var zikvcheckall = document.getElementById('ZikvCheckAll');
        zikvcheckall.checked = false;

        //Update Study Count
        var allcheckedcount = document.querySelectorAll('input.studyopt:checked').length;
        document.getElementById('studyCountText').innerText = allcheckedcount + " studies selected"; 

        document.getElementById('deselectDiv').style.visibility = "hidden";   
    
    }


    function submitCisData()
    {
        var cisfileList = document.getElementById("cisfileList");
        var list = document.createElement("ul");
        cisfileList.appendChild(list);

        var li = document.createElement("li");
        list.appendChild(li);
    Â    var info = document.createElement("span");

        userCisArr[userCisDataId] = document.getElementById("cisDataVal").value;
        
        var file_full = userCisArr[userCisDataId];
        var fileSplit = file_full.split('/');
        var fileName = fileSplit[fileSplit.length-1]; 
        var fileNoExt = fileName.split('.gz');
        var fileNameClean = fileNoExt[0];       
        
        info.innerHTML = fileNameClean; 
    Â    li.appendChild(info);

        userCisFileLink[userCisDataId] = file_full;
        userCisFilename[userCisDataId] = fileNameClean;

        userCisDataId++;
    }


    function sampleCisData()
    {
        var cisfileList = document.getElementById("cisfileList");
        var list = document.createElement("ul");
        cisfileList.appendChild(list);

        var li = document.createElement("li");
        list.appendChild(li);
    Â    var info = document.createElement("span");

        document.getElementById("cisDataVal").value = "https://kalays01.u.hpc.mssm.edu/sample-cis.txt.gz";

        userCisArr[userCisDataId] = document.getElementById("cisDataVal").value;
        
        var file_full = userCisArr[userCisDataId];
        var fileSplit = file_full.split('/');
        var fileName = fileSplit[fileSplit.length-1]; 
        var fileNoExt = fileName.split('.gz');
        var fileNameClean = fileNoExt[0];       
        
        info.innerHTML = fileNameClean; 
    Â    li.appendChild(info);

        userCisFileLink[userCisDataId] = file_full;
        userCisFilename[userCisDataId] = fileNameClean;

        userCisDataId++;
        document.getElementById("cisSampleUpload").className = "btn btn-primary disabled";
    }


    function submitTransData()
    {
        var transfileList = document.getElementById("transfileList");
        var list = document.createElement("ul");
        transfileList.appendChild(list);

        var li = document.createElement("li");
        list.appendChild(li);
    Â    var info = document.createElement("span");

        userTransArr[userTransDataId] = document.getElementById("transDataVal").value;
        
        var file_full = userTransArr[userTransDataId];
        var fileSplit = file_full.split('/');
        var fileName = fileSplit[fileSplit.length-1]; 
        var fileNoExt = fileName.split('.gz');
        var fileNameClean = fileNoExt[0];       
        
        info.innerHTML = fileNameClean; 
    Â    li.appendChild(info);

        userTransFileLink[userTransDataId] = file_full;
        userTransFilename[userTransDataId] = fileNameClean;

        userTransDataId++;
    }


    function sampleTransData()
    {
        var transfileList = document.getElementById("transfileList");
        var list = document.createElement("ul");
        transfileList.appendChild(list);

        var li = document.createElement("li");
        list.appendChild(li);
    Â    var info = document.createElement("span");

        document.getElementById("transDataVal").value = "https://kalays01.u.hpc.mssm.edu/sample-trans.txt";

        userTransArr[userTransDataId] = document.getElementById("transDataVal").value;
        
        var file_full = userTransArr[userTransDataId];
        var fileSplit = file_full.split('/');
        var fileName = fileSplit[fileSplit.length-1]; 
        var fileNoExt = fileName.split('.gz');
        var fileNameClean = fileNoExt[0];       
        
        info.innerHTML = fileNameClean; 
    Â    li.appendChild(info);

        userTransFileLink[userTransDataId] = file_full;
        userTransFilename[userTransDataId] = fileNameClean;

        userTransDataId++;
        document.getElementById("transSampleUpload").className = "btn btn-primary disabled";
    }


    function uploadStudy()
    {
        $("#userListMain").show();
        $("#userListCis").show();
        $("#userListTrans").show();

        //$("#allStudies li:contains(User)").show();

        var userListCount = document.getElementById('userListCount');
        userListCount.innerText = userCisArr.length + userTransArr.length;

        var userListCisCount = document.getElementById('userListCisCount');
        userListCisCount.innerText = userCisArr.length;

        var userListTransCount = document.getElementById('userListTransCount');
        userListTransCount.innerText = userTransArr.length;
        
        //$('.user').show();
        $("#UserListTitle").show();

        for(var i = 0; i < userTransArr.length; i++)
        {
            var sid = "UserListTrans" + i;
            $('#'+ sid).show();

            var spanid = 'span110' + i;
            var userTransSpan = document.getElementById(spanid);
            userTransSpan.innerText = userTransFilename[i] + " - (Trans)";
        } 

        for(var i = 0; i < userCisArr.length; i++)
        {
            var sid = "UserListCis" + i;
            $('#'+ sid).show();

            var spanid = 'span100' + i;
            var userCisSpan = document.getElementById(spanid);
            userCisSpan.innerText = userCisFilename[i] + " - (Cis)";
        } 
    }


    function loadFile()
    {       
        var file = document.querySelector('input[type=file]').files[0];
        var ddd = d3.tsv.parse(file)
        create_table(data);     
    }


    function parseFile()
    {
          d3.text(reader.result, function(text) {
            userArr[userDataId] = d3.tsv.parseRows(text).map(function(row) {
                return row.map(function(value) {
                    return +value;
                });
                return userArr[userDataId];
            });
            
            userDataId++;

        });
    }

    
    function searchGeneURL(studyType, searchGene, datasetVal) 
    {
        //geneData = []; //reset geneData

        //Reset GUI ...
            var infoRow = document.getElementById("immRegExpRow");
            infoRow.style.display = "none";

            var browseRow = document.getElementById("browseRow");
            browseRow.style.display = "none";

            var geneRow = document.getElementById("geneEntryRow");
            geneRow.style.display = "none";

            var privDiv = document.getElementById("privDiv");
            privDiv.style.display = "none";

            var missingDataDiv = document.getElementById("missingDataset");
            missingDataDiv.style.display = "none";

            var multipleGeneNode = document.getElementById("multipleGeneRow");
            multipleGeneNode.style.visibility = "hidden";

            var submitButtonDiv = document.getElementById("submitButton");
            submitButtonDiv.style.textAlign = "left";

            if(document.getElementById("studyEntryRow").style.display == "none")
            {
                var queryRow = document.getElementById("queryRow");
                queryRow.style.marginTop = "60px";
            }
            
            $("#operationEntryRow").hide();

            document.getElementById("geneSearch").innerText = "Modify Query";


        //eQTL Data Visualization Steps...    
        currenti = 0;
        currentj = 49;
        SNP_interest = "";
        datasetvalues = []; //reset datasetvalues

        //addDataSets();
        datasetvalues.push(datasetVal);

        /*var gene = document.getElementById("geneVal").value;
        var lastChar = gene.substr(gene.length - 1);

        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);
        
        var geneArray = gene.split(',');*/
        //study = 2;
        var gene = searchGene;

        $('#geneSetDiv').empty();

        //Remove cis Legend Clone Div
        $('#cisLegendClone').remove();


        //Cis-eQTL
        if(studyType == 1) 
        {
            $('#transRow').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details 
            //$('#plotRow').empty(); 

            var filterRow = document.getElementById("filterRow"); 
            filterRow.style.height = "0px";
            filterRow.style.visibility = "hidden";

            /*var transRow = document.getElementById("transRow");
            transRow.style.height = "30px";
            transRow.style.visibility = "hidden"; */

            var plotRow = document.getElementById("plotRow");
            plotRow.style.height = "0px";
            plotRow.style.visibility = "hidden";

            var manplot = document.getElementById("manPlot");
            manplot.style.height = "0px";

            var filterCol = document.getElementById("filterCol"); 
            filterCol.style.height = "0px";
            filterCol.style.visibility = "hidden";
            
            removeControls();
 
            drawCis(datasetvalues);
            gwasRead = 1;
        }
                
        // Trans-eQTL
        else if (studyType == 2) 
        { 
            $('#igvDiv').empty(); // Remove contents of igvDiv (cis eqtls)
            var igvDiv = document.getElementById("igvDiv"); 
            igvDiv.style.visibility = "hidden";

            //$('#eqtlDataDiv').empty(); // Remove eqtl details 

            var transDiv = document.getElementById("transDiv");
            $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details
            
            if(transDiv === null) 
                createTransDiv();

            if(gene == "")
                gene = "all";
            
            if(operation == 2 || operation == 0)
            {
                var plotRow = document.getElementById("plotRow"); 
                plotRow.style.visibility = "visible";
                
                var plotCol = document.getElementById("plotCol");
                plotCol.style.height = "50px";

                var manplot = document.getElementById("manPlot");
                manplot.style.height = "40px";

                var filterCol = document.getElementById("filterCol"); 
                filterCol.style.height = "45px";
                filterCol.style.visibility = "visible";
            }

            if(operation == 1)
            {

                var filterRow = document.getElementById("filterRow"); 
                    filterRow.style.height = "0px";
                    filterRow.style.visibility = "hidden";

                var filterCol = document.getElementById("filterCol"); 
                    filterCol.style.height = "0px";
                    filterCol.style.visibility = "hidden";

                var plotRow = document.getElementById("plotRow");
                    plotRow.style.height = "0px";
                    plotRow.style.visibility = "hidden";

                $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
                $('#geneDataDiv').remove(); //remove Gene Data
                $('#eqtlDataDiv').remove(); // Remove eqtl details
            }

            var vizRow = document.getElementById("transRow"); 
            vizRow.style.visibility = "visible";

            //Add All Trans-eqtl Results
            addTrans(gene);
           
            //dataCheck(gene, transData);
        }

        //datasetvalues = [];
    }


    function searchGene() 
    {
        geneData = []; //reset geneData
        currenti = 0;
        currentj = 49;
        SNP_interest = "";
        datasetvalues = []; //reset datasetvalues

        addDataSets();

        var gene = document.getElementById("geneVal").value;
        var lastChar = gene.substr(gene.length - 1);

        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);
        
        var geneArray = gene.split(',');

        $('#geneSetDiv').empty();

        //Remove cis Legend Clone Div
        $('#cisLegendClone').remove();


        //Cis-eQTL
        if(study == 1) 
        {
            $('#transRow').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details 
            //$('#plotRow').empty(); 

            var filterRow = document.getElementById("filterRow"); 
            filterRow.style.height = "0px";
            filterRow.style.visibility = "hidden";

            /*var transRow = document.getElementById("transRow");
            transRow.style.height = "30px";
            transRow.style.visibility = "hidden"; */

            var plotRow = document.getElementById("plotRow");
            plotRow.style.height = "0px";
            plotRow.style.visibility = "hidden";

            var manplot = document.getElementById("manPlot");
            manplot.style.height = "0px";

            var filterCol = document.getElementById("filterCol"); 
            filterCol.style.height = "0px";
            filterCol.style.visibility = "hidden";
            
            removeControls();
 
            drawCis(datasetvalues);
            gwasRead = 1;
        }
                
        // Trans-eQTL
        else if (study == 2) 
        { 
            $('#igvDiv').empty(); // Remove contents of igvDiv (cis eqtls)
            var igvDiv = document.getElementById("igvDiv"); 
            igvDiv.style.visibility = "hidden";

            //$('#eqtlDataDiv').empty(); // Remove eqtl details 

            var transDiv = document.getElementById("transDiv");
            $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
            $('#geneDataDiv').remove(); //remove Gene Data
            $('#eqtlDataDiv').remove(); // Remove eqtl details
            
            if(transDiv === null) 
                createTransDiv();

            if(gene == "")
                gene = "all";
            
            if(operation == 2 || operation == 0)
            {
                var plotRow = document.getElementById("plotRow"); 
                plotRow.style.visibility = "visible";
                
                var plotCol = document.getElementById("plotCol");
                plotCol.style.height = "50px";

                var manplot = document.getElementById("manPlot");
                manplot.style.height = "40px";

                var filterCol = document.getElementById("filterCol"); 
                filterCol.style.height = "45px";
                filterCol.style.visibility = "visible";
            }

            if(operation == 1)
            {

                var filterRow = document.getElementById("filterRow"); 
                    filterRow.style.height = "0px";
                    filterRow.style.visibility = "hidden";

                var filterCol = document.getElementById("filterCol"); 
                    filterCol.style.height = "0px";
                    filterCol.style.visibility = "hidden";

                var plotRow = document.getElementById("plotRow");
                    plotRow.style.height = "0px";
                    plotRow.style.visibility = "hidden";

                $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
                $('#geneDataDiv').remove(); //remove Gene Data
                $('#eqtlDataDiv').remove(); // Remove eqtl details
            }

            var vizRow = document.getElementById("transRow"); 
            vizRow.style.visibility = "visible";

            //Add All Trans-eqtl Results
            addTrans(gene);
           
            //dataCheck(gene, transData);
        }

        //datasetvalues = [];
    }



    function createTransDiv() 
    {      
        var vizRow = document.getElementById("transRow");
        var transDivNode = document.createElement('div');

        transDivNode.id = "transDiv";
        transDivNode.className = "col-md-6";
        //transDivNode.style.border = "1px solid #ccc";
        transDivNode.style.padding = "5px";
        transDivNode.style.marginLeft = "10px";
        transDivNode.style.marginTop = "10px";
        //transDivNode.style.paddingBottom = "5px";

        vizRow.appendChild(transDivNode);
    }


    function isInArray(value, array) 
    {
        return array.indexOf(value) > -1;
    }


    function addTrans(gene) 
    {
        var totalTrans = 0;
        transData = [];
        var toggle001 = 0, toggle11 = 0, toggle13 = 0, toggle15 = 0, toggle16 = 0, toggle17 = 0, toggle18 = 0, toggle19 = 0, toggle24 = 0, toggle25 = 0, toggle26 = 0, toggle34 = 0, toggle41 = 0, toggle51 = 0, toggle61 = 0, toggle62 = 0, toggle63 = 0, toggle81 = 0, toggle111 = 0, toggle112 = 0, toggle113 = 0, toggle211 = 0, toggle212 = 0, toggle213 = 0, toggle221 = 0, toggle224 = 0, toggle225 = 0, toggle302 = 0, toggle401 = 0, toggle402 = 0, toggle403 = 0, toggle791 = 0, toggle396 = 0, toggle94 = 0, toggle99 = 0, toggle131 = 0, toggle133 = 0, toggle134 = 0, toggle135 = 0;
        
        var geneArray = gene.trim().split(/\s*,\s*/);
        var geneSetData = [];
        
        var file11 = [];
        //var file15 = [];

        var file001 = "NA", file13 ="NA", file15 ="NA", file16 ="NA", file17 ="NA", file18 ="NA", file19 ="NA", file24 ="NA", file25 ="NA", file26 ="NA", file34 ="NA", file41 ="NA", file51 ="NA", file61 ="NA", file62 ="NA", file63 ="NA", file81 ="NA", file111 ="NA", file112 ="NA", file113 ="NA", file211 ="NA", file212 ="NA", file213 ="NA", file221 ="NA", file224 ="NA", file225 ="NA", file302 ="NA", file401 ="NA", file402 ="NA", file403 ="NA", file791 ="NA", file396 ="NA", file94 ="NA",file99 ="NA", file131 ="NA", file133 ="NA", file134 ="NA", file135 ="NA";

        for (var i = 0; i < 30; i++)
        {
            file11[i] = "NA"; //Framingham
            //file15[i] = "NA"; //Pierce
        }
 
        var getCsv = function (csvUrl) {
            var defer = $.Deferred();
            d3.tsv(csvUrl, function (error, rows) {
                //if (error) {
                //    defer.reject(error);
               // }
                defer.resolve(rows);
            });
            return defer.promise();
        };


        var missingDataDiv = document.getElementById("missingDataset");
            missingDataDiv.style.display = "block";
            missingDataDiv.style.visibility = "visible";
            var missingDataText = document.getElementById("missingDataText");
            missingDataText.innerHTML = "Loading, please wait...";


        for (var i = 0; i < datasetvalues.length; i++) 
        {
            if(datasetvalues[i] == 11)
            {
                toggle11 = 1;
                
                for(var j = 0; j < geneArray.length; j++)
                    file11[j] = "./data/fram_trans/" + geneArray[j];
            }
                
            else if(datasetvalues[i] == 001)
            {              
                toggle001 = 1;
                file001 = "./data/2020/Lung-trans.txt";
            }

            else if(datasetvalues[i] == 13)
            {              
                toggle13 = 1;
                file13 = "./data/westra-trans.txt";
            }

            else if(datasetvalues[i] == 15)
            {
                toggle15 = 1;

                file15 = "./data/pierce/pierce-trans-2020.txt";
                //for(var j = 0; j < geneArray.length; j++)
                  //  file15[j] = "./data/pierce/trans/" + geneArray[j];
            }

            else if(datasetvalues[i] == 16)
            {
                toggle16 = 1;
                file16 = "./data/yamada/yamada-trans.txt";
            }

            else if(datasetvalues[i] == 17)
            {
                toggle17 = 1;
                file17 = "./data/eqtlgen/eqtlgen-trans.txt";
            }

            else if(datasetvalues[i] == 18)
            {
                toggle18 = 1;
                file18 = "./data/2020/cage-trans.txt";
            }

            else if(datasetvalues[i] == 19)
            {
                toggle19 = 1;
                file19 = "./data/2020/life-trans.txt";
            }
                
            else if(datasetvalues[i] == 24)
            {
                toggle24 = 1;
                file24 = "./data/jager/monocytes-trans.txt";

            }
                
            else if(datasetvalues[i] == 25)
            {
                toggle25 = 1;
                file25 = "./data/fairfax/mono-trans.txt";
            }
                
            else if(datasetvalues[i] == 26)
            {
                toggle26 = 1;
                file26 = "./data/fairfax/mono-trans-naive.txt";
            }
                
            else if(datasetvalues[i] == 34)
            {
                toggle34 = 1;
                file34 = "./data/jager/cd4T-trans.txt";
            }
                
            else if(datasetvalues[i] == 41)
            {
                toggle41 = 1;
                file41 = "./data/fairfax/bcell-trans.txt";
            }
                
            else if(datasetvalues[i] == 51)
            {
                toggle51 = 1;
                file51 = "./data/fairfax/neutrophils-trans-final.txt";
            }
                
            else if(datasetvalues[i] == 61)
            {
                toggle61 = 1;
                file61 = "./data/fairfax/mono-trans-lps2.txt";
            }
                
            else if(datasetvalues[i] == 62)
            {
                toggle62 = 1;
                file62 = "./data/fairfax/mono-trans-lps24.txt";
            }
                
            else if(datasetvalues[i] == 63)
            {
                toggle63 = 1;
                file63 = "./data/fairfax/mono-trans-ifn.txt";
            }
                
            else if(datasetvalues[i] == 81)
            {
                toggle81 = 1;
                file81 = "./data/heijmans/trans-meqtl.txt";
            }

            else if(datasetvalues[i] == 111)
            {
                toggle111 = 1;
                file111 = userTransFileLink[0];
            }

            else if(datasetvalues[i] == 112)
            {
                toggle112 = 1;
                file112 = userTransFileLink[1];
            }

            else if(datasetvalues[i] == 113)
            {
                toggle113 = 1;
                file113 = userTransFileLink[2];
            }

            else if(datasetvalues[i] == 211)
            {
                toggle211 = 1;
                file211 = "./data/hipc/mayo/mayo-day0-trans.txt";
            }

            else if(datasetvalues[i] == 212)
            {
                toggle212 = 1;
                file212 = "./data/hipc/mayo/mayo-day3-trans.txt";
            }
                
            else if(datasetvalues[i] == 213)
            {
                toggle213 = 1;
                file213 = "./data/hipc/mayo/mayo-day28-trans.txt";
            }

            else if(datasetvalues[i] == 221)
            {
                toggle211 = 1;
                file211 = "./data/hipc/nih/nih-day-7-trans.txt";
            }

            else if(datasetvalues[i] == 224)
            {
                toggle212 = 1;
                file212 = "./data/hipc/nih/nih-day7-trans.txt";
            }
                
            else if(datasetvalues[i] == 225)
            {
                toggle213 = 1;
                file213 = "./data/hipc/nih/nih-day70-trans.txt";
            }

            else if(datasetvalues[i] == 302)
            {
                toggle302 = 1;
                file302 = "./data/hipc/chikv/chik-day9-trans.txt";
            }

            else if(datasetvalues[i] == 401)
            {
                toggle401 = 1;
                file401 = "./data/hipc/zikv/zika-acute-trans.txt";
            }

            else if(datasetvalues[i] == 402)
            {
                toggle402 = 1;
                file402 = "./data/hipc/zikv/zika-lateacute-trans.txt";
            }

            else if(datasetvalues[i] == 403)
            {
                toggle403 = 1;
                file403 = "./data/hipc/zikv/zika-conv-trans.txt";
            }

            else if(datasetvalues[i] == 791)
            {
                toggle791 = 1;
                file791 = "./data/2020/alspac-trans.txt";
            }

            else if(datasetvalues[i] == 396)
            {
                toggle396 = 1;
                file396 = "./data/2021/estonia-cd4-trans.txt";
            }

            else if(datasetvalues[i] == 94)
            {
                toggle94 = 1;
                file94 = "./data/2021/estonia-cd8-trans.txt";
            }

            else if(datasetvalues[i] == 99)
            {
                toggle99 = 1;
                file99 = "./data/2021/pat-baseline-trans.txt";
            }

            else if(datasetvalues[i] == 131)
            {
                toggle131 = 1;
                file131 = "./data/2021/pat-lps-trans.txt";
            }

            else if(datasetvalues[i] == 133)
            {
                toggle133 = 1;
                file133 = "./data/2021/pat-flu-trans.txt";
            }

            else if(datasetvalues[i] == 134)
            {
                toggle134 = 1;
                file134 = "./data/2021/pat-re-flu-trans.txt";
            }

            else if(datasetvalues[i] == 135)
            {
                toggle135 = 1;
                file135 = "./data/2021/pat-ifnb-trans.txt";
            }
        }
        
        //No Framingham
        if(toggle11 == 0)
        {
            $.when( getCsv(file001), getCsv(file13), getCsv(file15), getCsv(file16), getCsv(file17), getCsv(file18), getCsv(file19), getCsv(file24), getCsv(file25), getCsv(file26), getCsv(file34), getCsv(file41), getCsv(file51), getCsv(file61), getCsv(file62), getCsv(file63), getCsv(file81), getCsv(file111), getCsv(file112), getCsv(file113), getCsv(file211), getCsv(file212), getCsv(file213), getCsv(file221), getCsv(file224), getCsv(file225), getCsv(file302), getCsv(file401), getCsv(file402), getCsv(file403), getCsv(file791), getCsv(file396), getCsv(file94), getCsv(file99), getCsv(file131), getCsv(file133), getCsv(file134), getCsv(file135)
                    ).done(function (res001, res13, res15, res16, res17, res18, res19, res24, res25, res26, res34, res41, res51, res61, res62, res63, res81, res111, res112, res113, res211, res212, res213, res221, res224, res225, res302, res401, res402, res403, res791, res396, res94, res99, res131, res133, res134, res135) { 
                     
                        if(toggle001 == 1)
                            transData.push(res001);

                        if(toggle13 == 1)
                            transData.push(res13); 

                        if(toggle15 == 1)
                            transData.push(res15);                

                        if(toggle16 == 1)
                            transData.push(res16);

                        if(toggle17 == 1)
                            transData.push(res17);

                        if(toggle18 == 1)
                            transData.push(res18);

                        if(toggle19 == 1)
                            transData.push(res19);

                        if(toggle24 == 1)
                            transData.push(res24);

                        if(toggle25 == 1)
                            transData.push(res25);

                        if(toggle26 == 1)
                            transData.push(res26);

                        if(toggle34 == 1)
                            transData.push(res34);

                        if(toggle41 == 1)
                            transData.push(res41);

                        if(toggle51 == 1)
                            transData.push(res51);

                        if(toggle61 == 1)
                            transData.push(res61);

                        if(toggle62 == 1)
                            transData.push(res62);

                        if(toggle63 == 1)
                            transData.push(res63);

                        if(toggle81 == 1)
                            transData.push(res81);

                        if(toggle111 == 1)
                            transData.push(res111);

                        if(toggle112 == 1)
                            transData.push(res112);

                        if(toggle113 == 1)
                            transData.push(res113);

                        if(toggle211 == 1)
                            transData.push(res211);

                        if(toggle212 == 1)
                            transData.push(res212);

                        if(toggle213 == 1)
                            transData.push(res213);

                        if(toggle221 == 1)
                            transData.push(res221);

                        if(toggle224 == 1)
                            transData.push(res224);

                        if(toggle225 == 1)
                            transData.push(res225);

                        if(toggle302 == 1)
                            transData.push(res302);

                        if(toggle401 == 1)
                            transData.push(res401);

                        if(toggle402 == 1)
                            transData.push(res402);

                        if(toggle403 == 1)
                            transData.push(res403);

                        if(toggle791 == 1)
                            transData.push(res791);

                        if(toggle396 == 1)
                            transData.push(res396);

                        if(toggle94 == 1)
                            transData.push(res94);

                        if(toggle99 == 1)
                            transData.push(res99);

                        if(toggle131 == 1)
                            transData.push(res131);

                        if(toggle133 == 1)
                            transData.push(res133);

                        if(toggle134 == 1)
                            transData.push(res134);

                        if(toggle135 == 1)
                            transData.push(res135);
                        
                        if(gwasRead == 0)
                        {
                            $.when(
                                getCsv("./data/gwas.txt") 
                            ).done(function (res1) {
                                gwasData.push(res1);
                                gwasRead = 1;

                                if(geneArray.length > 1)
                                {
                                    //Union operation
                                    if(operation == 2)                                                  
                                        dataCheck(transData, gene, geneArray);
                                 
                                    // Intersection Operation
                                    else 
                                    {
                                        for(var i = 0; i < transData.length; i++) {
                                            for(var j = 0; j < transData[i].length; j++) {
                                                if(geneSnpToggle == 1) {
                                                    if( isInArray(transData[i][j].SNPName, geneArray) )
                                                        geneSetData.push(transData[i][j]);
                                                }
                                                
                                                else if(geneSnpToggle == 0) {
                                                    if( isInArray(transData[i][j].Gene, geneArray) )
                                                        geneSetData.push(transData[i][j]);  
                                                }
                                            }
                                        }
                                        
                                        calcCommonTrans(geneSetData);
                                    }
                                }

                                else
                                    dataCheck(transData, gene, geneArray);
                            
                            }).fail(function (err) {
                                console.log(err);
                            });

                        }
                        
                        else
                        {
                            if(geneArray.length > 1)
                            {
                                //Union operation
                                if(operation == 2)                                                  
                                    dataCheck(transData, gene, geneArray);
                             
                                // Intersection Operation
                                else 
                                {
                                    for(var i = 0; i < transData.length; i++) {
                                        for(var j = 0; j < transData[i].length; j++) {
                                            if(geneSnpToggle == 1) {
                                                if( isInArray(transData[i][j].SNPName, geneArray) )
                                                    geneSetData.push(transData[i][j]);
                                            }
                                            
                                            else if(geneSnpToggle == 0) {
                                                if( isInArray(transData[i][j].Gene, geneArray) )
                                                    geneSetData.push(transData[i][j]);  
                                            }
                                        }
                                    }  
                                    
                                    calcCommonTrans(geneSetData);
                                }
                            }

                            else
                                dataCheck(transData, gene, geneArray);
                        }
                    });
            }


            // YES Framingham
            else if (toggle11 == 1)
            {
                $.when(
                getCsv(file11[0]), getCsv(file11[1]), getCsv(file11[2]), getCsv(file11[3]), getCsv(file11[4]), getCsv(file11[5]), getCsv(file11[6]), getCsv(file11[7]), getCsv(file11[8]), getCsv(file11[9]), getCsv(file11[10]), getCsv(file11[11]), getCsv(file11[12]), getCsv(file11[13]), getCsv(file11[14]), getCsv(file11[15]), getCsv(file11[16]), getCsv(file11[17]), getCsv(file11[18]), getCsv(file11[19]), getCsv(file11[20]), getCsv(file11[21]), getCsv(file11[22]), getCsv(file11[23]), getCsv(file11[24]), getCsv(file11[25]), getCsv(file11[26]), getCsv(file11[27]), getCsv(file11[28]), getCsv(file11[29]), 
                    
                getCsv(file001), getCsv(file13), getCsv(file15),

                    getCsv(file16), getCsv(file17), getCsv(file18), getCsv(file19), getCsv(file24), getCsv(file25), getCsv(file26), getCsv(file34), getCsv(file41), getCsv(file51), getCsv(file61), getCsv(file62), getCsv(file63), getCsv(file81), getCsv(file111), getCsv(file112), getCsv(file113), getCsv(file211), getCsv(file212), getCsv(file213), getCsv(file221), getCsv(file224), getCsv(file225), getCsv(file302), getCsv(file401), getCsv(file402), getCsv(file403), getCsv(file791), getCsv(file396), getCsv(file94), getCsv(file99), getCsv(file131), getCsv(file133), getCsv(file134), getCsv(file135)
                
                ).done(function (res11_1, res11_2, res11_3, res11_4, res11_5, res11_6, res11_7, res11_8, res11_9, res11_10, res11_11, res11_12, res11_13, res11_14, res11_15, res11_16, res11_17, res11_18, res11_19, res11_20, res11_21, res11_22, res11_23, res11_24, res11_25, res11_26, res11_27, res11_28, res11_29, res11_30, res001, res13, res15, res16, res17, res18, res19, res24, res25, res26, res34, res41, res51, res61, res62, res63, res81, res111, res112, res113, res211, res212, res213, res221, res224, res225, res302, res401, res402, res403, res791, res396, res94, res99, res131, res133, res134, res135) { 

                    if(toggle11 == 1)
                    {
                        if(res11_1 != undefined){
                            transData.push(res11_1);
                        }

                        if(res11_2 != undefined)
                            transData.push(res11_2);

                        if(res11_3 != undefined){
                            transData.push(res11_3);
                        }

                        if(res11_4 != undefined)
                            transData.push(res11_4);

                        if(res11_5 != undefined){
                            transData.push(res11_5);
                        }

                        if(res11_6 != undefined)
                            transData.push(res11_6);

                        if(res11_7 != undefined){
                            transData.push(res11_7);
                        }

                        if(res11_8 != undefined)
                            transData.push(res11_8);

                        if(res11_9 != undefined){
                            transData.push(res11_9);
                        }

                        if(res11_10 != undefined)
                            transData.push(res11_10);

                        if(res11_11 != undefined){
                            transData.push(res11_11);
                        }

                        if(res11_12 != undefined)
                            transData.push(res11_12);

                        if(res11_13 != undefined){
                            transData.push(res11_13);
                        }

                        if(res11_14 != undefined)
                            transData.push(res11_14);

                        if(res11_15 != undefined){
                            transData.push(res11_15);
                        }

                        if(res11_16 != undefined)
                            transData.push(res11_16);

                        if(res11_17 != undefined){
                            transData.push(res11_17);
                        }

                        if(res11_18 != undefined)
                            transData.push(res11_18);

                        if(res11_19 != undefined){
                            transData.push(res11_19);
                        }

                        if(res11_20 != undefined)
                            transData.push(res11_20);

                        if(res11_21 != undefined){
                            transData.push(res11_21);
                        }

                        if(res11_22 != undefined)
                            transData.push(res11_22);

                        if(res11_23 != undefined){
                            transData.push(res11_23);
                        }

                        if(res11_24 != undefined)
                            transData.push(res11_24);

                        if(res11_25 != undefined){
                            transData.push(res11_25);
                        }

                        if(res11_26 != undefined)
                            transData.push(res11_26);

                        if(res11_27 != undefined){
                            transData.push(res11_27);
                        }

                        if(res11_28 != undefined)
                            transData.push(res11_28);

                        if(res11_29 != undefined){
                            transData.push(res11_29);
                        }

                        if(res11_30 != undefined)
                            transData.push(res11_30);
                    }

                    if(toggle001 == 1)
                        transData.push(res001);

                    if(toggle13 == 1)
                        transData.push(res13);

                    if(toggle15 == 1)
                        transData.push(res15);

                    if(toggle16 == 1)
                        transData.push(res16);

                    if(toggle17 == 1)
                        transData.push(res17);

                    if(toggle18 == 1)
                        transData.push(res18);

                    if(toggle19 == 1)
                        transData.push(res19);

                    if(toggle24 == 1)
                        transData.push(res24);

                    if(toggle25 == 1)
                        transData.push(res25);

                    if(toggle26 == 1)
                        transData.push(res26);

                    if(toggle34 == 1)
                        transData.push(res34);

                    if(toggle41 == 1)
                        transData.push(res41);

                    if(toggle51 == 1)
                        transData.push(res51);

                    if(toggle61 == 1)
                        transData.push(res61);

                    if(toggle62 == 1)
                        transData.push(res62);

                    if(toggle63 == 1)
                        transData.push(res63);

                    if(toggle81 == 1)
                        transData.push(res81);

                    if(toggle111 == 1)
                        transData.push(res111);

                    if(toggle112 == 1)
                        transData.push(res112);

                    if(toggle113 == 1)
                        transData.push(res113);

                    if(toggle211 == 1)
                        transData.push(res211);

                    if(toggle212 == 1)
                        transData.push(res212);

                    if(toggle213 == 1)
                        transData.push(res213);

                    if(toggle221 == 1)
                        transData.push(res221);

                    if(toggle224 == 1)
                        transData.push(res224);

                    if(toggle225 == 1)
                        transData.push(res225);

                    if(toggle302 == 1)
                        transData.push(res302);

                    if(toggle401 == 1)
                        transData.push(res401);

                    if(toggle402 == 1)
                        transData.push(res402);

                    if(toggle403 == 1)
                        transData.push(res403);

                    if(toggle791 == 1)
                        transData.push(res791);

                    if(toggle396 == 1)
                        transData.push(res396);

                    if(toggle94 == 1)
                        transData.push(res94);

                    if(toggle99 == 1)
                        transData.push(res99);
                    
                    if(toggle131 == 1)
                        transData.push(res131);

                    if(toggle133 == 1)
                        transData.push(res133);

                    if(toggle134 == 1)
                        transData.push(res134);

                    if(toggle135 == 1)
                        transData.push(res135);

                    if(gwasRead == 0)
                    {
                        $.when(
                            getCsv("./data/gwas.txt") 
                        ).done(function (res1) {
                            gwasData.push(res1);
                            gwasRead = 1;

                            if(geneArray.length > 1)
                            {
                                //Union operation
                                if(operation == 2)                                                  
                                    dataCheck(transData, gene, geneArray);
                             
                                // Intersection Operation
                                else 
                                {
                                    for(var i = 0; i < transData.length; i++) {
                                        for(var j = 0; j < transData[i].length; j++) {
                                            if(geneSnpToggle == 1) {
                                                if( isInArray(transData[i][j].SNPName, geneArray) )
                                                    geneSetData.push(transData[i][j]);
                                            }
                                            
                                            else if(geneSnpToggle == 0) {
                                                if( isInArray(transData[i][j].Gene, geneArray) )
                                                    geneSetData.push(transData[i][j]);  
                                            }
                                        }
                                    }  
                                    
                                    calcCommonTrans(geneSetData);
                                }
                            }

                            else
                                dataCheck(transData, gene, geneArray);
                        
                        }).fail(function (err) {
                            console.log(err);
                        });

                    }
                    
                    else
                    {
                        if(geneArray.length > 1)
                        {
                            //Union operation
                            if(operation == 2)                                                  
                                dataCheck(transData, gene, geneArray);
                         
                            // Intersection Operation
                            else 
                            {
                                for(var i = 0; i < transData.length; i++) {
                                    for(var j = 0; j < transData[i].length; j++) {
                                        if(geneSnpToggle == 1) {
                                            if( isInArray(transData[i][j].SNPName, geneArray) )
                                                geneSetData.push(transData[i][j]);
                                        }
                                        
                                        else if(geneSnpToggle == 0) {
                                            if( isInArray(transData[i][j].Gene, geneArray) )
                                                geneSetData.push(transData[i][j]);  
                                        }
                                    }
                                }  
                                
                                calcCommonTrans(geneSetData);
                            }
                        }

                        else
                            dataCheck(transData, gene, geneArray);
                    }
                });
            }

    }


    function addG() 
    {
        var btn = document.getElementById("wBtn");
        btn.className = "btn btn-primary active btn-block"; 
        btn.style.background = "lightgray";

        btn = document.getElementById("gBtn");

        btn.className = "btn btn-primary disabled btn-block"; 
        btn.style.background = "gray";

        d3.select("svg").remove();
    }
    

    function dataCheck(data, gene, geneArray) 
    {
        //var sliced = data.slice(i,j);
        geneData = [];

        //IF no gene is specified, show all data
        if(gene == "all") {

            for(var i = 0; i < data.length; i++)
                for(var j = 0; j < data[i].length; j++)
                    geneData.push(data[i][j]);

            geneData.sort(function(a, b){
                return a.PValue-b.PValue;
            });

            dataSlicer(geneData, 0, 49);
            currenti = 0;
            currentj = 49;
            updateRange(currenti, currentj);
        }

        else {
            
            for(var i = 0; i < data.length; i++) {
                for(var j = 0; j < data[i].length; j++) {
                    if(geneSnpToggle == 1) {
                        if( isInArray(data[i][j].SNPName, geneArray) )
                            geneData.push(data[i][j]);
                    }
                    
                    else if(geneSnpToggle == 0) {
                        if( isInArray(data[i][j].Gene, geneArray) )
                            geneData.push(data[i][j]);
                    }
                }
            }

                    //if(data[i][j].Gene == gene)
                        //geneData.push(data[i][j]);

            geneData.sort(function(a, b){
                return a.PValue-b.PValue;
            });

            var dataSize = geneData.length;
            
            var vizDiv = document.getElementById("transDiv");

            if(dataSize == 0) 
            {              
                vizDiv.style.fontSize = "16px";
                vizDiv.style.fontWeight = "bold";
                vizDiv.style.textAlign = "center";
                vizDiv.style.border = "1px solid black";
                vizDiv.innerHTML = "NO RESULTS FOUND...";

                var missingDataDiv = document.getElementById("missingDataset");
                missingDataDiv.style.visibility = "hidden";
                missingDataDiv.style.display = "none";

                removeControls();
                removefilterControl();
                $('#geneDataDiv').empty(); //remove Gene Data
                
                var plotRow = document.getElementById("plotRow");
                plotRow.style.height = "0px";
                plotRow.style.visibility = "hidden";
            }

            else 
            {
                $('#transDiv').empty();
                //vizDiv.innerHTML = "";
                vizDiv.style.border = "";
                vizDiv.style.textAlign = "left";
                
                removeControls();
                           
                addGeneData(gene, geneData, dataSize);
                addfilterControl();
                drawTrans(geneData, geneData[0].PValue, dataSize);
                
                if(dataSize > 50) {
                    addControls();
                    //updateRange(0, 49);
                    addTransLegends();
                    //addSaveButtons(true);
                }
                
                else
                    addTransLegends();
                    //addSaveButtons(false);
            }
         
        }
    }


    function dataSlicer(data, i, j) 
    {
        var sliced = data.slice(i,j);
        
        $('#transDiv').empty();
        var vizDiv = document.getElementById("transDiv");
        vizDiv.style.border = "";
        
        addGeneData("all", geneData, data.length);
        addfilterControl();

        drawTrans(sliced, data[0].PValue, sliced.length);  
        addControls();
        addTransLegends();
        //addSaveButtons(true);
    }


    function addControls() 
    {
        var controlsRow = document.getElementById("controlsRow"); 
        var vizDiv = document.getElementById("transDiv");
        
        var ControlClone = controlsRow.cloneNode(true);

        vizDiv.appendChild(ControlClone);
        ControlClone.style.visibility = "visible";
    }

    
    function addSaveButtons(addGap) 
    {
        var saveButtonsRow = document.getElementById("saveButtonsRow"); 
        var vizDiv = document.getElementById("transDiv");
        
        var saveButtonsClone = saveButtonsRow.cloneNode(true);

        vizDiv.appendChild(saveButtonsClone);

        saveButtonsClone.style.visibility = "visible";
    }


    function addLegends() 
    {
        if(geneSnpToggle == 1)
            var legendRow = document.getElementById("cisLegendRowSnp");
        
        else
            var legendRow = document.getElementById("cisLegendRow"); 
        
        var containerDiv = document.getElementById("igvDiv");
        
        var legendClone = legendRow.cloneNode(true);
        legendClone.id = "cisLegendClone";

        containerDiv.parentNode.insertBefore(legendClone, containerDiv.nextSibling);
        //containerDiv.appendChild(legendClone);
        legendClone.style.visibility = "visible";
    }


    function addTransLegends() 
    {
        var legendRow = document.getElementById("transLegendRow"); 
        var transDiv = document.getElementById("transDiv");
        
        var legendClone = legendRow.cloneNode(true);
        legendClone.id = "transLegendClone";

        transDiv.appendChild(legendClone);
        //containerDiv.parentNode.insertBefore(legendClone, containerDiv.nextSibling);
        //containerDiv.appendChild(legendClone);
        legendClone.style.visibility = "visible";
    }


    function removeControls() 
    {
        var controlsRow = document.getElementById("controlsRow"); 
        controlsRow.style.visibility = "hidden"; 
    }

    
    function addfilterControl() 
    {
        var filterRow = document.getElementById("filterRow"); 
        filterRow.style.visibility = "visible";

        var filterCol = document.getElementById("filterCol"); 
        filterCol.style.visibility = "visible";
        filterCol.style.height = "45x";
    }

    
    function removefilterControl() 
    {
        var filterRow = document.getElementById("filterRow"); 
        filterRow.style.visibility = "hidden";

        var filterCol = document.getElementById("filterCol"); 
        filterCol.style.visibility = "hidden";
        filterCol.style.height = "0px";
    }

    
    function updateRange(currenti, currentj) 
    {
        var RangeNode = document.getElementById("displayRange");
        var content = "" + (currenti+1) + " - " + (currentj+1);
        RangeNode.innerHTML = content;
    }

    
    function first() 
    {  
        currenti = 0;
        currentj = 49;

        dataSlicer(geneData, currenti, currentj);
        updateRange(currenti, currentj);
    }

    
    function last() 
    {
        var len = geneData.length;

        currenti = (parseInt(len / 50) * 50);
        currentj = currenti + 49;
        
        dataSlicer(geneData, currenti, currentj);
        updateRange(currenti, currentj);
    }

    
    function next() 
    {
        currenti = currentj + 1;
        currentj = currentj + 50;
        
        dataSlicer(geneData, currenti, currentj);
        updateRange(currenti, currentj);
    }

    
    function prev() 
    {
        if (currenti == 0)
            return;
        currenti = currenti - 50;
        currentj = currentj - 50;
        
        dataSlicer(geneData, currenti, currentj);
        updateRange(currenti, currentj);
    }


    function saveText()
    {
        var content = "";

        content += "P-Value" + "\t";
        content += "SNP-Name" + "\t";
        content += "SNP-Chr" + "\t";
        content += "SNP-Chr-Pos" + "\t";
        content += "GENE" + "\t";
        content += "Gene-Chr" + "\t";
        content += "Gene-Chr-Pos" + "\n";

        for(var i = 0; i < geneData.length; i++)
        {
            content += geneData[i].PValue + "\t";
            content += geneData[i].SNPName + "\t";
            content += geneData[i].SNPChr + "\t";
            content += geneData[i].SNPChrPos + "\t";
            content += geneData[i].Gene + "\t";
            content += geneData[i].ProbeChr + "\t";
            content += geneData[i].ProbeCenterChrPos + "\t";
            content += "\n";
        }
        
        uri = "data:application/octet-stream," + encodeURIComponent(content);
        location.href = uri;
    }


    function saveSVG() 
    {
        var tmp = document.getElementById("transDiv");
        var svg = tmp.getElementsByTagName("svg")[0];

        var svgData = svg.outerHTML;
        var svgBlob = new Blob([svgData], {type:"image/svg+xml;charset=utf-8"});
        var svgUrl = URL.createObjectURL(svgBlob);
        var downloadLink = document.createElement("a");
        downloadLink.href = svgUrl;
        downloadLink.download = "transEQTLs.svg";
        document.body.appendChild(downloadLink);
        downloadLink.click();
        document.body.removeChild(downloadLink);
    }


    function savePNG()
    {
        var svg = d3.select("svg");
        var w = parseInt(svg.style("width"), 10);
        var h = parseInt(svg.style("height"), 10);

        var svgString = getSVGString(svg.node());
        svgString2Image( svgString, 2 * w, 2 * h, 'png', save ); // passes Blob and filesize String to the callback

        function save( dataBlob, filesize ){
            saveAs( dataBlob, 'transEQTLs.png' ); // FileSaver.js function
        }
        
    }


    function savePDF()
    {
       // svg_to_pdf(document.querySelector("svg"), function (pdf) {
       //         download_pdf('SVG.pdf', pdf.output('dataurlstring'));
       //     });

        const svgElement = document.getElementById('trans-svg');
        const width = 800, height = 600;

        // create a new jsPDF instance
        const pdf = new jsPDF2('l', 'pt', [width, height]);

        // render the svg element
        svg2pdf(svgElement, pdf, {
            xOffset: 25,
            yOffset: 20,
            scale: 1
        });

        // get the data URI
        const uri = pdf.output('datauristring');

        // or simply safe the created pdf
        pdf.save('transEQTLs.pdf');
    }


    function saveCisPNG() 
    {

        html2canvas($("#igvDiv"), {
            onrendered: function(canvas) {
                theCanvas = canvas;


                canvas.toBlob(function(blob) {
                    saveAs(blob, "cisEQTLs.png"); 
                });
            }
        });

    }


    function saveCisPDF() 
    {
        html2canvas($("#igvDiv"), {
            onrendered: function(canvas) {
                $('#container').html("");
                $('#container').append(canvas);

                var imgData = canvas.toDataURL("image/png", 1.0);
                var pdf = new jsPDF();

                pdf.addImage(imgData, 0, 0, 200, 200, 0, 0);

                pdf.save('cisEQTLs.pdf');
            }
        });
    }
    


    function addGeneData(gene, geneData, dataSize) 
    {
        var geneNode = document.getElementById("geneDataDiv");
        var geneInnerNode = document.getElementById("geneDataInnerDiv");

        if(geneNode === null) {
            geneNode = document.createElement('div'); 
            
            geneNode.id = "geneDataDiv";
            geneNode.className = "row";
            geneNode.style.marginTop = "20px";
            geneNode.style.marginLeft = "-5px";
            geneNode.style.fontSize = "14px";
            geneNode.style.fontWeight = "bold";

            $("#queryRow").after(geneNode);
        }

        if(geneInnerNode === null) {
            geneInnerNode = document.createElement('div');

            geneInnerNode.id = "geneDataInnerDiv";
            geneInnerNode.className = "col-md-6";
            geneInnerNode.style.border = "1px solid #ccc";
            geneInnerNode.style.paddingLeft = "15px";
            geneInnerNode.style.paddingRight = "15px";
            geneInnerNode.style.paddingTop = "5px";
            geneInnerNode.style.paddingBottom = "5px";

            geneNode.appendChild(geneInnerNode);
        }

        geneInnerNode.innerHTML = " "
        
        if(gene == "all")
            htmlcontent = "Gene(s): ALL";

        //Append Data Size
        var d = document.createElement('span');
        d.className = "badge";
        d.style.background = "steelblue";
        d.style.color = "white";
        var dtext = dataSize + " trans-eQTLs";
        var dnode = document.createTextNode(dtext);
        //d.style.marginLeft = "10px";
        d.style.fontSize = "16px";
        d.style.fontWeight = "normal";
        d.appendChild(dnode);
        geneInnerNode.appendChild(d);

        //Append Gene List
        var p = document.createElement('span');
        var ptext = document.createTextNode("Selected Genes:");
        p.style.marginLeft = "35px";
        p.style.fontSize = "14px";
        p.style.verticalAlign = "middle";
        p.appendChild(ptext);
        geneInnerNode.appendChild(p);

        var genedataselect = document.getElementById("geneDataSelect");
        var genedataselectClone = genedataselect.cloneNode(true);

        //add each Gene 
        var geneArray = gene.trim().split(/\s*,\s*/);

        for(var i = 0; i < geneArray.length; i++)
        {
            var opt1 = document.createElement("option");
            
            //Calculate the count value
            var dcount = 0;
            for(var j = 0; j < geneData.length; j++)
            {
                if(geneData[j].Gene === geneArray[i])
                    dcount++
            }

            opt1.className = "filterGene";
            opt1.text = geneArray[i] + " (" + dcount + ")"; 
            opt1.id = "geneD" + i;
            opt1.selected = true;
            genedataselectClone.add(opt1);
        } 
        
        genedataselectClone.style.visibility = "visible";
        geneInnerNode.appendChild(genedataselectClone);
        $('#geneDataSelect').multiselect({
                enableFiltering: true,
                includeSelectAllOption: true,
                buttonText: function(options, select) {
                    return 'Filter Gene List';
                },
                maxHeight: 300,
                buttonWidth: '240px'
        }); 

        //Filter trans-eQTLs based on filter selection 
        var filtercheckboxes = document.querySelectorAll("label.checkbox > input");
        
        for(var i = 0; i < filtercheckboxes.length; i++) {

            filtercheckboxes[i].onclick = function() {

                var allcircles = document.querySelectorAll('circle.circle');
                
                if(this.checked === false)
                {
                    for (var j = 0; j < allcircles.length; j++)
                    {
                        if(allcircles[j].__data__.Gene === this.value.split(' ')[0] || this.value === "multiselect-all")
                            allcircles[j].style.fillOpacity = 0.1;
                    }
                } 

                else if(this.checked === true)
                {
                    for (var j = 0; j < allcircles.length; j++)
                    {
                        if(allcircles[j].__data__.Gene === this.value.split(' ')[0] || this.value === "multiselect-all")
                            allcircles[j].style.fillOpacity = 0.7;
                    }
                }
            }
          
        }        
    }


    function filter() 
    {
        var chr = document.getElementById("filterChr").value;
        var start = document.getElementById("filterStart").value;
        var end = document.getElementById("filterEnd").value;

        var endToggle = 0;

        if (end != "End")
            endToggle = 1;
        
        if (end == "")
            endToggle = 0;

        start = parseInt(start, 10);
        end = parseInt(end, 10);
        
        highlightC = chr;

        var selection = svg.selectAll("circle");

        selection.attr('r', function (d) {
            if(endToggle == 1)
                if (d.SNPChr == chr && d.SNPChrPos >= start && d.SNPChrPos <= end)
                    return 5;
                else
                    return 0;

            else if(endToggle == 0) {
                if ( ("c" + d.SNPChr) == chr && start <= d.SNPChrPos){
                    return 5;
                }
                else {
                    return 0;
                }
            }

        });

        var btn = document.getElementById("filter");
        btn.className = "btn btn-primary disabled";

        var btn = document.getElementById("unfilter");
        btn.className = "btn btn-primary active";
    }


    function unfilter() 
    {
        highlightC = 0;
        var selection = svg.selectAll("circle");

        selection.attr('r', '5');
        
        var btn = document.getElementById("filter");
        btn.className = "btn btn-primary active";

        var btn = document.getElementById("unfilter");
        btn.className = "btn btn-primary disabled";   
    }


    function openFilter() 
    {
        var filterDiv = document.getElementsByClassName('filter');
        var i = filterDiv.length;

        //var parent = document.getElementById("filterRow");
        var parent = document.getElementById("filterPl");
        var child = document.getElementById("openF");

        if(filterToggle == 0) {
            
            while(i--) {
                filterDiv[i].style.visibility = "visible";
            }         
            
            parent.className = "col-md-1";

            child.style.width = "70px";
            child.innerHTML = "Hide <<";

            filterToggle = 1;
        }

        else {
            while(i--) {
                filterDiv[i].style.visibility = "hidden";
            }         
            
            parent.className = "col-md-2";

            child.style.width = "155px";
            child.innerHTML = "Filter Results >>";

            filterToggle = 0;
        }

        //parent.removeChild(child);
    }


    function eqtlDetails(PValue, SNPName, SNPChr, SNPChrPos, Gene, ProbeChr, ProbeCenterChrPos ) 
    {
        var eqtlNode = document.getElementById("eqtlDataDiv");
        var eqtlTitleNode = document.getElementById("eqtlTitleDiv");
        var eqtlInnerNode = document.getElementById("eqtlDataInnerDiv");
        var rsTableNode = document.getElementById("rsTableDiv");
        var gwasTableNode = document.getElementById("gwasTableDiv");

        var htmlcontent = PValue;
        var ensembleLink = "http://www.ensembl.org/Homo_sapiens/Gene/ExpressionAtlas?g=" + Gene;
        var snpPosLink = "https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=" + SNPChr + SNPChrPos;
        var genePosLink = "https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=" + SNPChr + SNPChrPos;
        var dbSNPLink = "https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=" + SNPName;

        var span = "<span class='fa fa-external-link' style='margin-left: 2px; color: blue;'></span>";
        var space ="<span style='margin-left: 25px;'></span>";

        if(eqtlNode === null) {
            eqtlNode = document.createElement('div'); 
            
            eqtlNode.id = "eqtlDataDiv";
            eqtlNode.className = "col-md-5";
            eqtlNode.style.marginTop = "0px";
            eqtlNode.style.marginLeft = "80px";

            eqtlNode.style.fontSize = "14px";
            //eqtlNode.style.border = "1px solid #ccc";
            //eqtlNode.style.fontWeight = "bold";

            //geneNode.appendChild(eqtlNode);
            $("#transDiv").after(eqtlNode);
        }

        if(eqtlTitleNode === null) {
            eqtlTitleNode = document.createElement('div');

            eqtlTitleNode.id = "eqtlTitleDiv";
            eqtlTitleNode.style.textAlign = "center";
            //eqtlInnerNode.className = "col-md-4";
            //eqtlTitleNode.style.border = "1px solid #ccc";
            eqtlTitleNode.style.padding = "5px";

            eqtlNode.appendChild(eqtlTitleNode);
        }
        
        if(eqtlInnerNode === null) {
            eqtlInnerNode = document.createElement('div');

            eqtlInnerNode.id = "eqtlDataInnerDiv";
            //eqtlInnerNode.className = "col-md-4";
            eqtlInnerNode.style.border = "1px solid #ccc";
            eqtlInnerNode.style.padding = "10px";
            eqtlInnerNode.style.marginBottom = "20px";

            eqtlNode.appendChild(eqtlInnerNode);
        }

        if(rsTableNode === null) {
            rsTableNode = document.createElement('table');

            rsTableNode.id = "rsTableDiv";
            rsTableNode.className = "display";
            rsTableNode.style.border = "1px solid #ccc";
            rsTableNode.style.marginTop = "20px";
            rsTableNode.style.marginLeft = "0px";
            rsTableNode.style.padding = "5px";
            rsTableNode.style.fontSize = "small";

            eqtlNode.appendChild(rsTableNode);
        }

        if(gwasTableNode === null) {
            gwasTableNode = document.createElement('table');

            gwasTableNode.id = "gwasTableDiv";
            gwasTableNode.className = "display";
            gwasTableNode.style.border = "1px solid #ccc";
            gwasTableNode.style.marginTop = "20px";
            gwasTableNode.style.marginLeft = "0px";
            gwasTableNode.style.padding = "5px";
            gwasTableNode.style.fontSize = "small";

            eqtlNode.appendChild(gwasTableNode);
        }
    
        eqtlTitleNode.innerHTML = "<strong>" + "Trans-eQTL Details: " + "</strong>";

        eqtlInnerNode.innerHTML = "<div class=\"row\">" + "<div class=\"col-md-4\">" + 
            "<strong>" + "P-value: " + "</strong>" + "</div>" + "<div class=\"col-md-6\">" + PValue + "</div>" + "</div>" +
                "<div class=\"row\">" + "<div class=\"col-md-4\">" + 
            "<strong>" + "SNP-Pos: " + "</strong>" + "</div>" + "<div class=\"col-md-6\">" + SNPChr + ":" + SNPChrPos + "</div>" + "</div>" + 
                "<div class=\"row\">" + "<div class=\"col-md-4\">" +  
            "<strong>" + "Gene: " + "</strong>" + "</div>" + "<div class=\"col-md-3\">" + Gene + "</div>" + "<div class=\"col-md-3\">" + "<a href = " + ensembleLink + " target=\"_blank\">" + space + " Ensembl </a>" + span + "</div>" + "</div>" +
                "<div class=\"row\">" + "<div class=\"col-md-4\">" + 
            "<strong>" + "Gene-Pos: " + "</strong>" + "</div>" + "<div class=\"col-md-6\">" + ProbeChr + ":" + ProbeCenterChrPos + "</div>" + "</div>" +
                "<div class=\"row\">" + "<div class=\"col-md-4\">" + 
            "<strong>" + "rsid: " + "</strong>" + "</div>" + "<div class=\"col-md-3\">" + SNPName + "</div>" + "<div class=\"col-md-3\">" + "<a href = " + dbSNPLink + " target=\"_blank\">" + space + " dbSNP </a>" + span + "</div>" + "</div>";


            var b = document.createElement('a');
            var linktextcontent = "Other Gene Targets for: '" + SNPName + "'";
            var linkText = document.createTextNode(linktextcontent);
            b.style.textAlign = "center";
            b.appendChild(linkText);
            eqtlInnerNode.appendChild(b);

            
            var c = document.createElement('a');
            var newSpan = document.createElement('span');
            newSpan.id = "openTable";
            newSpan.style.marginLeft = "4px";
            newSpan.className = "glyphicon glyphicon-arrow-down";
            c.appendChild(newSpan);
            eqtlInnerNode.appendChild(c);

            //Table Initialization
            var header = rsTableNode.createTHead();                    
            var tblBody = document.createElement("tbody");
            rsTableNode.appendChild(tblBody);
            var th = document.createElement('th');
            th.innerHTML = "Gene";
            th.innerHTML = "P-Value";

            c.onclick = function() {  

                //Open Table Content
                if(newSpan.id == "openTable") 
                {     
                    //Change the icon
                    newSpan.className = "glyphicon glyphicon-arrow-up";
                    newSpan.id = "closeTable";

                    //Table exists
                    if (document.getElementById("rsTableDiv_wrapper")) 
                    {                                           
                        $("#rsTableDiv_wrapper").show();

                    }       

                    //Create Table
                    else
                    {
                        var formatP = d3.format(",.4e");         
                        var rsArr = [];
                        //SNP_interest = SNPName;

                        for(var i = 0; i < transData.length; i++) {
                            for(var j = 0; j < transData[i].length; j++)
                                if(transData[i][j].SNPName === SNPName){
                                    rsArr.push(transData[i][j]);
                                    rsArr[rsArr.length-1].PValue = formatP(rsArr[rsArr.length-1].PValue);
                                }
                        }

                        $('#rsTableDiv').DataTable( {
                            data: rsArr,

                            columns: [
                                { data: 'Gene', "title": 'Gene', 
                                    "render": function ( data, type, full, meta ) {
                                        return '<a href="#" onClick="addTrans(\''+data+'\')">'+data+'</a>';
                                    } 
                                },
                                { data: 'PValue', "title": 'P-Value' },                      
                                { data: 'ProbeChr', "title": 'Chr' },
                                { data: 'ProbeCenterChrPos', "title": 'Center-Pos'}         
                            ]
                        } );

                        $('#rsTableDiv').append('<caption>Table: eQTL Target Genes</caption>');

                    }

                }

                //Hide Table Content
                else
                {
                    newSpan.className = "glyphicon glyphicon-arrow-down";
                    newSpan.id = "openTable";

                    $("#rsTableDiv_wrapper").hide();
                }

            }


            //GWAS Data TABLE Creation...
            var gwasArr = gwasData[0].filter(function(x) { 
                                    return (x.Strongest_SNP_Risk_Allele).split('-')[0] === SNPName; 
                                });

            if(gwasArr.length > 0)
            {
                var br = document.createElement('br');
                eqtlInnerNode.appendChild(br);

                var g = document.createElement('a');
                var linktextcontent = "GWAS Results (" + gwasArr.length + ") for: '" + SNPName + "'";
                var linkText = document.createTextNode(linktextcontent);
                g.style.textAlign = "center";
                g.appendChild(linkText);
                eqtlInnerNode.appendChild(g);

                
                var h = document.createElement('a');
                var newSpan2 = document.createElement('span');
                newSpan2.id = "openTableGwas";
                newSpan2.style.marginLeft = "4px";
                newSpan2.className = "glyphicon glyphicon-arrow-down";
                h.appendChild(newSpan2);
                eqtlInnerNode.appendChild(h);

                //Table Initialization
                var header = gwasTableNode.createTHead();                    
                var tblBody = document.createElement("tbody");
                gwasTableNode.appendChild(tblBody);
                var th = document.createElement('th');
                th.innerHTML = "Gene";
                th.innerHTML = "P-Value";

                h.onclick = function() {  

                    //Open Table Content
                    if(newSpan2.id == "openTableGwas") 
                    {     
                        //Change the icon
                        newSpan2.className = "glyphicon glyphicon-arrow-up";
                        newSpan2.id = "closeTableGwas";

                        //Table exists
                        if (document.getElementById("gwasTableDiv_wrapper")) 
                        {                                           
                            $("#gwasTableDiv_wrapper").show();
                            
                        }       

                        //Create Table
                        else
                        {

                            $('#gwasTableDiv').DataTable( {
                                data: gwasArr,

                                columns: [
                                    { data: 'Disease_Phenotype', "title": 'Disease/Phenotype' },
                                    { data: 'P-value', "title": 'P-Value' },                      
                                    { data: 'Odds ratio or beta', "title": 'Odds-ratio/beta' },
                                    { data: 'PUBMEDID', "title": 'Pubmed ID' ,
                                     "render": function ( data, type, full, meta ) {
                                            var pmed = "https://www.ncbi.nlm.nih.gov/pubmed/?term=" + data;
                                            var pmedlink = "<a href = " + pmed + " target=\"_blank\">" + data + "</a>";
                                            return pmedlink;
                                        } 
                                    }         
                                ]
                            } );

                            $('#gwasTableDiv').append('<caption>Table: GWAS Results</caption>');

                        }

                    }

                    //Hide Table Content
                    else
                    {
                        newSpan2.className = "glyphicon glyphicon-arrow-down";
                        newSpan2.id = "openTableGwas";

                        $("#gwasTableDiv_wrapper").hide();
                    }

                }
            }          
    }


    function compare(a,b) 
    {
          if (a.Gene.length < b.Gene.length)
            return -1;
          if (a.Gene.length > b.Gene.length)
            return 1;
          
          return 0;
    }


    function geneSetTrans(frequency, source) 
    {
        var sourceArr = [], geneArr = [], fullArr = [], obj = {}, genes, seenGenes = [];
        var transNode = document.getElementById("transDiv");
        var geneSetNode = document.getElementById("geneSetDiv");

        $('#transDiv').empty(); // Remove contents of transDiv (trans eqtls)
        $('#geneDataDiv').remove(); //remove Gene Data
        $('#eqtlDataDiv').remove(); // Remove eqtl details

        var filterRow = document.getElementById("filterRow");
        filterRow.style.height = "0px";
        filterRow.style.visibility = "hidden";

        var filterCol = document.getElementById("filterCol");
        filterCol.style.height = "0px";
        filterCol.style.visibility = "hidden";

        var plotRow = document.getElementById("plotRow");
        plotRow.style.height = "0px";
        plotRow.style.visibility = "hidden";
        var plotCol = document.getElementById("plotCol");
        plotCol.style.height = "0px";

        sourceArr = Object.keys(source);
        
        for(var i = 0; i < sourceArr.length; i++) {
            seenGenes = [];
            genes = "";
            
            Object.keys(source[sourceArr[i]]).forEach(function(key) {

                    if( !isInArray(source[sourceArr[i]][key].Gene, seenGenes) ) {
                        seenGenes.push(source[sourceArr[i]][key].Gene);
                        genes = genes + source[sourceArr[i]][key].Gene + "; ";
                    }

            });

            genes = genes.slice(0, -2);
            
            if(seenGenes.length > 1) {
                obj = {rsid: sourceArr[i], Gene: genes };
                fullArr.push(obj); 
            }
        }

        if(geneSetNode === null) {
            geneSetNode = document.createElement('table');

            geneSetNode.id = "geneSetDiv";
            geneSetNode.className = "display";
            geneSetNode.style.border = "1px solid #ccc";
            geneSetNode.style.marginTop = "-20px";
            geneSetNode.style.padding = "5px";

            transNode.appendChild(geneSetNode);
        }

        //Table Initialization
            var header = geneSetNode.createTHead();                    
            var tblBody = document.createElement("tbody");
            geneSetNode.appendChild(tblBody);
            var th = document.createElement('th');

            fullArr.sort(compare);

        $('#geneSetDiv').DataTable( {
                    data: fullArr,
                    "order": [[ 0, "desc" ]],
                    retrieve: true,
                    columns: [                     
                        { data: 'Gene.length', "title": 'deneme', "visible": false},
                        { data: 'rsid', "title": 'SNP' },
                        { data: 'Gene', "title": 'Gene' }
                      
                    ]
                } );

    }


    function geneSetDrawTrans(source) 
    {
            var dataSize = geneData.length;
            
            var vizDiv = document.getElementById("transDiv");

                $('#transDiv').empty();
                vizDiv.style.border = "";
                vizDiv.style.textAlign = "left";
                
                removeControls();

                drawTrans(geneData, geneData[0].PValue, dataSize);
                
    }


    function drawTrans(data, topP, dataSize) 
    {       
        //draw in 'transDiv'
        var viz = d3.select('#transDiv');
        
        wdth = 635; //610
        
        var gene = document.getElementById("geneVal").value;
        var lastChar = gene.substr(gene.length - 1);

        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);

        if(gene == "")
            gene = "all";

        var genePlaceholder = document.getElementById("geneVal");
        genePlaceholder.placeholder = "Gene";

        var superscript = "â°Â¹Â²Â³â´âµâ¶â·â¸â¹",
            formatPower = function(d) { return (d + "").split("").map(function(c) { return superscript[c]; }).join(""); };
        
        svg = viz.append('svg')
                    .attr({
                        width: wdth + 30,
                        height: hght + 20
                    })
                    .append("svg:g")
                    .attr("class", "graph-svg-component")
                    .attr("id", "trans-svg")
                    .attr("transform", "translate(" + 0 + "," + 0 +")");

        var xScale = d3.scale.linear()
                        .domain([1, 100]) 
                        .range([0,wdth]);
        
        var xAxis = d3.svg.axis()
                        .scale(xScale)
                        .orient("bottom")
                        .ticks(0)
                        .tickSize(1);

        var yScale = d3.scale.log()
                        .domain([1, topP]) 
                        .range([hght - 10,10]); 
        
        var y = d3.scale.log()
                        .domain([1e-5, 100])
                        .range([hght - 10, 0]);


        var yAxis = d3.svg.axis()
                        .scale(yScale)                      
                        .tickFormat(function(d) { return "e" + Math.round(Math.log10(d)); })
                        .orient("left")
                        .tickValues([1e-100,1e-80,1e-40,1e-20])
                        .ticks(8)
                        .tickSize(1);
        

        // Define the div for the tooltip
        var div = d3.select("body")
                        .append("div")   
                        .attr("class", "tooltip")               
                        .style("opacity", 0);

        var formatP = d3.format(",.4e");

        var selection = svg.selectAll("circle")
                            .data(data, function(d) { return d.PValue+""+d.SNPName+""+d.Gene+""+d.ProbeCenterChrPos; });

        selection.exit().remove();
        
        selection.enter().append('circle')
                        .attr('class', 'circle')
                        .attr('cx', function (d, i) {
                            if(plot == 1)
                                return (i+2)*12 + 10;

                            else if(plot == 2) 
                            {
                                var chrStartInd = [45, 95, 143, 183, 211, 257, 291, 323, 352, 380, 407, 434, 461, 484, 505, 525, 543, 560, 576, 588, 601, 610, 620];
                                
                                if(geneSnpToggle == 0)
                                {
                                    if(d.SNPChr == 'X' || d.SNPChr == 'Y')
                                        return chrStartInd[22] + (d.SNPChrPos / 5000000);
                                    else
                                        return chrStartInd[d.SNPChr - 1] + (d.SNPChrPos / 5000000);
                                }
                                    
                                else
                                    return chrStartInd[d.ProbeChr - 1] + (d.ProbeCenterChrPos / 5000000);
                            }
                                //return (d.SNPChr)*24;
                        })  
                        .attr('cy', function (d) {return yScale(d.PValue)+5;})//yScale(Math.pow(d.PValue, 0.46))+5;})
                        .attr('r', function (d) {
                            if(d.SNPName == SNP_interest)
                                return 6;    
                            else
                                return 5;
                        })
                        .attr("transform", "translate(20,0)")
                        .style("fill", function (d) {
                               var gwaslist = gwasData[0].filter(function(x) { 
                                    return (x.Strongest_SNP_Risk_Allele).split('-')[0] === d.SNPName; 
                                });

                                if(d.SNPName == SNP_interest)
                                    return "red";

                                if(gwaslist.length > 0)
                                    return "black";
                                         
                        })
                        .on("mouseover", function(d) {
                            this.setAttribute('r', '8');
                            d3.select(this).style('fill', 'red');
                            //SNP_hover = d.SNPName;
                            
                            var tipHtml = "<strong>" + "P-value: " + "</strong>" + formatP(d.PValue) + "<br/>" + 
                                        "<strong>" + "rsid: " + "</strong>" + d.SNPName + "<br/>"  + 
                                        "<strong>" + "SNP-Pos: " + "</strong>" + d.SNPChr + ":" + d.SNPChrPos + "<br/>" +
                                        "<strong>" + "Gene: " + "</strong>" + d.Gene + "<br/>"  +
                                        "<strong>" + "Gene-Pos: " + "</strong>" + d.ProbeChr + ":" + d.ProbeCenterChrPos;

                            var gwaslist = gwasData[0].filter(function(x) { 
                                    return (x.Strongest_SNP_Risk_Allele).split('-')[0] === d.SNPName; 
                                });
                            
                            for (var i = 0; i < gwaslist.length; i++)
                            {
                                tipHtml = tipHtml + "<hr style=\"height:1px;\">" + "<strong>" + "Disease/Trait: " + "</strong>" +  gwaslist[i].Disease_Phenotype + "<br/>"   + "<strong>" + "Pubmed ID: " + "</strong>" + gwaslist[i].PUBMEDID + "<br/>";
                            }
                        

                            div.transition()        
                                .duration(100)      
                                .style("opacity", .9);     

                            div.html(tipHtml)  
                                .style("left", (d3.event.pageX + 10) + "px")     
                                .style("top", (d3.event.pageY - 28) + "px"); 

                        })
                        .on("click", function(d){
                            SNP_interest = d.SNPName;

                            d3.selectAll(".tooltip").remove();
                            
                            addTrans(gene);

                            $('.dataTables_wrapper').empty();
                            $('#gwasTableDiv_wrapper').remove();
                            //$('#saveTableClone').remove();
                            $('#rsTableDiv_wrapper').remove();
                            eqtlDetails(formatP(d.PValue), d.SNPName, d.SNPChr, d.SNPChrPos, d.Gene, d.ProbeChr, d.ProbeCenterChrPos );
                        })
                        .on("mouseout", function(d) {
                            
                            d3.select(this).style('fill', function (d) {
                                
                                var gwaslist = gwasData[0].filter(function(x) { 
                                    return (x.Strongest_SNP_Risk_Allele).split('-')[0] === d.SNPName; 
                                });

                                if(d.SNPName == SNP_interest)
                                    return "red";

                                if(gwaslist.length > 0)
                                    return "black";

                                }); 

                            if(d.SNPName == SNP_interest)
                                this.setAttribute('r', '6');
                            
                            else
                                this.setAttribute('r', '5');
                            
                            div.transition()        
                                .duration(100)      
                                .style("opacity", 0);
                        
                        });
     

        svg.append("text")
              .attr("transform", "rotate(-90)")
              .attr("y", 0)
              .attr("x", -200)
              .attr("dy", "1em")
              .style("text-anchor", "middle")
              .style("font-size", "12px")
              .style("font-weight", "normal")
              .text("-log10(P-value)");


        svg.append("text")
                .attr("x", (wdth / 2 ))             
                .attr("y", 440)
                .attr("text-anchor", "middle")  
                .style("font-size", "14px") 
                .text("Trans-eQTLs");
        
        
        // create left yAxis
        svg.append("g")
                .attr("class", "axis axis--y")
                .attr("transform", "translate(45, 0)")
                .call(yAxis);

        // create xAxis
        svg.append("g")
                .attr("class", "axis axis--x")
                .attr("transform", "translate(45, 400)")
                .call(xAxis);

        
        //add border
       svg.append("rect")
                .attr("x", 0)
                .attr("y", 0)
                .attr("height", hght+20)
                .attr("width", wdth+30)
                .style("stroke", "dodgerblue")
                .style("fill", "none")
                .style("stroke-width", 1);

        var missingDataDiv = document.getElementById("missingDataset");
        missingDataDiv.style.visibility = "hidden";
        missingDataDiv.style.display = "none";

        var missingDataText = document.getElementById("missingDataText");
        missingDataText.innerHTML = "Please select 1 or more datasets.";
    }


    function manhattanPlot()
    {
        
        if(document.getElementById("manPlot").innerText == "Manhattan Plot")
        {

            var svg = d3.select("body").transition();

            svg.selectAll(".circle")   // change the line
                .duration(750)
                .attr('cx', function (d, i) {         
                    var chrStartInd = [45, 95, 143, 183, 211, 257, 291, 323, 352, 380, 407, 434, 461, 484, 505, 525, 543, 560, 576, 588, 601, 610, 620]; 
                    
                    if(geneSnpToggle == 0)
                    {
                        if(d.SNPChr == 'X' || d.SNPChr == 'Y')
                            return chrStartInd[22] + (d.SNPChrPos / 5000000);
                        else
                            return chrStartInd[d.SNPChr - 1] + (d.SNPChrPos / 5000000);
                    }

                    else
                        return chrStartInd[d.ProbeChr - 1] + (d.ProbeCenterChrPos / 5000000);
                }); 

            plot = 2;

            document.getElementById("manPlot").innerText = "P-value Ordered Plot";
            document.getElementById("manPlot").style.width = "220px";

            var xScale = d3.scale.linear()
                        .domain([1, 620]) 
                        .range([0,620]);
        

            var xAxis = d3.svg.axis().scale(xScale).orient("bottom").tickValues([2, 70, 118, 158, 196, 232, 266, 298, 327, 355, 382, 409, 436, 459, 480, 500, 518, 535, 551, 563, 574, 585,600]).tickFormat(function(n, i) { return i+1 }).tickSize(5);

            svg.select(".axis--x").transition().duration(500).call(xAxis).style("font-size", "9px").style("font-weight", "400"); 

         }


        else if(document.getElementById("manPlot").innerText == "P-value Ordered Plot")
        {

            var svg = d3.select("body").transition();

            svg.selectAll(".circle")   // change the line
                .duration(750)
                .attr('cx', function (d, i) {
                    return (i+2)*12 + 10;
                }); 

            plot = 1;

            document.getElementById("manPlot").innerText = "Manhattan Plot";
            document.getElementById("manPlot").style.width = "170px";

            var xScale = d3.scale.linear()
                        .domain([1, 100]) 
                        .range([0,wdth]);

            var xAxis = d3.svg.axis()
                        .scale(xScale)
                        .orient("bottom")
                        .ticks(0)
                        .tickSize(1);

            svg.select(".axis--x").transition().duration(750).call(xAxis);

         }

    }



    function drawCis(datasetvalues) 
    {
        var cisDiv = document.getElementById("igvDiv");
        cisDiv.style.visibility = "visible";
        cisDiv.style.marginTop = "-10px";

        var genePlaceholder = document.getElementById("geneVal");
        var gene = document.getElementById("geneVal").value;

        var geneArray = gene.split(',');

        gene = geneArray[0];
        //var lastChar = gene.substr(gene.length - 1);
        
        //if(lastChar == ',')
            //gene = gene.substring(0, gene.length - 1);

        var location = gene;

        if (location == "") {
            location = "chr1:155,160,475-155,184,282";
            genePlaceholder.placeholder = location;
        }

        else
            genePlaceholder.placeholder = "Gene";

        var div = $("#igvDiv")[0],
                options = {
                    showNavigation: true,
                    showRuler: true,
                    locus: location,
                    flanking: 500000,
                    reference: {
                        id: "hg19"
                    },
                    formats: {
                            gwasSNPS: {
                                coords: 0,
                                chr: 0,
                                start: 1,
                                end: 2,
                                fields: ["Chr", "chromStart", "chromEnd", "Strongest_SNP_risk_allele", "Disease_Phenotype", "P-value", "Odds ratio or beta", "PUBMEDID"]
                            }
                        },
                    tracks: [
                            {
                                featureType: 'annotation',
                                format: 'gwasSNPS',
                                indexed: false,
                                color: 'rgb(100,200,200)',
                                displayMode: 'EXPANDED',
                                url: "./data/gwas.txt",
                                name: 'GWAS Catalog',
                                height: "150",
                                order: 9999
                            },
                        {
                            name: "Genes",
                            type: "annotation",
                            format: "gtf",
                            sourceType: "file",
                            height: "200",
                            url: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg19/genes/gencode.v18.annotation.sorted.gtf.gz",
                            indexURL: "https://s3.amazonaws.com/igv.broadinstitute.org/annotations/hg19/genes/gencode.v18.annotation.sorted.gtf.gz.tbi",
                            visibilityWindow: 10000000,
                            order: Number.MAX_VALUE,
                            order: 10000,
                            displayMode: "EXPANDED"
                        },

                    ]
                };

        igv.createBrowser(div, options);

        setTimeout(loadCisTracks, 1000, datasetvalues);
        igv.browser.zoomOut();
        igv.browser.zoomOut();
        igv.browser.zoomOut();

    }


    function loadCisTracks(datasetvalues) 
    {
        var dataset = 13;
        var commonCis = [];
        var remaining = datasetvalues.length;

        var spinner = new Spinner();
        spinner.spin(document.getElementById("queryRow"));

        var gene = document.getElementById("geneVal").value;

        $('#multipleGeneRow').empty();
        
        var lastChar = gene.substr(gene.length - 1);

        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);
        
        var geneArray = gene.split(',');

        gene = geneArray[0];

        if(geneArray.length > 1)
        {
            /*var extraGeneDiv = document.getElementById("extraGene");
            extraGeneDiv.style.visibility = "visible";
            extraGeneDiv.innerHTML = "Please enter a single gene in the gene-box for 'cis-eQTL' queries.";
            return true;*/
            var multipleGeneNode = document.getElementById("multipleGeneRow");
            multipleGeneNode.style.visibility = "visible";

            for (var i = 0; i < geneArray.length; i++)
            {
                var b = document.createElement('a');
                var linktextcontent = geneArray[i];
                var linkText = document.createTextNode(linktextcontent);
                b.style.textAlign = "center";
                if(i != 0)
                    b.style.marginLeft = "10px";
                b.appendChild(linkText);
                multipleGeneNode.appendChild(b);
                
                b.onclick = function() {
                                igv.browser.search($(event.target).text());
                            }
            }

        }

        for (var i = 0; i < datasetvalues.length; i++) {

            //Framingham dataset
            if(datasetvalues[i] == 11) {
                var file = "./data/fram/fram/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData11 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 001) {
                    cisData001 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 002) {
                    cisData002 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 003) {
                    cisData003 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            //GTEx dataset
            else if (datasetvalues[i] == 12) {
                var file = "./data/gtex/gtex/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData12 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            //Westra dataset
            else if (datasetvalues[i] == 13) {
                var file = "./data/westra/westra/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData13 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 14) {
                var file = "./data/lude/lude/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData14 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 15) {
                var file = "./data/pierce/pierce/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData15 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 16) {
                d3.tsv("./data/yamada/yamada.txt", function(rows16) {
                    cisData16 = rows16;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 17) {
                var file = "./data/eqtlgen/eqtlgen/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData17 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 18) {
                    cisData18 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
                /*d3.tsv("./data/user/user1.txt", function(rows101) { 
                    cisData101 = rows101;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                }); */
            }

            else if (datasetvalues[i] == 19) {
                    cisData19 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 190) {
                    cisData190 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 21) {                
                var file = "./data/jager/monocytes-aa/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData21 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                }); 
            }

            else if (datasetvalues[i] == 22) {
                var file = "./data/jager/monocytes-ea/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData22 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 23) {
                var file = "./data/jager/monocytes-eu/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData23 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 25) {               
                d3.tsv("./data/fairfax/mono.txt", function(rows25) {
                    cisData25 = rows25;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 26) {               
                var file = "./data/fairfax/mono-naive/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData26 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 27) {
                var file = "./data/dice/monocytes/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData27 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 28) {
                var file = "./data/dice/mono-non/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData28 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 29) {
                    cisData29 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 290) {
                    cisData290 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 31) {
                d3.tsv("./data/jager/cd4T-aa.txt", function(rows31) {
                    cisData31 = rows31;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 32) {
                d3.tsv("./data/jager/cd4T-ea.txt", function(rows32) {
                    cisData32 = rows32;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 33) {
                var file = "./data/jager/cd4T-eu/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData33 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 35) {
                var file = "./data/dice/cd4/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData35 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 36) {
                var file = "./data/dice/treg-naive/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData36 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 37) {
                var file = "./data/dice/treg-mem/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData37 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 38) {
                var file = "./data/dice/th1/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData38 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 39) {
                var file = "./data/dice/thstar/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData39 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 390) {
                var file = "./data/dice/th17/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData390 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 391) {
                var file = "./data/dice/th2/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData391 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 392) {
                var file = "./data/dice/tfh/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData392 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 393) {
                    cisData393 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 394) {
                    cisData394 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 395) {
                    cisData395 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 41) {
                d3.tsv("./data/fairfax/bcell.txt", function(rows41) {
                    cisData41 = rows41;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 42) {
                var file = "./data/dice/bcell/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData42 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 51) {
                d3.tsv("./data/fairfax/neutrophils.txt", function(rows51) {
                    cisData51 = rows51;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 52) {
                    cisData52 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 53) {
                    cisData53 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 61) {
                var file = "./data/fairfax/mono-lps2/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData61 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 62) {
                var file = "./data/fairfax/mono-lps24/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData62 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });              
            }

            else if (datasetvalues[i] == 63) {
                var file = "./data/fairfax/mono-ifn/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData63 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 71) {
                var file = "./data/dermitzakis/eur/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData71 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 72) {
                var file = "./data/dermitzakis/yri/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData72 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 73) {
                    cisData73 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 74) {
                    cisData74 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 75) {
                    cisData75 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 76) {
                    cisData76 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 77) {
                    cisData77 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 78) {
                    cisData78 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 79) {
                    cisData79 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 790) {
                    cisData790 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 791) {
                    cisData791 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 792) {
                    cisData792 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 81) {
                var file = "./data/heijmans/meqtl/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData81 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 82) {
                d3.tsv("./data/heijmans/eqtm.txt", function(rows82) {
                    cisData82 = rows82;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 83) {
                    cisData83 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 84) {
                    cisData84 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 85) {
                    cisData85 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 86) {
                    cisData86 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 87) {
                    cisData87 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 88) {
                    cisData88 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 89) {
                    cisData89 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 890) {
                    cisData890 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 891) {
                    cisData891 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 892) {
                    cisData892 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 893) {
                    cisData893 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 91) {
                var file = "./data/dice/nk/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData91 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 92) {
                var file = "./data/dice/CD8/" + gene;  
                
                d3.text(file, function(r) { 
                    var result = "GENE\tSNP\n" + r;
                    cisData92 = d3.tsv.parse(result);
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 93) {
                    cisData93 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 101) {
                d3.tsv("./data/user/user1.txt", function(rows101) { 
                    cisData101 = rows101;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                
            }

            else if (datasetvalues[i] == 102) {
                d3.tsv("./data/user/user2.txt", function(rows102) { 
                    cisData102 = rows102;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 103) {
                d3.tsv("./data/user/user3.txt", function(rows103) { 
                    cisData103 = rows103;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });               
            }

            else if (datasetvalues[i] == 211) {
                d3.tsv("./data/hipc/mayo/mayo-day0-snp.txt", function(rows211) {
                    cisData211 = rows211;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 212) {
                d3.tsv("./data/hipc/mayo/mayo-day3-snp.txt", function(rows212) {
                    cisData212 = rows212;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 213) {
                d3.tsv("./data/hipc/mayo/mayo-day28-snp.txt", function(rows213) {
                    cisData213 = rows213;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 221) {
                d3.tsv("./data/hipc/nih/nih-day-7-snp.txt", function(rows221) {
                    cisData221 = rows221;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 222) {
                d3.tsv("./data/hipc/nih/nih-day0-snp.txt", function(rows222) {
                    cisData222 = rows222;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 223) {
                d3.tsv("./data/hipc/nih/nih-day1-snp.txt", function(rows223) {
                    cisData223 = rows223;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 224) {
                d3.tsv("./data/hipc/nih/nih-day7-snp.txt", function(rows224) {
                    cisData224 = rows224;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 225) {
                d3.tsv("./data/hipc/nih/nih-day70-snp.txt", function(rows225) {
                    cisData225 = rows225;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 231) {
                d3.tsv("./data/hipc/yale/yale-day0-snp.txt", function(rows231) {
                    cisData231 = rows231;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 232) {
                d3.tsv("./data/hipc/yale/yale-day2-snp.txt", function(rows232) {
                    cisData232 = rows232;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 233) {
                d3.tsv("./data/hipc/yale/yale-day7-snp.txt", function(rows233) {
                    cisData233 = rows233;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 234) {
                d3.tsv("./data/hipc/yale/yale-day28-snp.txt", function(rows234) {
                    cisData234 = rows234;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });
            }

            else if (datasetvalues[i] == 241) {
                d3.tsv("./data/hipc/emory/emory-day0-snp.txt", function(rows241) {
                    cisData241 = rows241;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 242) {
                d3.tsv("./data/hipc/emory/emory-day1-snp.txt", function(rows242) {
                    cisData242 = rows242;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 243) {
                d3.tsv("./data/hipc/emory/emory-day3-snp.txt", function(rows243) {
                    cisData243 = rows243;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 244) {
                d3.tsv("./data/hipc/emory/emory-day7-snp.txt", function(rows244) {
                    cisData244 = rows244;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 245) {
                d3.tsv("./data/hipc/emory/emory-day14-snp.txt", function(rows245) {
                    cisData245 = rows245;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 301) {
                d3.tsv("./data/hipc/chikv/chik-day1-snp.txt", function(rows301) {
                    cisData301 = rows301;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 302) {
                d3.tsv("./data/hipc/chikv/chik-day9-snp.txt", function(rows302) {
                    cisData302 = rows302;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 401) {
                d3.tsv("./data/hipc/zikv/zika-acute-snp.txt", function(rows401) {
                    cisData401 = rows401;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 402) {
                d3.tsv("./data/hipc/zikv/zika-lateacute-snp.txt", function(rows402) {
                    cisData402 = rows402;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            else if (datasetvalues[i] == 403) {
                d3.tsv("./data/hipc/zikv/zika-conv-snp.txt", function(rows403) {
                    cisData403 = rows403;
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues); 
                });                 
            }

            // Datasets added in July 2020

            else if (datasetvalues[i] == 396) {
                    cisData396 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 94) {
                    cisData94 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

           /* else if (datasetvalues[i] == 191) {
                    cisData191 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            } */

            else if (datasetvalues[i] == 291) {
                    cisData291 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 292) {
                    cisData292 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 293) {
                    cisData293 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 397) {
                    cisData397 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 95) {
                    cisData95 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 43) {
                    cisData43 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 96) {
                    cisData96 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 97) {
                    cisData97 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 98) {
                    cisData98 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 990) {
                    cisData990 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 294) {
                    cisData294 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 64) {
                    cisData64 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 65) {
                    cisData65 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 66) {
                    cisData66 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 67) {
                    cisData67 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 68) {
                    cisData68 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 69) {
                    cisData69 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 99) {
                    cisData99 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 131) {
                    cisData131 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 132) {
                    cisData132 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 133) {
                    cisData133 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 134) {
                    cisData134 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 135) {
                    cisData135 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            else if (datasetvalues[i] == 136) {
                    cisData136 = {};
                    remaining--;
                    
                    if(remaining == 0)
                        drawCisTracks(datasetvalues);
            }

            //Study does not have cis-eqtl data
            else
                remaining--;
        }

        spinner.stop();

    }


    function drawCisTracks(datasetvalues) 
    {
        var geneCis = [];
        var gene = document.getElementById("geneVal").value;

        var lastChar = gene.substr(gene.length - 1);

        if(lastChar == ',')
            gene = gene.substring(0, gene.length - 1);
        
        var obj = {};

       
        for (var i = 0; i < datasetvalues.length; i++) {

            if(datasetvalues[i] == 11) {
                for(var j = 0; j < cisData11.length; j++)
                {
                    if(cisData11[j].GENE == gene ) 
                    {
                        obj = {source: "11", snp: cisData11[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 001) {
                for(var j = 0; j < cisData001.length; j++)
                {
                    if(cisData001[j].GENE == gene ) 
                    {
                        obj = {source: "001", snp: cisData001[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 003) {
                for(var j = 0; j < cisData003.length; j++)
                {
                    if(cisData003[j].GENE == gene ) 
                    {
                        obj = {source: "003", snp: cisData003[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 12) {
                for(var j = 0; j < cisData12.length; j++)
                {
                    if(cisData12[j].GENE == gene ) 
                    {
                        obj = {source: "12", snp: cisData12[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 13) {
                for(var j = 0; j < cisData13.length; j++)
                {
                    if(cisData13[j].GENE == gene ) 
                    {
                        obj = {source: "13", snp: cisData13[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 14) {
                for(var j = 0; j < cisData14.length; j++)
                {
                    if(cisData14[j].GENE == gene ) 
                    {
                        obj = {source: "14", snp: cisData14[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 15) {
                for(var j = 0; j < cisData15.length; j++)
                {
                    if(cisData15[j].GENE == gene ) 
                    {
                        obj = {source: "15", snp: cisData15[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 16) {
                for(var j = 0; j < cisData16.length; j++)
                {
                    if(cisData16[j].GENE == gene ) 
                    {
                        obj = {source: "16", snp: cisData16[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 17) {
                for(var j = 0; j < cisData17.length; j++)
                {
                    if(cisData17[j].GENE == gene ) 
                    {
                        obj = {source: "17", snp: cisData17[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 18) {
                for(var j = 0; j < cisData18.length; j++)
                {
                    if(cisData18[j].GENE == gene ) 
                    {
                        obj = {source: "18", snp: cisData18[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 19) {
                for(var j = 0; j < cisData19.length; j++)
                {
                    if(cisData19[j].GENE == gene ) 
                    {
                        obj = {source: "19", snp: cisData19[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 190) {
                for(var j = 0; j < cisData190.length; j++)
                {
                    if(cisData190[j].GENE == gene ) 
                    {
                        obj = {source: "190", snp: cisData190[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 21) {
                for(var j = 0; j < cisData21.length; j++)
                {
                    if(cisData21[j].GENE == gene ) 
                    {
                        obj = {source: "21", snp: cisData21[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 22) {
                for(var k = 0; k < cisData22.length; k++)
                {
                    if(cisData22[k].GENE == gene )
                    {
                        obj = {source: "22", snp: cisData22[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 23) {
                for(var k = 0; k < cisData23.length; k++)
                {
                    if(cisData23[k].GENE == gene )
                    {
                        obj = {source: "23", snp: cisData23[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 25) {
                for(var k = 0; k < cisData25.length; k++)
                {
                    if(cisData25[k].GENE == gene )
                    {
                        obj = {source: "25", snp: cisData25[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 26) {
                for(var k = 0; k < cisData26.length; k++)
                {
                    if(cisData26[k].GENE == gene )
                    {
                        obj = {source: "26", snp: cisData26[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 27) {
                for(var k = 0; k < cisData27.length; k++)
                {
                    if(cisData27[k].GENE == gene )
                    {
                        obj = {source: "27", snp: cisData27[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 28) {
                for(var k = 0; k < cisData28.length; k++)
                {
                    if(cisData28[k].GENE == gene )
                    {
                        obj = {source: "28", snp: cisData28[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 29) {
                for(var k = 0; k < cisData29.length; k++)
                {
                    if(cisData29[k].GENE == gene )
                    {
                        obj = {source: "29", snp: cisData29[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 290) {
                for(var k = 0; k < cisData290.length; k++)
                {
                    if(cisData290[k].GENE == gene )
                    {
                        obj = {source: "290", snp: cisData290[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 31) {
                for(var k = 0; k < cisData31.length; k++)
                {
                    if(cisData31[k].GENE == gene )
                    {
                        obj = {source: "31", snp: cisData31[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 32) {
                for(var k = 0; k < cisData32.length; k++)
                {
                    if(cisData32[k].GENE == gene )
                    {
                        obj = {source: "32", snp: cisData32[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 33) {
                for(var k = 0; k < cisData33.length; k++)
                {
                    if(cisData33[k].GENE == gene )
                    {
                        obj = {source: "33", snp: cisData33[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 35) {
                for(var k = 0; k < cisData35.length; k++)
                {
                    if(cisData35[k].GENE == gene )
                    {
                        obj = {source: "35", snp: cisData35[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 36) {
                for(var k = 0; k < cisData36.length; k++)
                {
                    if(cisData36[k].GENE == gene )
                    {
                        obj = {source: "36", snp: cisData36[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 37) {
                for(var k = 0; k < cisData37.length; k++)
                {
                    if(cisData37[k].GENE == gene )
                    {
                        obj = {source: "37", snp: cisData37[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 38) {
                for(var k = 0; k < cisData38.length; k++)
                {
                    if(cisData38[k].GENE == gene )
                    {
                        obj = {source: "38", snp: cisData38[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 39) {
                for(var k = 0; k < cisData39.length; k++)
                {
                    if(cisData39[k].GENE == gene )
                    {
                        obj = {source: "39", snp: cisData39[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 390) {
                for(var k = 0; k < cisData390.length; k++)
                {
                    if(cisData390[k].GENE == gene )
                    {
                        obj = {source: "390", snp: cisData390[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 391) {
                for(var k = 0; k < cisData391.length; k++)
                {
                    if(cisData391[k].GENE == gene )
                    {
                        obj = {source: "391", snp: cisData391[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 392) {
                for(var k = 0; k < cisData392.length; k++)
                {
                    if(cisData392[k].GENE == gene )
                    {
                        obj = {source: "392", snp: cisData392[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 393) {
                for(var k = 0; k < cisData393.length; k++)
                {
                    if(cisData393[k].GENE == gene )
                    {
                        obj = {source: "393", snp: cisData393[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 394) {
                for(var k = 0; k < cisData394.length; k++)
                {
                    if(cisData394[k].GENE == gene )
                    {
                        obj = {source: "394", snp: cisData394[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 395) {
                for(var k = 0; k < cisData395.length; k++)
                {
                    if(cisData395[k].GENE == gene )
                    {
                        obj = {source: "395", snp: cisData395[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 41) {
                for(var k = 0; k < cisData41.length; k++)
                {
                    if(cisData41[k].GENE == gene )
                    {
                        obj = {source: "41", snp: cisData41[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 42) {
                for(var k = 0; k < cisData42.length; k++)
                {
                    if(cisData42[k].GENE == gene )
                    {
                        obj = {source: "42", snp: cisData42[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 51) {
                for(var k = 0; k < cisData51.length; k++)
                {
                    if(cisData51[k].GENE == gene )
                    {
                        obj = {source: "51", snp: cisData51[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 52) {
                for(var k = 0; k < cisData52.length; k++)
                {
                    if(cisData52[k].GENE == gene )
                    {
                        obj = {source: "52", snp: cisData52[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 53) {
                for(var k = 0; k < cisData53.length; k++)
                {
                    if(cisData53[k].GENE == gene )
                    {
                        obj = {source: "53", snp: cisData53[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 61) {
                for(var k = 0; k < cisData61.length; k++)
                {
                    if(cisData61[k].GENE == gene )
                    {
                        obj = {source: "61", snp: cisData61[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 62) {
                for(var k = 0; k < cisData62.length; k++)
                {
                    if(cisData62[k].GENE == gene )
                    {
                        obj = {source: "62", snp: cisData62[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 63) {
                for(var k = 0; k < cisData63.length; k++)
                {
                    if(cisData63[k].GENE == gene )
                    {
                        obj = {source: "63", snp: cisData63[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 71) {
                for(var k = 0; k < cisData71.length; k++)
                {
                    if(cisData71[k].GENE == gene )
                    {
                        obj = {source: "71", snp: cisData71[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 72) {
                for(var k = 0; k < cisData72.length; k++)
                {
                        obj = {source: "72", snp: cisData72[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 73) {
                for(var k = 0; k < cisData73.length; k++)
                {
                        obj = {source: "73", snp: cisData73[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 74) {
                for(var k = 0; k < cisData74.length; k++)
                {
                        obj = {source: "74", snp: cisData74[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 75) {
                for(var k = 0; k < cisData75.length; k++)
                {
                        obj = {source: "75", snp: cisData75[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 76) {
                for(var k = 0; k < cisData76.length; k++)
                {
                        obj = {source: "76", snp: cisData76[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 77) {
                for(var k = 0; k < cisData77.length; k++)
                {
                        obj = {source: "77", snp: cisData77[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 78) {
                for(var k = 0; k < cisData78.length; k++)
                {
                        obj = {source: "78", snp: cisData78[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 79) {
                for(var k = 0; k < cisData79.length; k++)
                {
                        obj = {source: "79", snp: cisData79[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 790) {
                for(var k = 0; k < cisData790.length; k++)
                {
                        obj = {source: "790", snp: cisData790[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 791) {
                for(var k = 0; k < cisData791.length; k++)
                {
                        obj = {source: "791", snp: cisData791[k].SNP}
                        geneCis.push(obj);
                }
            }

            else if(datasetvalues[i] == 792) {
                for(var k = 0; k < cisData792.length; k++)
                {
                        obj = {source: "792", snp: cisData792[k].SNP}
                        geneCis.push(obj);
                }
            }


            else if(datasetvalues[i] == 81) {
                for(var k = 0; k < cisData81.length; k++)
                {
                    if(cisData81[k].GENE == gene )
                    {
                        obj = {source: "81", snp: cisData81[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 82) {
                for(var k = 0; k < cisData82.length; k++)
                {
                    if(cisData82[k].GENE == gene )
                    {
                        obj = {source: "82", snp: cisData82[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 83) {
                for(var k = 0; k < cisData83.length; k++)
                {
                    if(cisData83[k].GENE == gene )
                    {
                        obj = {source: "83", snp: cisData83[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 84) {
                for(var k = 0; k < cisData84.length; k++)
                {
                    if(cisData84[k].GENE == gene )
                    {
                        obj = {source: "84", snp: cisData84[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 85) {
                for(var k = 0; k < cisData85.length; k++)
                {
                    if(cisData85[k].GENE == gene )
                    {
                        obj = {source: "85", snp: cisData85[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 86) {
                for(var k = 0; k < cisData86.length; k++)
                {
                    if(cisData86[k].GENE == gene )
                    {
                        obj = {source: "86", snp: cisData86[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 87) {
                for(var k = 0; k < cisData87.length; k++)
                {
                    if(cisData87[k].GENE == gene )
                    {
                        obj = {source: "87", snp: cisData87[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 88) {
                for(var k = 0; k < cisData88.length; k++)
                {
                    if(cisData88[k].GENE == gene )
                    {
                        obj = {source: "88", snp: cisData88[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 89) {
                for(var k = 0; k < cisData89.length; k++)
                {
                    if(cisData89[k].GENE == gene )
                    {
                        obj = {source: "89", snp: cisData89[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 890) {
                for(var k = 0; k < cisData890.length; k++)
                {
                    if(cisData890[k].GENE == gene )
                    {
                        obj = {source: "890", snp: cisData890[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 891) {
                for(var k = 0; k < cisData891.length; k++)
                {
                    if(cisData891[k].GENE == gene )
                    {
                        obj = {source: "891", snp: cisData891[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 892) {
                for(var k = 0; k < cisData892.length; k++)
                {
                    if(cisData892[k].GENE == gene )
                    {
                        obj = {source: "892", snp: cisData892[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 893) {
                for(var k = 0; k < cisData893.length; k++)
                {
                    if(cisData893[k].GENE == gene )
                    {
                        obj = {source: "893", snp: cisData893[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 91) {
                for(var k = 0; k < cisData91.length; k++)
                {
                    if(cisData91[k].GENE == gene )
                    {
                        obj = {source: "91", snp: cisData91[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 92) {
                for(var k = 0; k < cisData92.length; k++)
                {
                    if(cisData92[k].GENE == gene )
                    {
                        obj = {source: "92", snp: cisData92[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 93) {
                for(var k = 0; k < cisData93.length; k++)
                {
                    if(cisData93[k].GENE == gene )
                    {
                        obj = {source: "93", snp: cisData93[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 101) {
                for(var k = 0; k < cisData101.length; k++)
                {
                    if(cisData101[k].GENE == gene )
                    {
                        obj = {source: "101", snp: cisData101[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 102) {
                for(var k = 0; k < cisData102.length; k++)
                {
                    if(cisData102[k].GENE == gene )
                    {
                        obj = {source: "102", snp: cisData102[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 103) {
                for(var k = 0; k < cisData103.length; k++)
                {
                    if(cisData103[k].GENE == gene )
                    {
                        obj = {source: "103", snp: cisData103[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 211) {
                for(var k = 0; k < cisData211.length; k++)
                {
                    if(cisData211[k].GENE == gene )
                    {
                        obj = {source: "211", snp: cisData211[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 212) {
                for(var k = 0; k < cisData212.length; k++)
                {
                    if(cisData212[k].GENE == gene )
                    {
                        obj = {source: "212", snp: cisData212[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 213) {
                for(var k = 0; k < cisData213.length; k++)
                {
                    if(cisData213[k].GENE == gene )
                    {
                        obj = {source: "213", snp: cisData213[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 221) {
                for(var k = 0; k < cisData221.length; k++)
                {
                    if(cisData221[k].GENE == gene )
                    {
                        obj = {source: "221", snp: cisData221[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 222) {
                for(var k = 0; k < cisData222.length; k++)
                {
                    if(cisData222[k].GENE == gene )
                    {
                        obj = {source: "222", snp: cisData222[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 223) {
                for(var k = 0; k < cisData223.length; k++)
                {
                    if(cisData223[k].GENE == gene )
                    {
                        obj = {source: "223", snp: cisData223[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 224) {
                for(var k = 0; k < cisData224.length; k++)
                {
                    if(cisData224[k].GENE == gene )
                    {
                        obj = {source: "224", snp: cisData224[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 225) {
                for(var k = 0; k < cisData225.length; k++)
                {
                    if(cisData225[k].GENE == gene )
                    {
                        obj = {source: "225", snp: cisData225[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 231) {
                for(var k = 0; k < cisData231.length; k++)
                {
                    if(cisData231[k].GENE == gene )
                    {
                        obj = {source: "231", snp: cisData231[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 232) {
                for(var k = 0; k < cisData232.length; k++)
                {
                    if(cisData232[k].GENE == gene )
                    {
                        obj = {source: "232", snp: cisData232[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 233) {
                for(var k = 0; k < cisData233.length; k++)
                {
                    if(cisData233[k].GENE == gene )
                    {
                        obj = {source: "233", snp: cisData233[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 234) {
                for(var k = 0; k < cisData234.length; k++)
                {
                    if(cisData234[k].GENE == gene )
                    {
                        obj = {source: "234", snp: cisData234[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 241) {
                for(var k = 0; k < cisData241.length; k++)
                {
                    if(cisData241[k].GENE == gene )
                    {
                        obj = {source: "241", snp: cisData241[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 242) {
                for(var k = 0; k < cisData242.length; k++)
                {
                    if(cisData242[k].GENE == gene )
                    {
                        obj = {source: "242", snp: cisData242[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 243) {
                for(var k = 0; k < cisData243.length; k++)
                {
                    if(cisData243[k].GENE == gene )
                    {
                        obj = {source: "243", snp: cisData243[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 244) {
                for(var k = 0; k < cisData244.length; k++)
                {
                    if(cisData244[k].GENE == gene )
                    {
                        obj = {source: "244", snp: cisData244[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 245) {
                for(var k = 0; k < cisData245.length; k++)
                {
                    if(cisData245[k].GENE == gene )
                    {
                        obj = {source: "245", snp: cisData245[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 301) {
                for(var k = 0; k < cisData301.length; k++)
                {
                    if(cisData301[k].GENE == gene )
                    {
                        obj = {source: "301", snp: cisData301[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 302) {
                for(var k = 0; k < cisData302.length; k++)
                {
                    if(cisData302[k].GENE == gene )
                    {
                        obj = {source: "302", snp: cisData302[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 401) {
                for(var k = 0; k < cisData401.length; k++)
                {
                    if(cisData401[k].GENE == gene )
                    {
                        obj = {source: "401", snp: cisData401[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 402) {
                for(var k = 0; k < cisData402.length; k++)
                {
                    if(cisData402[k].GENE == gene )
                    {
                        obj = {source: "402", snp: cisData402[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 403) {
                for(var k = 0; k < cisData403.length; k++)
                {
                    if(cisData403[k].GENE == gene )
                    {
                        obj = {source: "403", snp: cisData403[k].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            // Datasets added in July 2020

            else if(datasetvalues[i] == 396) {
                for(var j = 0; j < cisData396.length; j++)
                {
                    if(cisData396[j].GENE == gene ) 
                    {
                        obj = {source: "396", snp: cisData396[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 94) {
                for(var j = 0; j < cisData94.length; j++)
                {
                    if(cisData94[j].GENE == gene ) 
                    {
                        obj = {source: "94", snp: cisData94[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            /* else if(datasetvalues[i] == 191) {
                for(var j = 0; j < cisData191.length; j++)
                {
                    if(cisData191[j].GENE == gene ) 
                    {
                        obj = {source: "191", snp: cisData191[j].SNP}
                        geneCis.push(obj);
                    }
                }
            } */

            else if(datasetvalues[i] == 291) {
                for(var j = 0; j < cisData291.length; j++)
                {
                    if(cisData291[j].GENE == gene ) 
                    {
                        obj = {source: "291", snp: cisData291[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 292) {
                for(var j = 0; j < cisData292.length; j++)
                {
                    if(cisData292[j].GENE == gene ) 
                    {
                        obj = {source: "292", snp: cisData292[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 293) {
                for(var j = 0; j < cisData293.length; j++)
                {
                    if(cisData293[j].GENE == gene ) 
                    {
                        obj = {source: "293", snp: cisData293[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 397) {
                for(var j = 0; j < cisData397.length; j++)
                {
                    if(cisData397[j].GENE == gene ) 
                    {
                        obj = {source: "397", snp: cisData397[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 95) {
                for(var j = 0; j < cisData95.length; j++)
                {
                    if(cisData95[j].GENE == gene ) 
                    {
                        obj = {source: "95", snp: cisData95[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 43) {
                for(var j = 0; j < cisData43.length; j++)
                {
                    if(cisData43[j].GENE == gene ) 
                    {
                        obj = {source: "43", snp: cisData43[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 96) {
                for(var j = 0; j < cisData96.length; j++)
                {
                    if(cisData96[j].GENE == gene ) 
                    {
                        obj = {source: "96", snp: cisData96[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 97) {
                for(var j = 0; j < cisData97.length; j++)
                {
                    if(cisData97[j].GENE == gene ) 
                    {
                        obj = {source: "97", snp: cisData97[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 98) {
                for(var j = 0; j < cisData98.length; j++)
                {
                    if(cisData98[j].GENE == gene ) 
                    {
                        obj = {source: "98", snp: cisData98[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 990) {
                for(var j = 0; j < cisData990.length; j++)
                {
                    if(cisData990[j].GENE == gene ) 
                    {
                        obj = {source: "990", snp: cisData990[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 294) {
                for(var j = 0; j < cisData294.length; j++)
                {
                    if(cisData294[j].GENE == gene ) 
                    {
                        obj = {source: "294", snp: cisData294[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 64) {
                for(var j = 0; j < cisData64.length; j++)
                {
                    if(cisData64[j].GENE == gene ) 
                    {
                        obj = {source: "64", snp: cisData64[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 65) {
                for(var j = 0; j < cisData65.length; j++)
                {
                    if(cisData65[j].GENE == gene ) 
                    {
                        obj = {source: "65", snp: cisData65[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 66) {
                for(var j = 0; j < cisData66.length; j++)
                {
                    if(cisData66[j].GENE == gene ) 
                    {
                        obj = {source: "66", snp: cisData66[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 67) {
                for(var j = 0; j < cisData67.length; j++)
                {
                    if(cisData67[j].GENE == gene ) 
                    {
                        obj = {source: "67", snp: cisData67[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 68) {
                for(var j = 0; j < cisData68.length; j++)
                {
                    if(cisData68[j].GENE == gene ) 
                    {
                        obj = {source: "68", snp: cisData68[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 69) {
                for(var j = 0; j < cisData69.length; j++)
                {
                    if(cisData69[j].GENE == gene ) 
                    {
                        obj = {source: "69", snp: cisData69[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 99) {
                for(var j = 0; j < cisData99.length; j++)
                {
                    if(cisData99[j].GENE == gene ) 
                    {
                        obj = {source: "99", snp: cisData99[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 131) {
                for(var j = 0; j < cisData131.length; j++)
                {
                    if(cisData131[j].GENE == gene ) 
                    {
                        obj = {source: "131", snp: cisData131[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 132) {
                for(var j = 0; j < cisData132.length; j++)
                {
                    if(cisData132[j].GENE == gene ) 
                    {
                        obj = {source: "132", snp: cisData132[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 133) {
                for(var j = 0; j < cisData133.length; j++)
                {
                    if(cisData133[j].GENE == gene ) 
                    {
                        obj = {source: "133", snp: cisData133[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 134) {
                for(var j = 0; j < cisData134.length; j++)
                {
                    if(cisData134[j].GENE == gene ) 
                    {
                        obj = {source: "134", snp: cisData134[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 135) {
                for(var j = 0; j < cisData135.length; j++)
                {
                    if(cisData135[j].GENE == gene ) 
                    {
                        obj = {source: "135", snp: cisData135[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

            else if(datasetvalues[i] == 136) {
                for(var j = 0; j < cisData136.length; j++)
                {
                    if(cisData136[j].GENE == gene ) 
                    {
                        obj = {source: "136", snp: cisData136[j].SNP}
                        geneCis.push(obj);
                    }
                }
            }

           
        }

        var CommonCisData = calcCommonCis(geneCis);

        addLegends();
        

        for (var i = 0; i < datasetvalues.length; i++) {

            if(datasetvalues[i] == 11) {
                igv.browser.loadTrack({
                                    url: './data/fram-sorted.txt.gz',
                                    indexURL: './data/fram-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Euro-American',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    minLogP: 10,
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Framingham'});
            }

            else if(datasetvalues[i] == 001) {
                igv.browser.loadTrack({
                                    url: './data/2020/Lung-cis-sorted.txt.gz', 
                                    indexURL: './data/2020/Lung-cis-sorted.txt.gz.tbi', 
                                    name: 'Lung <br/> European <br/> Meta-analysis', 
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [5e-8, 5e-4, 0.05],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Lung'});
            }

            else if(datasetvalues[i] == 002) {
                igv.browser.loadTrack({
                                    url: './data/2020/gtex-Lung.txt.gz', 
                                    indexURL: './data/2020/gtex-Lung.txt.gz.tbi', 
                                    name: 'Lung <br/> Mixed <br/> GTEx', 
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [5e-8, 5e-4, 0.05],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'GTEx-Lung'});
            }

            else if(datasetvalues[i] == 003) {
                igv.browser.loadTrack({
                                    url: './data/2020/gtex-Prostate.txt.gz', 
                                    indexURL: './data/2020/gtex-Prostate.txt.gz.tbi', 
                                    name: 'Prostate <br/> Mixed <br/> GTEx', 
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [5e-8, 5e-4, 0.05],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'GTEx-Prostate'});
            }

            else if(datasetvalues[i] == 12) {
                igv.browser.loadTrack({
                                    url: './data/2020/gtex-Blood.txt.gz',   // v6: './data/gtex-sorted.txt.gz',
                                    indexURL: './data/2020/gtex-Blood.txt.gz.tbi',  // v6: './data/gtex-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Mixed <br/> GTEx', 
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [5e-8, 5e-4, 0.05],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'GTEx'});
            }

            else if(datasetvalues[i] == 13) {
              

                igv.browser.loadTrack({
                                    url: './data/westra-sorted.txt.gz',
                                    indexURL: './data/westra-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Meta-analysis <br/> European',
                                    format: 'gtexGWAS', 
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Westra'}); 
            }

            else if(datasetvalues[i] == 14) {
                igv.browser.loadTrack({
                                    url: './data/lude/lude-sorted.txt.gz',
                                    indexURL: './data/lude/lude-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Lude'});
            }

            else if(datasetvalues[i] == 15) {
                igv.browser.loadTrack({
                                    url: './data/pierce/pierce-cis-sorted.txt.gz',
                                    indexURL: './data/pierce/pierce-cis-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> East Asian <br/> PLoS Genetics',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    }, 
                                    minLogP: 10,                                
                                    maxLogP: 40,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Pierce'});
            }

            else if(datasetvalues[i] == 16) {
                igv.browser.loadTrack({
                                    url: './data/yamada/yamada-sorted.txt.gz',
                                    indexURL: './data/yamada/yamada-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> East Asian <br/> PLoS One',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Yamada'});
            }

            else if(datasetvalues[i] == 17) {
                igv.browser.loadTrack({
                                    url: './data/eqtlgen/eqtlgen-sorted.txt.gz',
                                    indexURL: './data/eqtlgen/eqtlgen-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Meta-analysis <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    minLogP: 10,
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'eQTLGen'});
            }

            else if(datasetvalues[i] == 18) {
                igv.browser.loadTrack({
                                    url: './data/2020/cage-cis-sorted.txt.gz',
                                    indexURL: './data/2020/cage-cis-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Mixed <br/> CAGE',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    minLogP: 6,
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'CAGE'});
            }

            else if(datasetvalues[i] == 19) {
                igv.browser.loadTrack({
                                    url: './data/2020/life-cis-sorted.txt.gz',
                                    indexURL: './data/2020/life-cis-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Mixed <br/> LIFE-HEART',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    minLogP: 6,
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LIFE-HEART'});
            }

            else if(datasetvalues[i] == 190) {
                igv.browser.loadTrack({
                                    url: './data/2020/UK-blood-cis-sorted.txt.gz',
                                    indexURL: './data/2020/UK-blood-cis-sorted.txt.gz.tbi',
                                    name: 'Whole Blood <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'UK10K'});
            }

            else if(datasetvalues[i] == 21) {             
                igv.browser.loadTrack({
                                    url: './data/jager/monocytes-aa-sorted.txt.gz',
                                    indexURL: './data/jager/monocytes-aa-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> African American',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    }, 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'MonoAA'});
            }

            else if(datasetvalues[i] == 22) {
                igv.browser.loadTrack({
                                    url: './data/jager/monocytes-ea-sorted.txt.gz',
                                    indexURL: './data/jager/monocytes-ea-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> East Asian',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'MonoEA'});
            }

            else if(datasetvalues[i] == 23) {            
                igv.browser.loadTrack({
                                    url: './data/jager/monocytes-eu-sorted.txt.gz',
                                    indexURL: './data/jager/monocytes-eu-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> Euro-American <br/> 377 samples',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'MonoEU'});
            }

            else if(datasetvalues[i] == 25) {           
                igv.browser.loadTrack({
                                    url: './data/fairfax/mono-sorted.txt.gz',
                                    indexURL: './data/fairfax/mono-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'MonoFairfax'});

            }

            else if(datasetvalues[i] == 26) {            
                igv.browser.loadTrack({
                                    url: './data/fairfax/mono-cis-naive-sorted.txt.gz',
                                    indexURL: './data/fairfax/mono-cis-naive-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> Euro-American <br/> 432 samples',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'MonoFairfax-EurAme'});
            }

            else if(datasetvalues[i] == 27) {                
                igv.browser.loadTrack({
                                    url: './data/dice/monocytes-cis-sorted.txt.gz',
                                    indexURL: './data/dice/monocytes-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-monocytes'});
            }

            else if(datasetvalues[i] == 28) {                
                igv.browser.loadTrack({
                                    url: './data/dice/mono-non-cis-sorted.txt.gz',
                                    indexURL: './data/dice/mono-non-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> non-classical <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-mono-nonclassical'});
            }

            else if(datasetvalues[i] == 29) {                
                igv.browser.loadTrack({
                                    url: './data/2020/Mono-Gene-Final-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Mono-Gene-Final-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> Mixed <br/> BLUEPRINT',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-Mixed'});
            }

            else if(datasetvalues[i] == 290) {                
                igv.browser.loadTrack({
                                    url: './data/2020/Mono-Psi-Final-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Mono-Psi-Final-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes - sQTL <br/> Mixed <br/> BLUEPRINT',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-sQTL-Mixed'});
            }

            else if(datasetvalues[i] == 31) {               
                igv.browser.loadTrack({
                                    url: './data/jager/cd4T-aa-sorted.txt.gz',
                                    indexURL: './data/jager/cd4T-aa-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-cells <br/> African American',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-AA'});
            }

            else if(datasetvalues[i] == 32) {           
                igv.browser.loadTrack({
                                    url: './data/jager/cd4T-ea-sorted.txt.gz',
                                    indexURL: './data/jager/cd4T-ea-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-cells <br/> East Asian',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-EA'});
            }

            else if(datasetvalues[i] == 33) {
                igv.browser.loadTrack({
                                    url: './data/jager/cd4T-eu-sorted.txt.gz',
                                    indexURL: './data/jager/cd4T-eu-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-cells <br/> Euro-American',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                   
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-EU'});
            }

            else if(datasetvalues[i] == 35) {                
                igv.browser.loadTrack({
                                    url: './data/dice/CD4-cis-sorted.txt.gz',
                                    indexURL: './data/dice/CD4-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-cells <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-Tcell'});
            }

            else if(datasetvalues[i] == 36) {                
                igv.browser.loadTrack({
                                    url: './data/dice/treg-naive-cis-sorted.txt.gz',
                                    indexURL: './data/dice/treg-naive-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ naive Treg <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-naive-Treg'});
            }

            else if(datasetvalues[i] == 37) {                
                igv.browser.loadTrack({
                                    url: './data/dice/treg-mem-cis-sorted.txt.gz',
                                    indexURL: './data/dice/treg-mem-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ memory Treg <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-memory-Treg'});
            }

            else if(datasetvalues[i] == 38) {                
                igv.browser.loadTrack({
                                    url: './data/dice/th1-cis-sorted.txt.gz',
                                    indexURL: './data/dice/th1-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ Th1 <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-Th1'});
            }

            else if(datasetvalues[i] == 39) {                
                igv.browser.loadTrack({
                                    url: './data/dice/thstar-cis-sorted.txt.gz',
                                    indexURL: './data/dice/thstar-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ Th1/17 <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-Th1/17'});
            }

            else if(datasetvalues[i] == 390) {                
                igv.browser.loadTrack({
                                    url: './data/dice/th17-cis-sorted.txt.gz',
                                    indexURL: './data/dice/th17-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ Th17 <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-Th17'});
            }

            else if(datasetvalues[i] == 391) {                
                igv.browser.loadTrack({
                                    url: './data/dice/th2-cis-sorted.txt.gz',
                                    indexURL: './data/dice/th2-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ Th2 <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-Th2'});
            }

            else if(datasetvalues[i] == 392) {                
                igv.browser.loadTrack({
                                    url: './data/dice/tfh-cis-sorted.txt.gz',
                                    indexURL: './data/dice/tfh-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ memory Tfh <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd4-Tfh'});
            }

            else if(datasetvalues[i] == 393) {                
                igv.browser.loadTrack({
                                    url: './data/2020/Tcell-Gene-Final-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Tcell-Gene-Final-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-Cells <br/> Mixed <br/> BLUEPRINT',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-Mixed'});
            }

            else if(datasetvalues[i] == 394) {                
                igv.browser.loadTrack({
                                    url: './data/2020/Tcell-Psi-Final-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Tcell-Psi-Final-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-Cells - sQTL <br/> Mixed <br/> BLUEPRINT',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-sQTL-Mixed'});
            }

            else if(datasetvalues[i] == 395) {                
                igv.browser.loadTrack({
                                    url: './data/2020/gencord-Tcell-cis-sorted.txt.gz',
                                    indexURL: './data/2020/gencord-Tcell-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-Cells <br/> asQTL <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-asQTL-European'});
            }

            else if(datasetvalues[i] == 41) {                
                igv.browser.loadTrack({
                                    url: './data/fairfax/bcell-sorted.txt.gz',
                                    indexURL: './data/fairfax/bcell-sorted.txt.gz.tbi',
                                    name: 'B-cells <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Bcell'});
            }

            else if(datasetvalues[i] == 42) {                
                igv.browser.loadTrack({
                                    url: './data/dice/bcell-cis-sorted.txt.gz',
                                    indexURL: './data/dice/bcell-cis-sorted.txt.gz.tbi',
                                    name: 'B-cells <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-Bcell'});
            }

            else if(datasetvalues[i] == 51) {            
                igv.browser.loadTrack({
                                    url: './data/fairfax/neutrophils-cis-final-sorted.txt.gz',
                                    indexURL: './data/fairfax/neutrophils-cis-final-sorted.txt.gz.tbi',
                                    name: 'CD16+ Neutrophils <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                 
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Neutrophils'});
            }

            else if(datasetvalues[i] == 52) {                
                igv.browser.loadTrack({
                                    url: './data/2020/Neut-Gene-Final-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Neut-Gene-Final-cis-sorted.txt.gz.tbi',
                                    name: 'CD16+ Neutrophils <br/> Mixed <br/> BLUEPRINT',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Neut-Mixed'});
            }

            else if(datasetvalues[i] == 53) {                
                igv.browser.loadTrack({
                                    url: './data/2020/Neut-Psi-Final-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Neut-Psi-Final-cis-sorted.txt.gz.tbi',
                                    name: 'CD16+ Neutrophils - sQTL <br/> Mixed <br/> BLUEPRINT',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Neut-sQTL-Mixed'});
            }
       
            else if(datasetvalues[i] == 61) {              
                igv.browser.loadTrack({
                                    url: './data/fairfax/mono-cis-lps2-sorted.txt.gz',
                                    indexURL: './data/fairfax/mono-cis-lps2-sorted.txt.gz.tbi',
                                    name: 'LPS - 2 hours <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-LPS2h-EurAme'});           
            }

            else if(datasetvalues[i] == 62) {              
               igv.browser.loadTrack({
                                    url: './data/fairfax/mono-cis-lps24-sorted.txt.gz',
                                    indexURL: './data/fairfax/mono-cis-lps24-sorted.txt.gz.tbi',
                                    name: 'LPS - 24 hours <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-LPS24h-EurAme'});               
            }

            else if(datasetvalues[i] == 63) {          
                igv.browser.loadTrack({
                                    url: './data/fairfax/mono-cis-ifn-sorted.txt.gz',
                                    indexURL: './data/fairfax/mono-cis-ifn-sorted.txt.gz.tbi',
                                    name: 'IFN-gamma <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-IFNgamma-EurAme'});
            }

            else if(datasetvalues[i] == 71) {              
                igv.browser.loadTrack({
                                    url: './data/dermitzakis/euro-temp-sorted.txt.gz',
                                    indexURL: './data/dermitzakis/euro-temp-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-European'});           
            }

            else if(datasetvalues[i] == 72) {              
               igv.browser.loadTrack({
                                    url: './data/dermitzakis/yri-temp-sorted.txt.gz',
                                    indexURL: './data/dermitzakis/yri-temp-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> African',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-African'});               
            }

            else if(datasetvalues[i] == 73) {              
               igv.browser.loadTrack({
                                    url: './data/2020/muther-lcl-cis-sorted.txt.gz',
                                    indexURL: './data/2020/muther-lcl-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> European (twins)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-European (twins)'});               
            }

            else if(datasetvalues[i] == 74) {              
               igv.browser.loadTrack({
                                    url: './data/2020/ceu-cis-sorted.txt.gz',
                                    indexURL: './data/2020/ceu-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> N/W European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-N/W European'});               
            }

            else if(datasetvalues[i] == 75) {              
               igv.browser.loadTrack({
                                    url: './data/2020/chb-cis-sorted.txt.gz',
                                    indexURL: './data/2020/chb-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Han Chinese',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-Han Chinese'});               
            }

            else if(datasetvalues[i] == 76) {              
               igv.browser.loadTrack({
                                    url: './data/2020/gih-cis-sorted.txt.gz',
                                    indexURL: './data/2020/gih-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Gujarati Indians',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-Gujarati Indians'});               
            }

            else if(datasetvalues[i] == 77) {              
               igv.browser.loadTrack({
                                    url: './data/2020/jpt-cis-sorted.txt.gz',
                                    indexURL: './data/2020/jpt-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Japanese',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-Japanese'});               
            }

            else if(datasetvalues[i] == 78) {              
               igv.browser.loadTrack({
                                    url: './data/2020/lwk-cis-sorted.txt.gz',
                                    indexURL: './data/2020/lwk-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Luhya Kenya',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-Luhya Kenya'});               
            }

            else if(datasetvalues[i] == 79) {              
               igv.browser.loadTrack({
                                    url: './data/2020/yri-cis-sorted.txt.gz',
                                    indexURL: './data/2020/yri-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Yoruba Nigeria',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-Yoruba Nigeria'});               
            }

            else if(datasetvalues[i] == 790) {              
               igv.browser.loadTrack({
                                    url: './data/2020/UK-lcl-cis-sorted.txt.gz',
                                    indexURL: './data/2020/UK-lcl-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-UK10K'});               
            }

            else if(datasetvalues[i] == 791) {              
               igv.browser.loadTrack({
                                    url: './data/2020/alspac-cis-final-sorted.txt.gz',
                                    indexURL: './data/2020/alspac-cis-final-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> Mixed <br/> ALSPAC',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-ALSPAC'});               
            }

            else if(datasetvalues[i] == 792) {              
               igv.browser.loadTrack({
                                    url: './data/2020/gencord-lcl-cis-sorted.txt.gz',
                                    indexURL: './data/2020/gencord-lcl-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> asQTL <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-Gencord'});               
            }

            else if(datasetvalues[i] == 81) {              
                igv.browser.loadTrack({
                                    url: './data/heijmans/meqtl-sorted.txt.gz',
                                    indexURL: './data/heijmans/meqtl-sorted.txt.gz.tbi',
                                    name: 'meQTL <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'meQTL'});           
            }

            else if(datasetvalues[i] == 82) {              
               igv.browser.loadTrack({
                                    url: './data/heijmans/eqtm-sorted.txt.gz',
                                    indexURL: './data/heijmans/eqtm-sorted.txt.gz.tbi',
                                    name: 'eQTM <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'eQTM'});               
            }

            else if(datasetvalues[i] == 83) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Mono-Meth-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Mono-Meth-Full-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> meQTL',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Monocytes-meQTL'});               
            }

            else if(datasetvalues[i] == 84) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Mono-K4ME1-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Mono-K4ME1-Full-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> hmQTL(H3K4me1)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Monocytes-hmQTL(H3K4me1)'});               
            }

            else if(datasetvalues[i] == 85) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Mono-K27AC-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Mono-K27AC-Full-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> hmQTL(H3K27ac)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Monocytes-hmQTL(H3K27ac)'});               
            }

            else if(datasetvalues[i] == 86) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Neut-Meth-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Neut-Meth-Full-cis-sorted.txt.gz.tbi',
                                    name: 'Neutrophils <br/> meQTL',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Neutrophils-meQTL'});               
            }

            else if(datasetvalues[i] == 87) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Neut-K4ME1-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Neut-K4ME1-Full-cis-sorted.txt.gz.tbi',
                                    name: 'Neutrophils <br/> hmQTL(H3K4me1)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Neutrophils-hmQTL(H3K4me1)'});               
            }

            else if(datasetvalues[i] == 88) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Neut-K27AC-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Neut-K27AC-Full-cis-sorted.txt.gz.tbi',
                                    name: 'Neutrophils <br/> hmQTL(H3K27ac)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Neutrophils-hmQTL(H3K27ac)'});               
            }

            else if(datasetvalues[i] == 89) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Tcell-Meth-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Tcell-Meth-Full-cis-sorted.txt.gz.tbi',
                                    name: 'CD4 T-cells <br/> meQTL',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'CD4 T-cells-meQTL'});               
            }

            else if(datasetvalues[i] == 890) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Tcell-K4ME1-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Tcell-K4ME1-Full-cis-sorted.txt.gz.tbi',
                                    name: 'CD4 T-cells <br/> hmQTL(H3K4me1)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'CD4 T-cells-hmQTL(H3K4me1)'});               
            }

            else if(datasetvalues[i] == 891) {              
               igv.browser.loadTrack({
                                    url: './data/2020/Tcell-K27AC-Full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/Tcell-K27AC-Full-cis-sorted.txt.gz.tbi',
                                    name: 'CD4 T-cells <br/> hmQTL(H3K27ac)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'CD4 T-cells-hmQTL(H3K27ac)'});               
            }

            else if(datasetvalues[i] == 892) {              
               igv.browser.loadTrack({
                                    url: './data/2020/gilad-full-cis-sorted.txt.gz',
                                    indexURL: './data/2020/gilad-full-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> meQTL(Yoruba)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-meQTL(Yoruba)'});               
            }

            else if(datasetvalues[i] == 893) {              
               igv.browser.loadTrack({
                                    url: './data/2020/chromo-cis-sorted.txt.gz',
                                    indexURL: './data/2020/chromo-cis-sorted.txt.gz.tbi',
                                    name: 'LCL <br/> hmQTL(Yoruba)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'LCL-hmQTL(Yoruba)'});               
            }

            else if(datasetvalues[i] == 91) {                
                igv.browser.loadTrack({
                                    url: './data/dice/nk-cis-sorted.txt.gz',
                                    indexURL: './data/dice/nk-cis-sorted.txt.gz.tbi',
                                    name: 'CD56dim CD16+ NK <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd16+-NK'});
            }

            else if(datasetvalues[i] == 92) {                
                igv.browser.loadTrack({
                                    url: './data/dice/CD8-cis-sorted.txt.gz',
                                    indexURL: './data/dice/CD8-cis-sorted.txt.gz.tbi',
                                    name: 'CD8+ T-cell <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DICE-cd8-Tcell'});
            }

            else if(datasetvalues[i] == 93) {                
                igv.browser.loadTrack({
                                    url: './data/2020/gencord-fibro-cis-sorted.txt.gz',
                                    indexURL: './data/2020/gencord-fibro-cis-sorted.txt.gz.tbi',
                                    name: 'Fibroblast <br/> asQTL <br/> European',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Fibroblast-European'});
            }

            else if(datasetvalues[i] == 101) {  

                var dataURL = userCisFileLink[0];
                var indURL = userCisFileLink[0] + ".tbi";
           
                igv.browser.loadTrack({
                                    url: dataURL, 
                                    indexURL: indURL, 
                                    name: userCisFilename[0],
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: userCisFilename[0]});           
            }

            else if(datasetvalues[i] == 102) {    

                var dataURL = userCisFileLink[1];
                var indURL = userCisFileLink[1] + ".tbi";

               igv.browser.loadTrack({
                                    url: dataURL,
                                    indexURL: indURL,
                                    name: userCisFilename[1],
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: userCisFilename[1]});               
            }

            else if(datasetvalues[i] == 103) {  

                var dataURL = userCisFileLink[2];
                var indURL = userCisFileLink[2] + ".tbi";            
                igv.browser.loadTrack({
                                    url: dataURL,
                                    indexURL: indURL,
                                    name: userCisFilename[2],
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: userCisFilename[2]});           
            }

            else if(datasetvalues[i] == 211) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/mayo/mayo-day0-sorted.txt.gz',
                                    indexURL: './data/hipc/mayo/mayo-day0-sorted.txt.gz.tbi',
                                    name: 'Mayo <br/> Day0',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mayo-Day0'});
            }

            else if(datasetvalues[i] == 212) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/mayo/mayo-day3-sorted.txt.gz',
                                    indexURL: './data/hipc/mayo/mayo-day3-sorted.txt.gz.tbi',
                                    name: 'Mayo <br/> Day3',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mayo-Day3'});
            }

            else if(datasetvalues[i] == 213) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/mayo/mayo-day28-sorted.txt.gz',
                                    indexURL: './data/hipc/mayo/mayo-day28-sorted.txt.gz.tbi',
                                    name: 'Mayo <br/> Day28',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mayo-Day28'});
            }

            else if(datasetvalues[i] == 221) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/nih/nih-day-7-sorted.txt.gz',
                                    indexURL: './data/hipc/nih/nih-day-7-sorted.txt.gz.tbi',
                                    name: 'NIH <br/> Day-7',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'NIH-Day-7'});
            }

            else if(datasetvalues[i] == 222) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/nih/nih-day0-sorted.txt.gz',
                                    indexURL: './data/hipc/nih/nih-day0-sorted.txt.gz.tbi',
                                    name: 'NIH <br/> Day0',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'NIH-Day0'});
            }

            else if(datasetvalues[i] == 223) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/nih/nih-day1-sorted.txt.gz',
                                    indexURL: './data/hipc/nih/nih-day1-sorted.txt.gz.tbi',
                                    name: 'NIH <br/> Day1',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'NIH-Day1'});
            }

            else if(datasetvalues[i] == 224) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/nih/nih-day7-sorted.txt.gz',
                                    indexURL: './data/hipc/nih/nih-day7-sorted.txt.gz.tbi',
                                    name: 'NIH <br/> Day7',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'NIH-Day7'});
            }

            else if(datasetvalues[i] == 225) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/nih/nih-day70-sorted.txt.gz',
                                    indexURL: './data/hipc/nih/nih-day70-sorted.txt.gz.tbi',
                                    name: 'NIH <br/> Day70',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'NIH-Day70'});
            }

            else if(datasetvalues[i] == 231) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/yale/yale-day0-sorted.txt.gz',
                                    indexURL: './data/hipc/yale/yale-day0-sorted.txt.gz.tbi',
                                    name: 'Yale <br/> Day0',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Yale-Day0'});
            }

            else if(datasetvalues[i] == 232) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/yale/yale-day2-sorted.txt.gz',
                                    indexURL: './data/hipc/yale/yale-day2-sorted.txt.gz.tbi',
                                    name: 'Yale <br/> Day2',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Yale-Day2'});
            }

            else if(datasetvalues[i] == 233) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/yale/yale-day7-sorted.txt.gz',
                                    indexURL: './data/hipc/yale/yale-day7-sorted.txt.gz.tbi',
                                    name: 'Yale <br/> Day7',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Yale-Day7'});
            }

            else if(datasetvalues[i] == 234) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/yale/yale-day28-sorted.txt.gz',
                                    indexURL: './data/hipc/yale/yale-day28-sorted.txt.gz.tbi',
                                    name: 'Yale <br/> Day28',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Yale-Day28'});
            }

            else if(datasetvalues[i] == 241) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/emory/emory-day0-sorted.txt.gz',
                                    indexURL: './data/hipc/emory/emory-day0-sorted.txt.gz.tbi',
                                    name: 'Emory <br/> Day0',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Emory-Day0'});
            }

            else if(datasetvalues[i] == 242) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/emory/emory-day1-sorted.txt.gz',
                                    indexURL: './data/hipc/emory/emory-day1-sorted.txt.gz.tbi',
                                    name: 'Emory <br/> Day1',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Emory-Day1'});
            }

            else if(datasetvalues[i] == 243) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/emory/emory-day3-sorted.txt.gz',
                                    indexURL: './data/hipc/emory/emory-day3-sorted.txt.gz.tbi',
                                    name: 'Emory <br/> Day3',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Emory-Day3'});
            }

            else if(datasetvalues[i] == 244) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/emory/emory-day7-sorted.txt.gz',
                                    indexURL: './data/hipc/emory/emory-day7-sorted.txt.gz.tbi',
                                    name: 'Emory <br/> Day7',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Emory-Day7'});
            }

            else if(datasetvalues[i] == 245) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/emory/emory-day14-sorted.txt.gz',
                                    indexURL: './data/hipc/emory/emory-day14-sorted.txt.gz.tbi',
                                    name: 'Emory <br/> Day14',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Emory-Day14'});
            }

            else if(datasetvalues[i] == 301) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/chikv/chik-day1-sorted.txt.gz',
                                    indexURL: './data/hipc/chikv/chik-day1-sorted.txt.gz.tbi',
                                    name: 'CHIKV <br/> Acute',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'CHIKV-Day1'});
            }

            else if(datasetvalues[i] == 302) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/chikv/chik-day9-sorted.txt.gz',
                                    indexURL: './data/hipc/chikv/chik-day9-sorted.txt.gz.tbi',
                                    name: 'CHIKV <br/> Convalescent',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'CHIKV-Day9'});
            }

            else if(datasetvalues[i] == 401) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/zikv/zika-acute-cis-sorted.txt.gz',
                                    indexURL: './data/hipc/zikv/zika-acute-cis-sorted.txt.gz.tbi',
                                    name: 'ZIKV <br/> Acute',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'ZIKV-Acute'});
            }

            else if(datasetvalues[i] == 402) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/zikv/zika-lateacute-cis-sorted.txt.gz',
                                    indexURL: './data/hipc/zikv/zika-lateacute-cis-sorted.txt.gz.tbi',
                                    name: 'ZIKV <br/> Late Acute',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'ZIKV-LateAcute'});
            }

            else if(datasetvalues[i] == 403) {                
                igv.browser.loadTrack({
                                    url: './data/hipc/zikv/zika-conv-cis-sorted.txt.gz',
                                    indexURL: './data/hipc/zikv/zika-conv-cis-sorted.txt.gz.tbi',
                                    name: 'ZIKV <br/> Convalescent',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'ZIKV-Conv'});
            }

            // Datasets added in July 2020

            else if(datasetvalues[i] == 396) {                
                igv.browser.loadTrack({
                                    url: './data/2021/estonia-CD4-cis-sorted.txt.gz',
                                    indexURL: './data/2021/estonia-CD4-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-Cells <br/> European <br/> Estonian Biobank',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-European-Estonia'});
            }

            else if(datasetvalues[i] == 94) {                
                igv.browser.loadTrack({
                                    url: './data/2021/estonia-CD8-cis-sorted.txt.gz',
                                    indexURL: './data/2021/estonia-CD8-cis-sorted.txt.gz.tbi',
                                    name: 'CD8+ T-Cells <br/> European <br/> Estonian Biobank',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd8T-European-Estonia'});
            }

            else if(datasetvalues[i] == 291) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-mono-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-mono-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 292) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-cmono-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-cmono-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes - classical <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cMono-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 293) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-ncmono-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-ncmono-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes - non-classical <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'ncMono-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 397) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-cd4-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-cd4-cis-sorted.txt.gz.tbi',
                                    name: 'CD4+ T-Cells <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd4T-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 95) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-cd8-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-cd8-cis-sorted.txt.gz.tbi',
                                    name: 'CD8+ T-Cells <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'cd8T-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 43) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-bcells-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-bcells-cis-sorted.txt.gz.tbi',
                                    name: 'B-Cells <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'bcells-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 96) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-nk-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-nk-cis-sorted.txt.gz.tbi',
                                    name: 'NK Cells <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'nk-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 97) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-pbmc-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-pbmc-cis-sorted.txt.gz.tbi',
                                    name: 'PBMC <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'pbmc-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 98) {                
                igv.browser.loadTrack({
                                    url: './data/2021/lifelines-dend-cis-sorted.txt.gz',
                                    indexURL: './data/2021/lifelines-dend-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> European <br/> Lifelines (single-cell)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'DC-European-Lifelines(sc)'});
            }

            else if(datasetvalues[i] == 990) {                
                igv.browser.loadTrack({
                                    url: './data/2021/ss-mdm-cis-sorted.txt.gz',
                                    indexURL: './data/2021/ss-mdm-cis-sorted.txt.gz.tbi',
                                    name: 'MDMs <br/> Mixed <br/> Systemic Sclerosis',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'MDM-Mixed-SS'});
            }

            else if(datasetvalues[i] == 294) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-ctrl-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-ctrl-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> ImmunPop',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-ImmunPop'});
            }

            else if(datasetvalues[i] == 64) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-lps90h-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-lps90h-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> LPS-90mins',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-LPS90'});
            }

            else if(datasetvalues[i] == 65) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-lps6h-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-lps6h-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> LPS-6hours',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-LPS6h'});
            }

            else if(datasetvalues[i] == 66) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-mdp90h-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-mdp90h-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> MDP-90mins',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-MDP90'});
            }

            else if(datasetvalues[i] == 67) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-mdp6h-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-mdp6h-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> MDP-6hours',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-MDP6h'});
            }

            else if(datasetvalues[i] == 68) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-rna90h-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-rna90h-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> RNA-90mins',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-RNA90'});
            }

            else if(datasetvalues[i] == 69) {                
                igv.browser.loadTrack({
                                    url: './data/2021/impop-rna6h-cis-sorted.txt.gz',
                                    indexURL: './data/2021/impop-rna6h-cis-sorted.txt.gz.tbi',
                                    name: 'Monocytes <br/> European <br/> RNA-6hours',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Mono-European-RNA6h'});
            }

            else if(datasetvalues[i] == 99) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-baseline-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-baseline-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed'});
            }

            else if(datasetvalues[i] == 131) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-lps-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-lps-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed <br/> LPS',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed-LPS'});
            }

            else if(datasetvalues[i] == 132) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-re-lps-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-re-lps-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed <br/> LPS (reQTLs)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed-LPSre'});
            }

            else if(datasetvalues[i] == 133) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-flu-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-flu-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed <br/> Influenza',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed-FLU'});
            }

            else if(datasetvalues[i] == 134) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-re-flu-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-re-flu-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed <br/> Influenza (reQTLs)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed-FLUre'});
            }

            else if(datasetvalues[i] == 135) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-ifnb-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-ifnb-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed <br/> IFNbeta',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed-IFNb'});
            }

            else if(datasetvalues[i] == 136) {                
                igv.browser.loadTrack({
                                    url: './data/2021/pat-re-ifnb-cis-sorted.txt.gz',
                                    indexURL: './data/2021/pat-re-ifnb-cis-sorted.txt.gz.tbi',
                                    name: 'Dendritic Cells <br/> Mixed <br/> IFNbeta (reQTLs)',
                                    format: 'gtexGWAS',
                                    sourceType: 'file',
                                    type: 'gwas',
                                    colorScale: {
                                        thresholds: [],
                                        colors:['rgb(30,144,255)']
                                    },                                  
                                    maxLogP: 16,
                                    height: 150,
                                    commonCis: CommonCisData,
                                    label: 'Dend-Mixed-IFNbre'});
            }
           

        }


    }


    function calcCommonCis(geneCis)
    {
        var frequency = {}, source = {}, value, src;

        // compute frequencies of each value
        for(var i = 0; i < geneCis.length; i++) {
            value = geneCis[i].snp;
            src = geneCis[i].source;
            
            if(value in frequency) {
                if(source[value] != src)
                    frequency[value]++;
            }
            
            else {
                frequency[value] = 1;
                source[value] = src;
            }
        }

        var common = [];

        for(value in frequency) {
            if(frequency[value] >= 2)
                common.push(value);
        }

        return common;
    }


    function calcCommonTrans(geneSetData)
    {
        var snpData = [], junkArr = [], snpObj = {}, frequency = {}, source = {}, value, gene, pValue, snppos, genepos, geneObj = {}, index = {}, snp = [];

        // compute frequencies of each value
        for(var i = 0; i < geneSetData.length; i++) {
            value = geneSetData[i].SNPName;
            snp = [];
            //snp.push(value);
            gene = geneSetData[i].Gene;
            pValue = geneSetData[i].PValue;
            snppos = geneSetData[i].SNPChr + ":" + geneSetData[i].SNPChrPos; 
            genepos = geneSetData[i].ProbeChr + ":" + geneSetData[i].ProbeCenterChrPos;

            if(value in frequency) {
                if(source[value].Gene != gene)
                    frequency[value]++;
                    
                geneObj = {Pvalue: pValue, rsid: value, SNPPos: snppos, Gene: gene, GenePos: genepos};
                //snp.push(geneObj);
                source[value].push(geneObj);
            }
            
            else {
                frequency[value] = 1;
                geneObj = {Pvalue: pValue, rsid: value, SNPPos: snppos, Gene: gene, GenePos: genepos};
                
                snp.push(geneObj);
                source[value] = snp; 
                //index[value] = i;
            }
        }

        var common = [];

        for(value in frequency) {
            if(frequency[value] >= 2)
                ;
        }

        //return common;
        geneSetTrans(frequency, source);
    }


</script>

</body>

</html>
